// @privy-io/public-api/dist/index.mjs
import $83Qas$zod, {z as $83Qas$z} from "zod";
import {InvalidInputError as $83Qas$InvalidInputError, PrivyErrorCode as $83Qas$PrivyErrorCode, UnauthorizedError as $83Qas$UnauthorizedError, LegacyInvalidInputError as $83Qas$LegacyInvalidInputError} from "@privy-io/api-base";
import $83Qas$bs58 from "bs58";
import {ethers as $83Qas$ethers} from "ethers";
import {parsePhoneNumber as $83Qas$parsePhoneNumber} from "libphonenumber-js/max";


function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || Object.prototype.hasOwnProperty.call(dest, key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $852b49dc7ea96c7c$exports = {};
var $b8155828bfbee467$exports = {};

$parcel$export($b8155828bfbee467$exports, "AnalyticsEvent", () => $b8155828bfbee467$export$30b7662faa307ee);
const $b8155828bfbee467$export$30b7662faa307ee = {
    path: "/api/v1/analytics_events",
    method: "POST"
};


var $bf4e8a2ca74ea9d9$exports = {};

$parcel$export($bf4e8a2ca74ea9d9$exports, "AppConfig", () => $bf4e8a2ca74ea9d9$export$d67e1898834e3885);
const $bf4e8a2ca74ea9d9$export$d67e1898834e3885 = {
    path: "/api/v1/apps/:app_id",
    method: "GET"
};


var $97c9540320dc068b$exports = {};

$parcel$export($97c9540320dc068b$exports, "CustomJWTAuthenticate", () => $97c9540320dc068b$export$cca9af2746f9b569);
const $97c9540320dc068b$export$cca9af2746f9b569 = {
    path: "/api/v1/custom_jwt_account/authenticate",
    method: "POST"
};


var $4a90d090fac716ad$exports = {};

$parcel$export($4a90d090fac716ad$exports, "FarcasterInit", () => $4a90d090fac716ad$export$7ed9dd4c28ecc8b6);
$parcel$export($4a90d090fac716ad$exports, "FarcasterAuthenticate", () => $4a90d090fac716ad$export$c5985ffc97c7fdbb);
$parcel$export($4a90d090fac716ad$exports, "FarcasterLink", () => $4a90d090fac716ad$export$d51549470a941fce);
$parcel$export($4a90d090fac716ad$exports, "FarcasterUnlink", () => $4a90d090fac716ad$export$6017b5ccb660b0c8);
$parcel$export($4a90d090fac716ad$exports, "FarcasterStatus", () => $4a90d090fac716ad$export$a14f766edcdfe001);
const $4a90d090fac716ad$export$7ed9dd4c28ecc8b6 = {
    path: "/api/v1/farcaster/init",
    method: "POST"
};
const $4a90d090fac716ad$export$c5985ffc97c7fdbb = {
    path: "/api/v1/farcaster/authenticate",
    method: "POST"
};
const $4a90d090fac716ad$export$d51549470a941fce = {
    path: "/api/v1/farcaster/link",
    method: "POST"
};
const $4a90d090fac716ad$export$6017b5ccb660b0c8 = {
    path: "/api/v1/farcaster/unlink",
    method: "POST"
};
const $4a90d090fac716ad$export$a14f766edcdfe001 = {
    path: "/api/v1/farcaster/status",
    method: "GET"
};


var $2d6a6e81ed9cd5aa$exports = {};

$parcel$export($2d6a6e81ed9cd5aa$exports, "GuestAuthenticate", () => $2d6a6e81ed9cd5aa$export$bed08221a49a43ad);
const $2d6a6e81ed9cd5aa$export$bed08221a49a43ad = {
    path: "/api/v1/guest/authenticate",
    method: "POST"
};


var $93c65b76e5ca94a2$exports = {};

$parcel$export($93c65b76e5ca94a2$exports, "MfaPasswordlessSmsEnroll", () => $93c65b76e5ca94a2$export$2ca9d9da76a9e126);
$parcel$export($93c65b76e5ca94a2$exports, "MfaPasswordlessSmsVerify", () => $93c65b76e5ca94a2$export$3308cba1d8516f90);
$parcel$export($93c65b76e5ca94a2$exports, "MfaPasswordlessSmsInit", () => $93c65b76e5ca94a2$export$cf4de95cb6da600f);
$parcel$export($93c65b76e5ca94a2$exports, "MfaPasswordlessSmsUnenroll", () => $93c65b76e5ca94a2$export$503f112538d6ca09);
const $93c65b76e5ca94a2$export$2ca9d9da76a9e126 = {
    path: "/api/v1/mfa/passwordless_sms/enroll",
    method: "POST"
};
const $93c65b76e5ca94a2$export$3308cba1d8516f90 = {
    path: "/api/v1/mfa/passwordless_sms/verify",
    method: "POST"
};
const $93c65b76e5ca94a2$export$cf4de95cb6da600f = {
    path: "/api/v1/mfa/passwordless_sms/init",
    method: "POST"
};
const $93c65b76e5ca94a2$export$503f112538d6ca09 = {
    path: "/api/v1/mfa/passwordless_sms/unenroll",
    method: "POST"
};


var $0c1aa725aa4f001d$exports = {};

$parcel$export($0c1aa725aa4f001d$exports, "MfaTotpEnroll", () => $0c1aa725aa4f001d$export$8cde043d756d30a7);
$parcel$export($0c1aa725aa4f001d$exports, "MfaTotpVerify", () => $0c1aa725aa4f001d$export$68c952ebd3c622ec);
$parcel$export($0c1aa725aa4f001d$exports, "MfaTotpInit", () => $0c1aa725aa4f001d$export$9cfb70b9ef75b776);
$parcel$export($0c1aa725aa4f001d$exports, "MfaTotpUnenroll", () => $0c1aa725aa4f001d$export$2241a8a66d2e6498);
const $0c1aa725aa4f001d$export$8cde043d756d30a7 = {
    path: "/api/v1/mfa/totp/enroll",
    method: "POST"
};
const $0c1aa725aa4f001d$export$68c952ebd3c622ec = {
    path: "/api/v1/mfa/totp/verify",
    method: "POST"
};
const $0c1aa725aa4f001d$export$9cfb70b9ef75b776 = {
    path: "/api/v1/mfa/totp/init",
    method: "POST"
};
const $0c1aa725aa4f001d$export$2241a8a66d2e6498 = {
    path: "/api/v1/mfa/totp/unenroll",
    method: "POST"
};


var $918825365cfa7807$exports = {};

$parcel$export($918825365cfa7807$exports, "OAuthAuthenticate", () => $918825365cfa7807$export$866d3936787f21ca);
$parcel$export($918825365cfa7807$exports, "OAuthInit", () => $918825365cfa7807$export$7dd94108c3c957b6);
$parcel$export($918825365cfa7807$exports, "OAuthLink", () => $918825365cfa7807$export$b555c05c68965095);
$parcel$export($918825365cfa7807$exports, "OAuthUnlink", () => $918825365cfa7807$export$7246a40b726a26ea);
const $918825365cfa7807$export$866d3936787f21ca = {
    path: "/api/v1/oauth/authenticate",
    method: "POST"
};
const $918825365cfa7807$export$7dd94108c3c957b6 = {
    path: "/api/v1/oauth/init",
    method: "POST"
};
const $918825365cfa7807$export$b555c05c68965095 = {
    path: "/api/v1/oauth/link",
    method: "POST"
};
const $918825365cfa7807$export$7246a40b726a26ea = {
    path: "/api/v1/oauth/unlink",
    method: "POST"
};


var $7968453470d2e50e$exports = {};

$parcel$export($7968453470d2e50e$exports, "PasswordlessAuthenticate", () => $7968453470d2e50e$export$2399ea1e38345a53);
$parcel$export($7968453470d2e50e$exports, "PasswordlessInit", () => $7968453470d2e50e$export$90dfa8041dde28f7);
$parcel$export($7968453470d2e50e$exports, "PasswordlessLink", () => $7968453470d2e50e$export$c9ba5b5d6d996387);
$parcel$export($7968453470d2e50e$exports, "PasswordlessUnlink", () => $7968453470d2e50e$export$770e866201444bcd);
$parcel$export($7968453470d2e50e$exports, "PasswordlessUpdate", () => $7968453470d2e50e$export$95d6a76319eb3229);
const $7968453470d2e50e$export$2399ea1e38345a53 = {
    path: "/api/v1/passwordless/authenticate",
    method: "POST"
};
const $7968453470d2e50e$export$90dfa8041dde28f7 = {
    path: "/api/v1/passwordless/init",
    method: "POST"
};
const $7968453470d2e50e$export$c9ba5b5d6d996387 = {
    path: "/api/v1/passwordless/link",
    method: "POST"
};
const $7968453470d2e50e$export$770e866201444bcd = {
    path: "/api/v1/passwordless/unlink",
    method: "POST"
};
const $7968453470d2e50e$export$95d6a76319eb3229 = {
    path: "/api/v1/passwordless/update",
    method: "POST"
};


var $ff2f0b212f862ec3$exports = {};

$parcel$export($ff2f0b212f862ec3$exports, "PasswordlessSmsAuthenticate", () => $ff2f0b212f862ec3$export$7e171ae579c30d39);
$parcel$export($ff2f0b212f862ec3$exports, "PasswordlessSmsInit", () => $ff2f0b212f862ec3$export$5c9a94090bb8ee8b);
$parcel$export($ff2f0b212f862ec3$exports, "PasswordlessSmsLink", () => $ff2f0b212f862ec3$export$e1b8c2813604d498);
$parcel$export($ff2f0b212f862ec3$exports, "PasswordlessSmsUnlink", () => $ff2f0b212f862ec3$export$140d5f525bd53d81);
$parcel$export($ff2f0b212f862ec3$exports, "PasswordlessSmsUpdate", () => $ff2f0b212f862ec3$export$3a4526cc549be74f);
const $ff2f0b212f862ec3$export$7e171ae579c30d39 = {
    path: "/api/v1/passwordless_sms/authenticate",
    method: "POST"
};
const $ff2f0b212f862ec3$export$5c9a94090bb8ee8b = {
    path: "/api/v1/passwordless_sms/init",
    method: "POST"
};
const $ff2f0b212f862ec3$export$e1b8c2813604d498 = {
    path: "/api/v1/passwordless_sms/link",
    method: "POST"
};
const $ff2f0b212f862ec3$export$140d5f525bd53d81 = {
    path: "/api/v1/passwordless_sms/unlink",
    method: "POST"
};
const $ff2f0b212f862ec3$export$3a4526cc549be74f = {
    path: "/api/v1/passwordless_sms/update",
    method: "POST"
};


var $2047bf8d239cddb9$exports = {};

$parcel$export($2047bf8d239cddb9$exports, "RecoveryKeyMaterial", () => $2047bf8d239cddb9$export$53205c78d6e0d37f);
$parcel$export($2047bf8d239cddb9$exports, "RecoveryOAuthInit", () => $2047bf8d239cddb9$export$f75c08c106aa5263);
$parcel$export($2047bf8d239cddb9$exports, "RecoveryOAuthAuthenticate", () => $2047bf8d239cddb9$export$c4a18348874706ae);
$parcel$export($2047bf8d239cddb9$exports, "RecoveryOAuthInitICloud", () => $2047bf8d239cddb9$export$4af466c8016ee40a);
$parcel$export($2047bf8d239cddb9$exports, "RecoveryOAuthCallbackICloudExpo", () => $2047bf8d239cddb9$export$7b1580cc69e35e8);
$parcel$export($2047bf8d239cddb9$exports, "RecoveryConfigurationICloud", () => $2047bf8d239cddb9$export$bce86382d6c75077);
const $2047bf8d239cddb9$export$53205c78d6e0d37f = {
    path: "/api/v1/embedded_wallets/:address/recovery/key_material",
    method: "POST"
};
const $2047bf8d239cddb9$export$f75c08c106aa5263 = {
    path: "/api/v1/recovery/oauth/init",
    method: "POST"
};
const $2047bf8d239cddb9$export$c4a18348874706ae = {
    path: "/api/v1/recovery/oauth/authenticate",
    method: "POST"
};
const $2047bf8d239cddb9$export$4af466c8016ee40a = {
    path: "/api/v1/recovery/oauth/init_icloud",
    method: "POST"
};
const $2047bf8d239cddb9$export$7b1580cc69e35e8 = {
    path: "/api/v1/recovery/oauth/callback_icloud/:app_id/expo_ios",
    method: "GET"
};
const $2047bf8d239cddb9$export$bce86382d6c75077 = {
    path: "/api/v1/recovery/configuration_icloud",
    method: "POST"
};


var $48277fa58f01a4d6$exports = {};

$parcel$export($48277fa58f01a4d6$exports, "RefreshSession", () => $48277fa58f01a4d6$export$99faf8b86d913c70);
$parcel$export($48277fa58f01a4d6$exports, "Logout", () => $48277fa58f01a4d6$export$cad1a703886b4e3a);
const $48277fa58f01a4d6$export$99faf8b86d913c70 = {
    path: "/api/v1/sessions",
    method: "POST"
};
const $48277fa58f01a4d6$export$cad1a703886b4e3a = {
    path: "/api/v1/sessions/logout",
    method: "POST"
};


var $8263a276deed55ab$exports = {};

$parcel$export($8263a276deed55ab$exports, "SiweInit", () => $8263a276deed55ab$export$81e59ade100f8964);
$parcel$export($8263a276deed55ab$exports, "SiweAuthenticate", () => $8263a276deed55ab$export$5caf5d9f95cfdd58);
$parcel$export($8263a276deed55ab$exports, "SiweLink", () => $8263a276deed55ab$export$626623f305a79065);
$parcel$export($8263a276deed55ab$exports, "SiweUnlink", () => $8263a276deed55ab$export$cb415236257a3f80);
const $8263a276deed55ab$export$81e59ade100f8964 = {
    path: "/api/v1/siwe/init",
    method: "POST"
};
const $8263a276deed55ab$export$5caf5d9f95cfdd58 = {
    path: "/api/v1/siwe/authenticate",
    method: "POST"
};
const $8263a276deed55ab$export$626623f305a79065 = {
    path: "/api/v1/siwe/link",
    method: "POST"
};
const $8263a276deed55ab$export$cb415236257a3f80 = {
    path: "/api/v1/siwe/unlink",
    method: "POST"
};


var $7b32172f6de04ef0$exports = {};

$parcel$export($7b32172f6de04ef0$exports, "SiwsInit", () => $7b32172f6de04ef0$export$dd0e464f35c701ad);
$parcel$export($7b32172f6de04ef0$exports, "SiwsAuthenticate", () => $7b32172f6de04ef0$export$93f5e2370a853629);
$parcel$export($7b32172f6de04ef0$exports, "SiwsLink", () => $7b32172f6de04ef0$export$c309d30a2a2815ee);
$parcel$export($7b32172f6de04ef0$exports, "SiwsUnlink", () => $7b32172f6de04ef0$export$553e646318a6c92f);
const $7b32172f6de04ef0$export$dd0e464f35c701ad = {
    path: "/api/v1/siws/init",
    method: "POST"
};
const $7b32172f6de04ef0$export$93f5e2370a853629 = {
    path: "/api/v1/siws/authenticate",
    method: "POST"
};
const $7b32172f6de04ef0$export$c309d30a2a2815ee = {
    path: "/api/v1/siws/link",
    method: "POST"
};
const $7b32172f6de04ef0$export$553e646318a6c92f = {
    path: "/api/v1/siws/unlink",
    method: "POST"
};


var $5e8055405b755abe$exports = {};

$parcel$export($5e8055405b755abe$exports, "FarcasterSignerInit", () => $5e8055405b755abe$export$42654a8cac183ef0);
$parcel$export($5e8055405b755abe$exports, "FarcasterSignerStatus", () => $5e8055405b755abe$export$7f204ee171a84aa2);
const $5e8055405b755abe$export$42654a8cac183ef0 = {
    path: "/api/v1/farcaster/signer/init",
    method: "POST"
};
const $5e8055405b755abe$export$7f204ee171a84aa2 = {
    path: "/api/v1/farcaster/signer/status",
    method: "POST"
};


var $bc8d7bb5b40ee608$exports = {};

$parcel$export($bc8d7bb5b40ee608$exports, "MfaPasskeyInit", () => $bc8d7bb5b40ee608$export$5e38d17588971983);
$parcel$export($bc8d7bb5b40ee608$exports, "MfaPasskeyVerify", () => $bc8d7bb5b40ee608$export$52d08a59e63ca968);
$parcel$export($bc8d7bb5b40ee608$exports, "MfaPasskeyEnrollment", () => $bc8d7bb5b40ee608$export$d79952c9effdb95b);
const $bc8d7bb5b40ee608$export$5e38d17588971983 = {
    path: "/api/v1/mfa/passkey/init",
    method: "POST"
};
const $bc8d7bb5b40ee608$export$52d08a59e63ca968 = {
    path: "/api/v1/mfa/passkey/verify",
    method: "POST"
};
const $bc8d7bb5b40ee608$export$d79952c9effdb95b = {
    path: "/api/v1/mfa/passkey/enrollment",
    method: "POST"
};


var $4af29a4ebf5db253$exports = {};

$parcel$export($4af29a4ebf5db253$exports, "AcceptTermsOnUser", () => $4af29a4ebf5db253$export$cae4629dc339a1d3);
const $4af29a4ebf5db253$export$cae4629dc339a1d3 = {
    path: "/api/v1/users/me/accept_terms",
    method: "POST"
};


var $bfc4ba45a1fc4c5e$exports = {};

$parcel$export($bfc4ba45a1fc4c5e$exports, "PasskeyLink", () => $bfc4ba45a1fc4c5e$export$8cc9b733d1d27bf);
$parcel$export($bfc4ba45a1fc4c5e$exports, "PasskeyAuthenticate", () => $bfc4ba45a1fc4c5e$export$dd91b2a2913f179);
$parcel$export($bfc4ba45a1fc4c5e$exports, "PasskeyAuthenticateInit", () => $bfc4ba45a1fc4c5e$export$fbd0752c8f04789a);
$parcel$export($bfc4ba45a1fc4c5e$exports, "PasskeyLinkInit", () => $bfc4ba45a1fc4c5e$export$908a107a443c8819);
$parcel$export($bfc4ba45a1fc4c5e$exports, "PasskeyUnlink", () => $bfc4ba45a1fc4c5e$export$73b251c34ecfbbd2);
const $bfc4ba45a1fc4c5e$export$8cc9b733d1d27bf = {
    path: "/api/v1/passkeys/link",
    method: "POST"
};
const $bfc4ba45a1fc4c5e$export$dd91b2a2913f179 = {
    path: "/api/v1/passkeys/authenticate",
    method: "POST"
};
const $bfc4ba45a1fc4c5e$export$fbd0752c8f04789a = {
    path: "/api/v1/passkeys/authenticate/init",
    method: "POST"
};
const $bfc4ba45a1fc4c5e$export$908a107a443c8819 = {
    path: "/api/v1/passkeys/link/init",
    method: "POST"
};
const $bfc4ba45a1fc4c5e$export$73b251c34ecfbbd2 = {
    path: "/api/v1/passkeys/unlink",
    method: "POST"
};


var $aa12a54bc3b2d48d$exports = {};

$parcel$export($aa12a54bc3b2d48d$exports, "TelegramAuthenticate", () => $aa12a54bc3b2d48d$export$7429ccdbcca0d2ad);
$parcel$export($aa12a54bc3b2d48d$exports, "TelegramLink", () => $aa12a54bc3b2d48d$export$db55e97c1ee79975);
$parcel$export($aa12a54bc3b2d48d$exports, "TelegramUnlink", () => $aa12a54bc3b2d48d$export$513feadc9dae72fd);
const $aa12a54bc3b2d48d$export$7429ccdbcca0d2ad = {
    path: "/api/v1/telegram/authenticate",
    method: "POST"
};
const $aa12a54bc3b2d48d$export$db55e97c1ee79975 = {
    path: "/api/v1/telegram/link",
    method: "POST"
};
const $aa12a54bc3b2d48d$export$513feadc9dae72fd = {
    path: "/api/v1/telegram/unlink",
    method: "POST"
};


var $5d50be17907f558e$exports = {};

$parcel$export($5d50be17907f558e$exports, "getPathWithParams", () => $5d50be17907f558e$export$e2939f516ef264b4);
const $5d50be17907f558e$export$e2939f516ef264b4 = (path, params)=>{
    if (!params) return path;
    return Object.entries(params).reduce((acc, [key, value])=>{
        return acc.replace(`:${key}`, `${value}`);
    }, path);
};


var $9e6262beb329f0fb$exports = {};

$parcel$export($9e6262beb329f0fb$exports, "GetCrossAppConnections", () => $9e6262beb329f0fb$export$b9aa74fef6e2a10e);
const $9e6262beb329f0fb$export$b9aa74fef6e2a10e = {
    path: "/api/v1/apps/:app_id/cross-app/connections",
    method: "GET"
};


var $656516c52c57c728$exports = {};

$parcel$export($656516c52c57c728$exports, "OAuthProviderAuthorize", () => $656516c52c57c728$export$3251d162865f5fbf);
$parcel$export($656516c52c57c728$exports, "OAuthProviderVerify", () => $656516c52c57c728$export$e2c685498411d101);
const $656516c52c57c728$export$3251d162865f5fbf = {
    path: "/oauth/authorization_code",
    method: "POST"
};
const $656516c52c57c728$export$e2c685498411d101 = {
    path: "/oauth/verify",
    method: "POST"
};


var $61a6e48966d976eb$exports = {};

$parcel$export($61a6e48966d976eb$exports, "CoinbaseOnRampInit", () => $61a6e48966d976eb$export$4d795ff6fee0e993);
$parcel$export($61a6e48966d976eb$exports, "CoinbaseOnRampStatus", () => $61a6e48966d976eb$export$56adef0ec94545fa);
const $61a6e48966d976eb$export$4d795ff6fee0e993 = {
    path: "/api/v1/funding/coinbase_on_ramp/init",
    method: "POST"
};
const $61a6e48966d976eb$export$56adef0ec94545fa = {
    path: "/api/v1/funding/coinbase_on_ramp/status",
    method: "GET"
};


$parcel$exportWildcard($852b49dc7ea96c7c$exports, $b8155828bfbee467$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $bf4e8a2ca74ea9d9$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $97c9540320dc068b$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $4a90d090fac716ad$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $2d6a6e81ed9cd5aa$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $93c65b76e5ca94a2$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $0c1aa725aa4f001d$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $918825365cfa7807$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $7968453470d2e50e$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $ff2f0b212f862ec3$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $2047bf8d239cddb9$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $48277fa58f01a4d6$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $8263a276deed55ab$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $7b32172f6de04ef0$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $5e8055405b755abe$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $bc8d7bb5b40ee608$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $4af29a4ebf5db253$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $bfc4ba45a1fc4c5e$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $aa12a54bc3b2d48d$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $5d50be17907f558e$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $9e6262beb329f0fb$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $656516c52c57c728$exports);
$parcel$exportWildcard($852b49dc7ea96c7c$exports, $61a6e48966d976eb$exports);


var $b119bab29258e0d9$exports = {};
var $812875ae794ecf76$exports = {};

$parcel$export($812875ae794ecf76$exports, "AnalyticsEventInput", () => $812875ae794ecf76$export$5ac03ce02eff89f4);

const $812875ae794ecf76$export$5ac03ce02eff89f4 = (0, $83Qas$z).object({
    event_name: (0, $83Qas$z).string().max(255),
    client_id: (0, $83Qas$z).string().max(255).nullable(),
    payload: (0, $83Qas$z).record((0, $83Qas$z).any()).optional()
});


var $fb8f2a65df112faf$exports = {};

$parcel$export($fb8f2a65df112faf$exports, "AccountIdFromPath", () => $fb8f2a65df112faf$export$570fdd1008d53ff0);
$parcel$export($fb8f2a65df112faf$exports, "AppIdFromPath", () => $fb8f2a65df112faf$export$85be2ce1a760a71d);
$parcel$export($fb8f2a65df112faf$exports, "Pagination", () => $fb8f2a65df112faf$export$68f5e1453c188a82);


var $aab3573de5b449b0$exports = {};

$parcel$export($aab3573de5b449b0$exports, "UserId", () => $aab3573de5b449b0$export$22a69335b09c6066);
$parcel$export($aab3573de5b449b0$exports, "Email", () => $aab3573de5b449b0$export$5a6b8595f311cbb0);
$parcel$export($aab3573de5b449b0$exports, "PasswordlessCode", () => $aab3573de5b449b0$export$6051224a4b788b5f);
$parcel$export($aab3573de5b449b0$exports, "WalletAddress", () => $aab3573de5b449b0$export$e1ae011f7e31ddec);
$parcel$export($aab3573de5b449b0$exports, "normalizeEthereumAddress", () => $aab3573de5b449b0$export$52d99941c9f2e1af);
$parcel$export($aab3573de5b449b0$exports, "SolanaWalletAddress", () => $aab3573de5b449b0$export$a6c60903ce1c32fc);
$parcel$export($aab3573de5b449b0$exports, "PhoneNumber", () => $aab3573de5b449b0$export$3d623a7758cab61e);
$parcel$export($aab3573de5b449b0$exports, "normalizePhoneNumber", () => $aab3573de5b449b0$export$eeb94ade8d38f116);
$parcel$export($aab3573de5b449b0$exports, "isValidAddress", () => $aab3573de5b449b0$export$1987372119294ebb);
$parcel$export($aab3573de5b449b0$exports, "AuthenticateMode", () => $aab3573de5b449b0$export$1d53282181ae94da);




const $aab3573de5b449b0$export$22a69335b09c6066 = (0, $83Qas$z).string({
    required_error: "Privy DID must be included",
    invalid_type_error: "Privy DID is not a valid string"
}).min(1).transform((id)=>id.replace("did:privy:", ""));
const $aab3573de5b449b0$export$5a6b8595f311cbb0 = (0, $83Qas$z).string({
    required_error: "Email address must be provided",
    invalid_type_error: "Email address must be a valid string"
}).email("Invalid email address").transform((email)=>email.toLowerCase());
const $aab3573de5b449b0$export$6051224a4b788b5f = (0, $83Qas$z).string({
    required_error: "Verification code must be provided",
    invalid_type_error: "Invalid verification code"
}).length(6, "Verification code must have 6 digits.");
const $aab3573de5b449b0$export$e1ae011f7e31ddec = (0, $83Qas$z).string({
    required_error: "Ethereum address must be provided",
    invalid_type_error: "Invalid Ethereum address"
}).transform((address, ctx)=>{
    // After verifying that the address is a string, we attempt to
    // checksum it per EIP-55.
    const normalizedAddress = $aab3573de5b449b0$export$52d99941c9f2e1af(address);
    // If our normalization method returns falsy, the address is not
    // a valid Ethereum address, and we mark the input as invalid.
    if (!normalizedAddress) {
        ctx.addIssue({
            code: (0, $83Qas$z).ZodIssueCode.custom,
            message: "Invalid Ethereum address"
        });
        return (0, $83Qas$z).NEVER;
    }
    // Otherwise, we return the normalized (checksummed) address.
    return normalizedAddress;
});
const $aab3573de5b449b0$export$a6c60903ce1c32fc = (0, $83Qas$z).string({
    required_error: "Solana address must be provided",
    invalid_type_error: "Invalid Solana address"
}).trim().transform((address, ctx)=>{
    try {
        // Try to decode. If it fails the address is not a valid base58 string.
        (0, $83Qas$bs58).decode(address);
    } catch (error) {
        ctx.addIssue({
            code: (0, $83Qas$z).ZodIssueCode.custom,
            message: "Invalid Solana address"
        });
        return (0, $83Qas$z).NEVER;
    }
    return address;
});
const $aab3573de5b449b0$export$3d623a7758cab61e = (0, $83Qas$z).string({
    required_error: "Phone number must be included",
    invalid_type_error: "Phone number is not a valid string"
}).refine((val)=>{
    try {
        return $aab3573de5b449b0$export$eeb94ade8d38f116(val) !== undefined;
    } catch (e) {
        return false;
    }
}, {
    message: "Phone number is not valid"
}).transform((val)=>$aab3573de5b449b0$export$eeb94ade8d38f116(val) || ""); // This operation is safe because we previously refine the schema and break if it's undefined
function $aab3573de5b449b0$export$eeb94ade8d38f116(inputPhoneNumber) {
    const phoneNumber = (0, $83Qas$parsePhoneNumber)(inputPhoneNumber, "US");
    // https://www.npmjs.com/package/libphonenumber-js#using-phone-number-validation-feature
    if (phoneNumber?.isPossible()) return phoneNumber.formatInternational();
}
function $aab3573de5b449b0$export$1987372119294ebb(address) {
    return !!$aab3573de5b449b0$export$52d99941c9f2e1af(address);
}
function $aab3573de5b449b0$export$52d99941c9f2e1af(address) {
    if (typeof address !== "string") return;
    try {
        return (0, $83Qas$ethers).utils.getAddress(address.trim());
    } catch (e) {}
}
const $aab3573de5b449b0$export$1d53282181ae94da = (0, $83Qas$z).object({
    mode: (0, $83Qas$z).enum([
        "no-signup",
        "login-or-sign-up"
    ]).optional()
});


const $fb8f2a65df112faf$export$570fdd1008d53ff0 = (0, $83Qas$z).object({
    account_id: (0, $83Qas$z).string({
        required_error: "Account ID must be provided",
        invalid_type_error: "Account ID is not a valid string"
    }).min(1)
});
const $fb8f2a65df112faf$export$85be2ce1a760a71d = (0, $83Qas$z).object({
    app_id: (0, $83Qas$z).string({
        required_error: "App ID must be provided",
        invalid_type_error: "App ID is not a valid string"
    }).min(1)
}).transform(({ app_id: app_id })=>({
        app_id: app_id,
        appId: app_id
    })).catch(({ error: error })=>{
    throw new (0, $83Qas$InvalidInputError)(error.message, (0, $83Qas$PrivyErrorCode).MISSING_OR_INVALID_PRIVY_APP_ID);
});
const $fb8f2a65df112faf$export$68f5e1453c188a82 = (0, $83Qas$z).object({
    cursor: (0, $aab3573de5b449b0$export$22a69335b09c6066).optional(),
    limit: (0, $83Qas$z).coerce.number().max(100, {
        message: "Query limit can be at maximum 100 records."
    }).optional()
});


var $8952c42c3be19228$exports = {};

$parcel$export($8952c42c3be19228$exports, "AppResponseSchema", () => $8952c42c3be19228$export$dc5469a0fff5641e);

var $0e5934ad39a0ffa1$exports = {};

$parcel$export($0e5934ad39a0ffa1$exports, "EmbeddedWalletConfigSchema", () => $0e5934ad39a0ffa1$export$95372738a64f3b79);
$parcel$export($0e5934ad39a0ffa1$exports, "TelegramAuthConfigSchema", () => $0e5934ad39a0ffa1$export$367c50094464beeb);
$parcel$export($0e5934ad39a0ffa1$exports, "FundingMethodEnum", () => $0e5934ad39a0ffa1$export$fe80bb11c98cc392);
$parcel$export($0e5934ad39a0ffa1$exports, "FundingMethodArraySchema", () => $0e5934ad39a0ffa1$export$f5d20e939383f95);
$parcel$export($0e5934ad39a0ffa1$exports, "FundingConfigSchema", () => $0e5934ad39a0ffa1$export$4d23ce301550190a);


const $08338dc0d7cd4ec8$var$SUPPORTED_CHAIN_TYPES = [
    "evm",
    "solana"
];
const $08338dc0d7cd4ec8$export$b733e092f9f44b2c = (0, $83Qas$zod).object({
    /**
   * CAIP-2 formatted chain ID
   * Resource: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md
   */ chain: (0, $83Qas$zod).string(),
    // Defaulting to `native` token for now, so not specifying anything beyond chain
    // Currency asset
    asset: (0, $83Qas$zod).enum([
        "native-currency",
        "USDC"
    ]).optional()
});


const $0e5934ad39a0ffa1$export$95372738a64f3b79 = (0, $83Qas$zod).object({
    /**
   * Whether to create embedded wallets on login. Note that this still requires distinct API
   * calls, rather than the server creating one on the /authenticate call.
   *
   * Default: 'users-without-wallets'
   *
   * Overridable from client.
   */ create_on_login: (0, $83Qas$zod).enum([
        "users-without-wallets",
        "all-users",
        "off"
    ]),
    /**
   * Will be one or more of `user-passcode` | `google-drive` | `icloud`
   * - all enabled methods other than `password` require prior configuration
   */ user_owned_recovery_options: (0, $83Qas$zod).array((0, $83Qas$zod).enum([
        "user-passcode",
        "google-drive",
        "icloud"
    ])),
    /**
   * If true, this will prompt the user use one of the enabled recovery methods
   * to secure the recovery share of their embedded wallet.
   *
   * Otherwise (the default), Privy will secure the recovery share, and the embedded wallet
   * will be created without showing any UIs to the user.
   *
   * Overridable from client.
   */ require_user_owned_recovery_on_create: (0, $83Qas$zod).boolean().optional(),
    /**
   * @deprecated use `require_user_owned_recovery_on_create`
   *
   * If true, this will prompt the user to enter a password to secure the recovery share of
   * their embedded wallet.
   *
   * Otherwise (the default), Privy will secure the recovery share, and the embedded wallet
   * will be created without showing any UIs to the user.
   *
   * Overridable from client.
   */ require_user_password_on_create: (0, $83Qas$zod).boolean().optional()
});
const $0e5934ad39a0ffa1$export$367c50094464beeb = (0, $83Qas$zod).object({
    bot_id: (0, $83Qas$zod).string(),
    bot_name: (0, $83Qas$zod).string(),
    link_enabled: (0, $83Qas$zod).boolean(),
    seamless_auth_enabled: (0, $83Qas$zod).boolean()
});
const $0e5934ad39a0ffa1$export$fe80bb11c98cc392 = (0, $83Qas$zod).enum([
    "moonpay",
    "coinbase-onramp",
    "external"
]);
const $0e5934ad39a0ffa1$export$f5d20e939383f95 = (0, $83Qas$zod).array($0e5934ad39a0ffa1$export$fe80bb11c98cc392);
const $0e5934ad39a0ffa1$export$4d23ce301550190a = (0, $83Qas$zod).object({
    default_recommended_currency: (0, $08338dc0d7cd4ec8$export$b733e092f9f44b2c),
    default_recommended_amount: (0, $83Qas$zod).string(),
    methods: $0e5934ad39a0ffa1$export$f5d20e939383f95,
    prompt_funding_on_wallet_creation: (0, $83Qas$zod).boolean()
});


const $8952c42c3be19228$export$dc5469a0fff5641e = (0, $83Qas$z).object({
    id: (0, $83Qas$z).string(),
    name: (0, $83Qas$z).string(),
    logo_url: (0, $83Qas$z).string().nullable(),
    icon_url: (0, $83Qas$z).string().nullable(),
    terms_and_conditions_url: (0, $83Qas$z).string().nullable(),
    privacy_policy_url: (0, $83Qas$z).string().nullable(),
    require_users_accept_terms: (0, $83Qas$z).boolean().nullable(),
    theme: (0, $83Qas$z).string(),
    accent_color: (0, $83Qas$z).string().nullable(),
    show_wallet_login_first: (0, $83Qas$z).boolean(),
    allowed_domains: (0, $83Qas$z).array((0, $83Qas$z).string()),
    allowed_native_app_ids: (0, $83Qas$z).array((0, $83Qas$z).string()),
    allowed_native_app_url_schemes: (0, $83Qas$z).array((0, $83Qas$z).string()),
    wallet_auth: (0, $83Qas$z).boolean(),
    email_auth: (0, $83Qas$z).boolean(),
    sms_auth: (0, $83Qas$z).boolean(),
    google_oauth: (0, $83Qas$z).boolean(),
    twitter_oauth: (0, $83Qas$z).boolean(),
    discord_oauth: (0, $83Qas$z).boolean(),
    github_oauth: (0, $83Qas$z).boolean(),
    spotify_oauth: (0, $83Qas$z).boolean(),
    instagram_oauth: (0, $83Qas$z).boolean(),
    tiktok_oauth: (0, $83Qas$z).boolean(),
    linkedin_oauth: (0, $83Qas$z).boolean(),
    apple_oauth: (0, $83Qas$z).boolean(),
    farcaster_auth: (0, $83Qas$z).boolean(),
    passkey_auth: (0, $83Qas$z).boolean(),
    telegram_auth: (0, $83Qas$z).boolean(),
    guest_auth: (0, $83Qas$z).boolean(),
    solana_wallet_auth: (0, $83Qas$z).boolean(),
    // TODO: allowlist_enabled should be moved to AdminOnlyAppResponseSchema but is technically in older
    // versions of the server-auth getAppSettings response
    allowlist_enabled: (0, $83Qas$z).boolean(),
    allowlist_config: (0, $83Qas$z).object({
        error_title: (0, $83Qas$z).string().nullable(),
        error_detail: (0, $83Qas$z).string().nullable(),
        cta_text: (0, $83Qas$z).string().nullable(),
        cta_link: (0, $83Qas$z).string().nullable()
    }),
    wallet_connect_cloud_project_id: (0, $83Qas$z).string().nullable(),
    custom_api_url: (0, $83Qas$z).string().nullable(),
    embedded_wallet_config: (0, $0e5934ad39a0ffa1$export$95372738a64f3b79),
    enforce_wallet_uis: (0, $83Qas$z).boolean(),
    legacy_wallet_ui_config: (0, $83Qas$z).boolean(),
    fiat_on_ramp_enabled: (0, $83Qas$z).boolean(),
    captcha_enabled: (0, $83Qas$z).boolean(),
    twitter_oauth_on_mobile_enabled: (0, $83Qas$z).boolean(),
    mfa_methods: (0, $83Qas$z).array((0, $83Qas$z).enum([
        "sms",
        "totp",
        "passkey"
    ])),
    captcha_site_key: (0, $83Qas$z).string().optional(),
    verification_key: (0, $83Qas$z).string(),
    telegram_auth_config: (0, $0e5934ad39a0ffa1$export$367c50094464beeb).optional(),
    funding_config: (0, $0e5934ad39a0ffa1$export$4d23ce301550190a).optional(),
    max_linked_wallets_per_user: (0, $83Qas$z).number().nullable(),
    farcaster_link_wallets_enabled: (0, $83Qas$z).boolean()
});



var $87659a5aa8799d23$exports = {};

$parcel$export($87659a5aa8799d23$exports, "AuthenticateJwtInput", () => $87659a5aa8799d23$export$ad1e2c0abc3d279d);

const $87659a5aa8799d23$export$ad1e2c0abc3d279d = (0, $83Qas$zod).object({
    /** A JWT from the custom authentication provider */ token: (0, $83Qas$zod).string().optional()
});



var $8423ca9a1079f6d7$exports = {};

$parcel$export($8423ca9a1079f6d7$exports, "VerifyEmailInput", () => $8423ca9a1079f6d7$export$ca08e252312baf4);
$parcel$export($8423ca9a1079f6d7$exports, "AuthenticateEmailInput", () => $8423ca9a1079f6d7$export$161958b06841b78e);
$parcel$export($8423ca9a1079f6d7$exports, "InitEmailInput", () => $8423ca9a1079f6d7$export$ef1330eae1875774);
$parcel$export($8423ca9a1079f6d7$exports, "UnlinkEmailInput", () => $8423ca9a1079f6d7$export$b0a53354e7257ebb);
$parcel$export($8423ca9a1079f6d7$exports, "UpdateEmailInput", () => $8423ca9a1079f6d7$export$9ad98d5ba2865624);


const $8423ca9a1079f6d7$export$ca08e252312baf4 = (0, $83Qas$z).object({
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f),
    email: (0, $aab3573de5b449b0$export$5a6b8595f311cbb0)
});
const $8423ca9a1079f6d7$export$161958b06841b78e = $8423ca9a1079f6d7$export$ca08e252312baf4.merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $8423ca9a1079f6d7$export$ef1330eae1875774 = (0, $83Qas$z).object({
    email: (0, $aab3573de5b449b0$export$5a6b8595f311cbb0),
    token: (0, $83Qas$z).string().optional()
});
const $8423ca9a1079f6d7$export$b0a53354e7257ebb = (0, $83Qas$z).object({
    address: (0, $aab3573de5b449b0$export$5a6b8595f311cbb0)
});
const $8423ca9a1079f6d7$export$9ad98d5ba2865624 = (0, $83Qas$z).object({
    oldAddress: (0, $aab3573de5b449b0$export$5a6b8595f311cbb0),
    newAddress: (0, $aab3573de5b449b0$export$5a6b8595f311cbb0),
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f)
});


var $0777f26061100062$exports = {};

$parcel$export($0777f26061100062$exports, "JsonWebKey", () => $0777f26061100062$export$d3aaee921473d84f);
$parcel$export($0777f26061100062$exports, "JsonWebKeySet", () => $0777f26061100062$export$c8fd35505bb1704a);

const $0777f26061100062$export$d3aaee921473d84f = (0, $83Qas$z).object({
    kty: (0, $83Qas$z).string(),
    use: (0, $83Qas$z).enum([
        "sig",
        "enc"
    ]).optional(),
    key_ops: (0, $83Qas$z).array((0, $83Qas$z).enum([
        "sign",
        "verify",
        "encrypt",
        "decrypt",
        "wrapKey",
        "unwrapKey",
        "deriveKey",
        "deriveBits"
    ])).optional(),
    alg: (0, $83Qas$z).enum([
        "RS256"
    ]),
    kid: (0, $83Qas$z).string().optional(),
    x5u: (0, $83Qas$z).string().optional(),
    x5c: (0, $83Qas$z).array((0, $83Qas$z).string()).optional(),
    x5t: (0, $83Qas$z).string().optional(),
    "x5t#S256": (0, $83Qas$z).string().optional(),
    n: (0, $83Qas$z).string().optional(),
    e: (0, $83Qas$z).string().optional(),
    d: (0, $83Qas$z).string().optional()
});
const $0777f26061100062$export$c8fd35505bb1704a = (0, $83Qas$z).object({
    keys: (0, $83Qas$z).array($0777f26061100062$export$d3aaee921473d84f)
});


var $a666dad64df3d92a$exports = {};

$parcel$export($a666dad64df3d92a$exports, "FarcasterConnectInitResponse", () => $a666dad64df3d92a$export$6ddd112193bb4f42);
$parcel$export($a666dad64df3d92a$exports, "FarcasterConnectStatusCompletedResponse", () => $a666dad64df3d92a$export$10e572b3d0f4fe4f);
$parcel$export($a666dad64df3d92a$exports, "FarcasterConnectStatusPendingResponse", () => $a666dad64df3d92a$export$430853814f05cdc4);
$parcel$export($a666dad64df3d92a$exports, "FarcasterAuthenticateInput", () => $a666dad64df3d92a$export$2c59d6b32b518c6e);
$parcel$export($a666dad64df3d92a$exports, "FarcasterLinkInput", () => $a666dad64df3d92a$export$40d6546edc9ccfd3);
$parcel$export($a666dad64df3d92a$exports, "FarcasterInitInput", () => $a666dad64df3d92a$export$8dd30088cf36d3f5);
$parcel$export($a666dad64df3d92a$exports, "FarcasterUnlinkInput", () => $a666dad64df3d92a$export$325ad26071643d09);


const $a666dad64df3d92a$export$6ddd112193bb4f42 = (0, $83Qas$z).object({
    channel_token: (0, $83Qas$z).string(),
    connect_uri: (0, $83Qas$z).string()
});
const $a666dad64df3d92a$export$10e572b3d0f4fe4f = (0, $83Qas$z).object({
    state: (0, $83Qas$z).literal("completed"),
    nonce: (0, $83Qas$z).string(),
    message: (0, $83Qas$z).string(),
    signature: (0, $83Qas$z).string(),
    fid: (0, $83Qas$z).number(),
    username: (0, $83Qas$z).string(),
    display_name: (0, $83Qas$z).string(),
    bio: (0, $83Qas$z).string(),
    pfp_url: (0, $83Qas$z).string()
});
const $a666dad64df3d92a$export$430853814f05cdc4 = (0, $83Qas$z).object({
    state: (0, $83Qas$z).literal("pending"),
    nonce: (0, $83Qas$z).string()
});
const $a666dad64df3d92a$export$2c59d6b32b518c6e = (0, $83Qas$z).object({
    channel_token: (0, $83Qas$z).string(),
    message: (0, $83Qas$z).string(),
    signature: (0, $83Qas$z).string(),
    fid: (0, $83Qas$z).number()
}).merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $a666dad64df3d92a$export$40d6546edc9ccfd3 = (0, $83Qas$z).object({
    channel_token: (0, $83Qas$z).string(),
    message: (0, $83Qas$z).string(),
    signature: (0, $83Qas$z).string(),
    fid: (0, $83Qas$z).number()
});
const $a666dad64df3d92a$export$8dd30088cf36d3f5 = (0, $83Qas$z).object({
    token: (0, $83Qas$z).string().optional(),
    redirect_url: (0, $83Qas$z).string().url().optional(),
    relying_party: (0, $83Qas$z).string().optional()
});
const $a666dad64df3d92a$export$325ad26071643d09 = (0, $83Qas$z).object({
    fid: (0, $83Qas$z).number()
});


var $bfb33230d402f72a$exports = {};

$parcel$export($bfb33230d402f72a$exports, "AuthenticateGuestInput", () => $bfb33230d402f72a$export$494be569a247ec9a);

const $bfb33230d402f72a$export$494be569a247ec9a = (0, $83Qas$z).object({
    /**
     * Locally-persisted guest secret.
     * This credential should have as much entropy as embedded wallet entropy.
     */ guest_credential: (0, $83Qas$z).string().length(43)
}).strict();


var $37a005f91989be72$exports = {};

$parcel$export($37a005f91989be72$exports, "MfaVerifyResponse", () => $37a005f91989be72$export$2a073761fe65eeb);
$parcel$export($37a005f91989be72$exports, "MfaSmsInitInput", () => $37a005f91989be72$export$5f85d1e59e4b8093);
$parcel$export($37a005f91989be72$exports, "MfaSmsAuthenticateEnrollInput", () => $37a005f91989be72$export$d79e41bdbeb33bc8);
$parcel$export($37a005f91989be72$exports, "MfaSmsEnrollInput", () => $37a005f91989be72$export$44ba6bee6f8c49bc);
$parcel$export($37a005f91989be72$exports, "MfaSmsVerifyInput", () => $37a005f91989be72$export$826840b7b6a37c22);
$parcel$export($37a005f91989be72$exports, "MfaTotpInput", () => $37a005f91989be72$export$c5079467361455e6);
$parcel$export($37a005f91989be72$exports, "ResponseTotpInitMfa", () => $37a005f91989be72$export$2e0b8857111f6a95);


const $37a005f91989be72$export$2a073761fe65eeb = (0, $83Qas$zod).object({
    token: (0, $83Qas$zod).string()
}).strict();
/**
 * SMS MFA enrollment and verification inputs.
 */ const $37a005f91989be72$var$MfaSmsInitVerifyInput = (0, $83Qas$zod).object({
    action: (0, $83Qas$zod).literal("verify")
}).strict();
const $37a005f91989be72$var$MfaSmsInitEnrollInput = (0, $83Qas$zod).object({
    action: (0, $83Qas$zod).literal("enroll"),
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e)
}).strict();
const $37a005f91989be72$export$5f85d1e59e4b8093 = (0, $83Qas$zod).union([
    $37a005f91989be72$var$MfaSmsInitVerifyInput,
    $37a005f91989be72$var$MfaSmsInitEnrollInput
]);
const $37a005f91989be72$export$d79e41bdbeb33bc8 = (0, $83Qas$zod).object({
    action: (0, $83Qas$zod).literal("enroll").optional(),
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f),
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e)
}).strict();
const $37a005f91989be72$export$44ba6bee6f8c49bc = (0, $83Qas$zod).object({
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f),
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e)
}).strict();
const $37a005f91989be72$export$826840b7b6a37c22 = (0, $83Qas$zod).object({
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f)
}).strict();
const $37a005f91989be72$export$c5079467361455e6 = (0, $83Qas$zod).object({
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f)
}).strict();
const $37a005f91989be72$export$2e0b8857111f6a95 = (0, $83Qas$zod).object({
    totpSecret: (0, $83Qas$zod).string(),
    totpAuthUrl: (0, $83Qas$zod).string()
}).strict();


var $8dc16b6adb6078fa$exports = {};

$parcel$export($8dc16b6adb6078fa$exports, "ExternalOAuthProvider", () => $8dc16b6adb6078fa$export$16a6de659e03cb8c);
$parcel$export($8dc16b6adb6078fa$exports, "OAuthProvider", () => $8dc16b6adb6078fa$export$96d4e4cf27d59030);
$parcel$export($8dc16b6adb6078fa$exports, "AuthenticateOauthInput", () => $8dc16b6adb6078fa$export$3a815e8ae7821b6e);
$parcel$export($8dc16b6adb6078fa$exports, "OAuthInitInput", () => $8dc16b6adb6078fa$export$25d68298fd675d8b);
$parcel$export($8dc16b6adb6078fa$exports, "LinkOAuthInput", () => $8dc16b6adb6078fa$export$5169dcfff85dfbff);
$parcel$export($8dc16b6adb6078fa$exports, "OAuthUnlinkInput", () => $8dc16b6adb6078fa$export$100da24bb64f0e09);
$parcel$export($8dc16b6adb6078fa$exports, "OAuthInitResponse", () => $8dc16b6adb6078fa$export$98ade43b6adcef0b);
$parcel$export($8dc16b6adb6078fa$exports, "SUPPORTED_OAUTH_PROVIDERS", () => $7d4d0e60b9c76b9d$export$b5266d2768e29008);


// Casting as const makes it immutable.
const $7d4d0e60b9c76b9d$export$b5266d2768e29008 = [
    "google",
    "discord",
    "twitter",
    "github",
    "spotify",
    "instagram",
    "tiktok",
    "linkedin",
    "apple"
];



const $8dc16b6adb6078fa$export$16a6de659e03cb8c = (0, $83Qas$z).enum((0, $7d4d0e60b9c76b9d$export$b5266d2768e29008), {
    invalid_type_error: "Please specify a supported provider.",
    required_error: "Please specify a supported provider."
});
const $8dc16b6adb6078fa$export$96d4e4cf27d59030 = $8dc16b6adb6078fa$export$16a6de659e03cb8c.or((0, $83Qas$z).custom((value)=>{
    return typeof value === "string" ? /^privy:[a-z0-9]{25}$/i.test(value) : false;
}, {
    message: "Please specify a supported provider."
}));
const $8dc16b6adb6078fa$export$3a815e8ae7821b6e = (0, $83Qas$z).object({
    authorization_code: (0, $83Qas$z).string(),
    state_code: (0, $83Qas$z).string(),
    code_verifier: (0, $83Qas$z).string().min(43).max(128).optional(),
    code_type: (0, $83Qas$z).enum([
        "raw"
    ]).optional()
}).merge((0, $aab3573de5b449b0$export$1d53282181ae94da)).catch(()=>{
    throw new (0, $83Qas$UnauthorizedError)("Please specify authorization_code and state_code.", (0, $83Qas$PrivyErrorCode).INVALID_CREDENTIALS);
});
const $8dc16b6adb6078fa$export$25d68298fd675d8b = (0, $83Qas$z).object({
    redirect_to: (0, $83Qas$z).string({
        required_error: "Please provide a redirect_to value."
    }),
    provider: $8dc16b6adb6078fa$export$96d4e4cf27d59030,
    token: (0, $83Qas$z).string().optional(),
    state_code: (0, $83Qas$z).string().optional(),
    code_challenge: (0, $83Qas$z).string().optional()
});
const $8dc16b6adb6078fa$export$5169dcfff85dfbff = (0, $83Qas$z).object({
    authorization_code: (0, $83Qas$z).string(),
    state_code: (0, $83Qas$z).string(),
    code_verifier: (0, $83Qas$z).string().min(43).max(128).optional(),
    code_type: (0, $83Qas$z).enum([
        "raw"
    ]).optional()
});
const $8dc16b6adb6078fa$export$100da24bb64f0e09 = (0, $83Qas$z).object({
    subject: (0, $83Qas$z).string(),
    provider: $8dc16b6adb6078fa$export$96d4e4cf27d59030
});
const $8dc16b6adb6078fa$export$98ade43b6adcef0b = (0, $83Qas$z).object({
    url: (0, $83Qas$z).string()
});


var $f91fb55607e72b51$exports = {};

$parcel$export($f91fb55607e72b51$exports, "EmptyObject", () => $f91fb55607e72b51$export$2441567ff18621e0);
$parcel$export($f91fb55607e72b51$exports, "SuccessObject", () => $f91fb55607e72b51$export$8aed050e83c45a78);

const $f91fb55607e72b51$export$2441567ff18621e0 = (0, $83Qas$z).record((0, $83Qas$z).string(), (0, $83Qas$z).never());
const $f91fb55607e72b51$export$8aed050e83c45a78 = (0, $83Qas$z).object({
    success: (0, $83Qas$z).boolean()
});


var $e4527135bfc60402$exports = {};

$parcel$export($e4527135bfc60402$exports, "PasskeyAuthenticatorVerifyOptions", () => $e4527135bfc60402$export$7b01305e0ab33ea3);
$parcel$export($e4527135bfc60402$exports, "PasskeyAuthenticatorEnrollmentOptions", () => $e4527135bfc60402$export$1c8fd79a94eba09d);
$parcel$export($e4527135bfc60402$exports, "PasskeyAuthenticatorVerifyResponse", () => $e4527135bfc60402$export$96203b761bf7cadb);
$parcel$export($e4527135bfc60402$exports, "PasskeyAuthenticatorEnrollmentResponse", () => $e4527135bfc60402$export$49424bcfec0159aa);
$parcel$export($e4527135bfc60402$exports, "PasskeyInitInput", () => $e4527135bfc60402$export$8c48c68e39a52ea);
$parcel$export($e4527135bfc60402$exports, "ResponsePasskeyInitAuthenticate", () => $e4527135bfc60402$export$2741be68c8db9b43);
$parcel$export($e4527135bfc60402$exports, "ResponsePasskeyInitLink", () => $e4527135bfc60402$export$b968e9f4fdc5f0b2);
$parcel$export($e4527135bfc60402$exports, "PasskeyLinkInput", () => $e4527135bfc60402$export$9bd14c1a22f65d2);
$parcel$export($e4527135bfc60402$exports, "PasskeyAuthenticateInput", () => $e4527135bfc60402$export$8525265affc59e4b);
$parcel$export($e4527135bfc60402$exports, "UnlinkPasskeyInput", () => $e4527135bfc60402$export$1d32dd5b05e1d4dd);

const $e4527135bfc60402$export$7b01305e0ab33ea3 = (0, $83Qas$zod).object({
    challenge: (0, $83Qas$zod).string(),
    timeout: (0, $83Qas$zod).number().optional(),
    rp_id: (0, $83Qas$zod).string().optional(),
    allow_credentials: (0, $83Qas$zod).array((0, $83Qas$zod).object({
        id: (0, $83Qas$zod).string(),
        type: (0, $83Qas$zod).string(),
        transports: (0, $83Qas$zod).array((0, $83Qas$zod).string()).optional()
    })).optional(),
    user_verification: (0, $83Qas$zod).string().optional(),
    extensions: (0, $83Qas$zod).object({
        app_id: (0, $83Qas$zod).string().optional(),
        cred_props: (0, $83Qas$zod).boolean().optional(),
        hmac_create_secret: (0, $83Qas$zod).boolean().optional()
    }).optional()
});
const $e4527135bfc60402$export$1c8fd79a94eba09d = (0, $83Qas$zod).object({
    rp: (0, $83Qas$zod).object({
        name: (0, $83Qas$zod).string(),
        id: (0, $83Qas$zod).string().optional()
    }),
    user: (0, $83Qas$zod).object({
        id: (0, $83Qas$zod).string(),
        name: (0, $83Qas$zod).string(),
        display_name: (0, $83Qas$zod).string()
    }),
    challenge: (0, $83Qas$zod).string(),
    pub_key_cred_params: (0, $83Qas$zod).array((0, $83Qas$zod).object({
        alg: (0, $83Qas$zod).number(),
        type: (0, $83Qas$zod).literal("public-key")
    })),
    timeout: (0, $83Qas$zod).number().optional(),
    exclude_credentials: (0, $83Qas$zod).array((0, $83Qas$zod).object({
        id: (0, $83Qas$zod).string(),
        type: (0, $83Qas$zod).string(),
        transports: (0, $83Qas$zod).array((0, $83Qas$zod).string()).optional()
    })).optional(),
    authenticator_selection: (0, $83Qas$zod).object({
        authenticator_attachment: (0, $83Qas$zod).string().optional(),
        require_resident_key: (0, $83Qas$zod).boolean().optional(),
        resident_key: (0, $83Qas$zod).string().optional(),
        user_verification: (0, $83Qas$zod).string().optional()
    }).optional(),
    attestation: (0, $83Qas$zod).string().optional(),
    extensions: (0, $83Qas$zod).object({
        app_id: (0, $83Qas$zod).string().optional(),
        cred_props: (0, $83Qas$zod).object({
            rk: (0, $83Qas$zod).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, $83Qas$zod).boolean().optional()
    }).optional()
});
const $e4527135bfc60402$export$96203b761bf7cadb = (0, $83Qas$zod).object({
    id: (0, $83Qas$zod).string(),
    raw_id: (0, $83Qas$zod).string(),
    response: (0, $83Qas$zod).object({
        client_data_json: (0, $83Qas$zod).string(),
        authenticator_data: (0, $83Qas$zod).string(),
        signature: (0, $83Qas$zod).string(),
        user_handle: (0, $83Qas$zod).string().optional()
    }),
    authenticator_attachment: (0, $83Qas$zod).string().optional(),
    client_extension_results: (0, $83Qas$zod).object({
        app_id: (0, $83Qas$zod).boolean().optional(),
        cred_props: (0, $83Qas$zod).object({
            rk: (0, $83Qas$zod).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, $83Qas$zod).boolean().optional()
    }),
    type: (0, $83Qas$zod).literal("public-key")
});
const $e4527135bfc60402$export$49424bcfec0159aa = (0, $83Qas$zod).object({
    id: (0, $83Qas$zod).string(),
    raw_id: (0, $83Qas$zod).string(),
    response: (0, $83Qas$zod).object({
        client_data_json: (0, $83Qas$zod).string(),
        attestation_object: (0, $83Qas$zod).string(),
        authenticator_data: (0, $83Qas$zod).string().optional(),
        // TODO: Type this to AuthenticatorTransportFuture[]
        transports: (0, $83Qas$zod).array((0, $83Qas$zod).any()).optional(),
        public_key_algorithm: (0, $83Qas$zod).number().optional(),
        public_key: (0, $83Qas$zod).string().optional()
    }),
    authenticator_attachment: (0, $83Qas$zod).string().optional(),
    client_extension_results: (0, $83Qas$zod).object({
        app_id: (0, $83Qas$zod).boolean().optional(),
        cred_props: (0, $83Qas$zod).object({
            rk: (0, $83Qas$zod).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, $83Qas$zod).boolean().optional()
    }),
    type: (0, $83Qas$zod).literal("public-key")
});
const $e4527135bfc60402$export$8c48c68e39a52ea = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional(),
    token: (0, $83Qas$zod).string().optional()
}).strict();
const $e4527135bfc60402$export$2741be68c8db9b43 = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional(),
    options: $e4527135bfc60402$export$7b01305e0ab33ea3
}).strict();
const $e4527135bfc60402$export$b968e9f4fdc5f0b2 = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional(),
    options: $e4527135bfc60402$export$1c8fd79a94eba09d
}).strict();
const $e4527135bfc60402$export$9bd14c1a22f65d2 = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional(),
    authenticator_response: $e4527135bfc60402$export$49424bcfec0159aa
}).strict();
const $e4527135bfc60402$export$8525265affc59e4b = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional(),
    challenge: (0, $83Qas$zod).string(),
    authenticator_response: $e4527135bfc60402$export$96203b761bf7cadb
}).strict();
const $e4527135bfc60402$export$1d32dd5b05e1d4dd = (0, $83Qas$zod).object({
    credential_id: (0, $83Qas$zod).string()
});


var $dd6f6c1fe379a225$exports = {};

$parcel$export($dd6f6c1fe379a225$exports, "RecoveryKeyMaterialInput", () => $dd6f6c1fe379a225$export$5133f8b99ced310e);
$parcel$export($dd6f6c1fe379a225$exports, "RecoveryKeyMaterialResponse", () => $dd6f6c1fe379a225$export$52cacd413024ecd);
$parcel$export($dd6f6c1fe379a225$exports, "OAuthAuthenticateRecoveryResponse", () => $dd6f6c1fe379a225$export$8d23e6b066801647);
$parcel$export($dd6f6c1fe379a225$exports, "OAuthInitRecoveryInput", () => $dd6f6c1fe379a225$export$eb513813549270a);
$parcel$export($dd6f6c1fe379a225$exports, "OAuthInitICloudRecoveryInput", () => $dd6f6c1fe379a225$export$425e6a14ed44c52f);
$parcel$export($dd6f6c1fe379a225$exports, "OAuthCallbackICloudExpoInput", () => $dd6f6c1fe379a225$export$bec1efaa57e00983);
$parcel$export($dd6f6c1fe379a225$exports, "OAuthCallbackICloudExpoResponse", () => $dd6f6c1fe379a225$export$ea90bf451d6cb856);
$parcel$export($dd6f6c1fe379a225$exports, "RecoveryConfigurationICloudInput", () => $dd6f6c1fe379a225$export$dfb75001bba12cf3);
$parcel$export($dd6f6c1fe379a225$exports, "RecoveryConfigurationICloudResponse", () => $dd6f6c1fe379a225$export$a53156909e5ca052);



const $dd6f6c1fe379a225$export$5133f8b99ced310e = (0, $83Qas$z).object({
    chain_type: (0, $83Qas$z).string().optional()
});
const $dd6f6c1fe379a225$export$52cacd413024ecd = (0, $83Qas$z).object({
    recovery_type: (0, $83Qas$z).enum([
        "user_passcode_derived_recovery_key",
        "privy_passcode_derived_recovery_key",
        "privy_generated_recovery_key",
        "google_drive_recovery_secret",
        "icloud_recovery_secret"
    ]),
    recovery_key_derivation_salt: (0, $83Qas$z).string(),
    recovery_code: (0, $83Qas$z).string(),
    recovery_key: (0, $83Qas$z).string(),
    file_id: (0, $83Qas$z).string(),
    icloud_record_name: (0, $83Qas$z).string()
});
const $dd6f6c1fe379a225$export$8d23e6b066801647 = (0, $83Qas$z).object({
    access_token: (0, $83Qas$z).string()
});
const $dd6f6c1fe379a225$export$eb513813549270a = (0, $83Qas$z).object({
    redirect_to: (0, $83Qas$z).string({
        required_error: "Please provide a redirect_to value."
    }),
    token: (0, $83Qas$z).string().optional(),
    state_code: (0, $83Qas$z).string().optional(),
    code_challenge: (0, $83Qas$z).string().optional()
});
const $dd6f6c1fe379a225$export$425e6a14ed44c52f = (0, $83Qas$z).object({
    client_type: (0, $83Qas$z).enum([
        "web",
        "expo-ios"
    ])
});
const $dd6f6c1fe379a225$export$bec1efaa57e00983 = (0, $83Qas$z).object({
    /**
     * The authorization code from apple
     */ ckWebAuthToken: (0, $83Qas$z).string()
}).transform((input)=>({
        ckWebAuthToken: input.ckWebAuthToken
    })).catch(()=>{
    throw new (0, $83Qas$UnauthorizedError)("Missing ckWebAuthToken.", (0, $83Qas$PrivyErrorCode).INVALID_DATA);
});
const $dd6f6c1fe379a225$export$ea90bf451d6cb856 = (0, $f91fb55607e72b51$export$2441567ff18621e0);
const $dd6f6c1fe379a225$export$dfb75001bba12cf3 = (0, $83Qas$z).object({
    client_type: (0, $83Qas$z).enum([
        "web",
        "expo-ios"
    ])
});
const $dd6f6c1fe379a225$export$a53156909e5ca052 = (0, $83Qas$z).object({
    api_token: (0, $83Qas$z).string(),
    container_identifier: (0, $83Qas$z).string(),
    environment: (0, $83Qas$z).string()
});


var $3cbf49bd3049c518$exports = {};

$parcel$export($3cbf49bd3049c518$exports, "SiweInput", () => $3cbf49bd3049c518$export$1c8c55e22e1d8d9);
$parcel$export($3cbf49bd3049c518$exports, "AuthenticateSiweInput", () => $3cbf49bd3049c518$export$3435c4883fbaeb7f);
$parcel$export($3cbf49bd3049c518$exports, "SiweAddressInput", () => $3cbf49bd3049c518$export$b5b65f417dc17888);
$parcel$export($3cbf49bd3049c518$exports, "SiweInitInput", () => $3cbf49bd3049c518$export$6c957bbdcf62135c);
$parcel$export($3cbf49bd3049c518$exports, "SiweNonce", () => $3cbf49bd3049c518$export$8535d2f477730820);


const $3cbf49bd3049c518$export$1c8c55e22e1d8d9 = (0, $83Qas$z).object({
    message: (0, $83Qas$z).string({
        invalid_type_error: "Invalid SIWE message"
    }),
    signature: (0, $83Qas$z).string({
        invalid_type_error: "Invalid SIWE signature"
    }),
    chainId: (0, $83Qas$z).string().max(41).nullable().default(null),
    walletClientType: (0, $83Qas$z).string().max(64).nullable().default(null),
    connectorType: (0, $83Qas$z).string().max(64).nullable().default(null)
});
const $3cbf49bd3049c518$export$3435c4883fbaeb7f = $3cbf49bd3049c518$export$1c8c55e22e1d8d9.merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $3cbf49bd3049c518$export$b5b65f417dc17888 = (0, $83Qas$z).object({
    address: (0, $aab3573de5b449b0$export$e1ae011f7e31ddec)
});
const $3cbf49bd3049c518$export$6c957bbdcf62135c = (0, $83Qas$z).object({
    address: (0, $aab3573de5b449b0$export$e1ae011f7e31ddec),
    token: (0, $83Qas$z).string().optional()
});
const $3cbf49bd3049c518$export$8535d2f477730820 = (0, $83Qas$z).object({
    nonce: (0, $83Qas$z).string(),
    address: (0, $83Qas$z).string(),
    expires_at: (0, $83Qas$z).string()
});


var $0b8e24f996a08aa3$exports = {};

$parcel$export($0b8e24f996a08aa3$exports, "SiwsInput", () => $0b8e24f996a08aa3$export$fa2883d1fd7d2f4d);
$parcel$export($0b8e24f996a08aa3$exports, "AuthenticateSiwsInput", () => $0b8e24f996a08aa3$export$defac42dc31eb54f);
$parcel$export($0b8e24f996a08aa3$exports, "SiwsAddressInput", () => $0b8e24f996a08aa3$export$767d61e052aa619c);
$parcel$export($0b8e24f996a08aa3$exports, "SiwsInitInput", () => $0b8e24f996a08aa3$export$334006f5cf1e2e10);
$parcel$export($0b8e24f996a08aa3$exports, "SiwsNonce", () => $0b8e24f996a08aa3$export$678e538743698961);


const $0b8e24f996a08aa3$export$fa2883d1fd7d2f4d = (0, $83Qas$z).object({
    message: (0, $83Qas$z).string({
        invalid_type_error: "Invalid SIWS message"
    }),
    signature: (0, $83Qas$z).string({
        invalid_type_error: "Invalid SIWS signature"
    }),
    walletClientType: (0, $83Qas$z).string().max(64).nullable().default(null),
    connectorType: (0, $83Qas$z).string().max(64).nullable().default(null)
});
const $0b8e24f996a08aa3$export$defac42dc31eb54f = $0b8e24f996a08aa3$export$fa2883d1fd7d2f4d.merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $0b8e24f996a08aa3$export$767d61e052aa619c = (0, $83Qas$z).object({
    address: (0, $aab3573de5b449b0$export$a6c60903ce1c32fc)
});
const $0b8e24f996a08aa3$export$334006f5cf1e2e10 = (0, $83Qas$z).object({
    address: (0, $aab3573de5b449b0$export$a6c60903ce1c32fc),
    token: (0, $83Qas$z).string().optional()
});
const $0b8e24f996a08aa3$export$678e538743698961 = (0, $83Qas$z).object({
    nonce: (0, $83Qas$z).string(),
    address: (0, $83Qas$z).string(),
    expires_at: (0, $83Qas$z).string()
});


var $c55f91d419fc395f$exports = {};

$parcel$export($c55f91d419fc395f$exports, "FarcasterSignerInitInput", () => $c55f91d419fc395f$export$809802d572486db);
$parcel$export($c55f91d419fc395f$exports, "FarcasterSignerInitResponse", () => $c55f91d419fc395f$export$327461233dec5bf6);
$parcel$export($c55f91d419fc395f$exports, "FarcasterSignerStatusResponse", () => $c55f91d419fc395f$export$2a58492abd647af5);

const $c55f91d419fc395f$export$809802d572486db = (0, $83Qas$z).object({
    // TODO: Down the line, we should support specifying developer's own app FID
    // request_fid: z.bigint().optional(),
    /** Must be an ed25519 key, NOT secp256k1 (ethereum native curve) */ ed25519_public_key: (0, $83Qas$z).string().refine((key)=>key.startsWith("0x") && key.length === 66),
    /**
   * The deadline for a signer request to succeed. After the deadline, the request will no longer
   * succeed. Defaults to 1 hour.
   */ deadline: (0, $83Qas$z).bigint().optional().default(()=>{
        const now = Math.floor(Date.now() / 1000);
        const oneHour = 3600;
        return BigInt(now + oneHour);
    })
});
const $c55f91d419fc395f$export$327461233dec5bf6 = (0, $83Qas$z).discriminatedUnion("status", [
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("pending_approval"),
        signer_approval_url: (0, $83Qas$z).string()
    }),
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("approved"),
        /** The FID of the user who approved the signer */ fid: (0, $83Qas$z).bigint()
    }),
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("revoked"),
        /** The FID of the user who approved the signer */ fid: (0, $83Qas$z).bigint()
    })
]);
const $c55f91d419fc395f$export$2a58492abd647af5 = (0, $83Qas$z).discriminatedUnion("status", [
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("pending_approval")
    }),
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("approved"),
        /** The FID of the user who approved the signer */ fid: (0, $83Qas$z).bigint()
    }),
    (0, $83Qas$z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $83Qas$z).string(),
        status: (0, $83Qas$z).literal("revoked"),
        /** The FID of the user who approved the signer */ fid: (0, $83Qas$z).bigint()
    })
]);


var $2a629451b24ff1c8$exports = {};

$parcel$export($2a629451b24ff1c8$exports, "VerifyPhoneInput", () => $2a629451b24ff1c8$export$1bbb8302b8922d52);
$parcel$export($2a629451b24ff1c8$exports, "AuthenticatePhoneInput", () => $2a629451b24ff1c8$export$d57783821ffbf377);
$parcel$export($2a629451b24ff1c8$exports, "PasswordlessSmsPhoneInput", () => $2a629451b24ff1c8$export$794b219acd817434);
$parcel$export($2a629451b24ff1c8$exports, "UnlinkPhoneInput", () => $2a629451b24ff1c8$export$63fc94c9f0d4eea8);
$parcel$export($2a629451b24ff1c8$exports, "UpdatePhoneInput", () => $2a629451b24ff1c8$export$bbf6b1d1316adec);



const $2a629451b24ff1c8$export$1bbb8302b8922d52 = (0, $83Qas$z).object({
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f).catch(()=>{
        throw new (0, $83Qas$LegacyInvalidInputError)("Invalid SMS verification code", (0, $83Qas$PrivyErrorCode).INVALID_CREDENTIALS);
    }),
    // Not a mistake, it's actually camelCase
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e).catch(()=>{
        throw new (0, $83Qas$LegacyInvalidInputError)("Invalid phone number", (0, $83Qas$PrivyErrorCode).INVALID_CREDENTIALS);
    })
});
const $2a629451b24ff1c8$export$d57783821ffbf377 = $2a629451b24ff1c8$export$1bbb8302b8922d52.merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $2a629451b24ff1c8$export$794b219acd817434 = (0, $83Qas$z).object({
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e).catch(()=>{
        throw new (0, $83Qas$LegacyInvalidInputError)("Invalid phone number", (0, $83Qas$PrivyErrorCode).INVALID_DATA);
    }),
    token: (0, $83Qas$z).string().optional()
});
const $2a629451b24ff1c8$export$63fc94c9f0d4eea8 = (0, $83Qas$z).object({
    // Not a mistake, it's actually camelCase
    phoneNumber: (0, $aab3573de5b449b0$export$3d623a7758cab61e)
});
const $2a629451b24ff1c8$export$bbf6b1d1316adec = (0, $83Qas$z).object({
    old_phone_number: (0, $aab3573de5b449b0$export$3d623a7758cab61e),
    new_phone_number: (0, $aab3573de5b449b0$export$3d623a7758cab61e),
    code: (0, $aab3573de5b449b0$export$6051224a4b788b5f)
});


var $dbeea6fd043fc091$exports = {};

$parcel$export($dbeea6fd043fc091$exports, "RefreshTokenInput", () => $dbeea6fd043fc091$export$6aa3af8166ea5386);
$parcel$export($dbeea6fd043fc091$exports, "OptionalRefreshTokenInput", () => $dbeea6fd043fc091$export$3ea9bae16087a241);
$parcel$export($dbeea6fd043fc091$exports, "ForkedToken", () => $dbeea6fd043fc091$export$874b0538e4f42518);

const $dbeea6fd043fc091$export$6aa3af8166ea5386 = (0, $83Qas$z).object({
    refresh_token: (0, $83Qas$z).string()
});
const $dbeea6fd043fc091$export$3ea9bae16087a241 = $dbeea6fd043fc091$export$6aa3af8166ea5386.partial();
const $dbeea6fd043fc091$export$874b0538e4f42518 = (0, $83Qas$z).object({
    token: (0, $83Qas$z).string(),
    refresh_token: (0, $83Qas$z).string(),
    new_session_refresh_token: (0, $83Qas$z).string()
});


var $ff5f4339e0989169$exports = {};

$parcel$export($ff5f4339e0989169$exports, "EmailAccount", () => $ff5f4339e0989169$export$b291b2a8281f162e);
$parcel$export($ff5f4339e0989169$exports, "PhoneAccount", () => $ff5f4339e0989169$export$e461068a75eb0fb3);
$parcel$export($ff5f4339e0989169$exports, "EthereumAccount", () => $ff5f4339e0989169$export$e72ed3751242ab66);
$parcel$export($ff5f4339e0989169$exports, "SolanaAccount", () => $ff5f4339e0989169$export$d08a38b431a135bc);
$parcel$export($ff5f4339e0989169$exports, "FarcasterAccount", () => $ff5f4339e0989169$export$70b11e99555e84cb);
$parcel$export($ff5f4339e0989169$exports, "PasskeyAccount", () => $ff5f4339e0989169$export$c7ead6bbfb2c42a0);
$parcel$export($ff5f4339e0989169$exports, "TelegramAccount", () => $ff5f4339e0989169$export$70ddd50daefc295f);
$parcel$export($ff5f4339e0989169$exports, "EthereumEmbeddedWalletAccount", () => $ff5f4339e0989169$export$42e79fc33a58a2f5);
$parcel$export($ff5f4339e0989169$exports, "SolanaEmbeddedWalletAccount", () => $ff5f4339e0989169$export$734c44f2e990ba65);
$parcel$export($ff5f4339e0989169$exports, "GoogleOauthAccount", () => $ff5f4339e0989169$export$ab3c40bebc4d6f36);
$parcel$export($ff5f4339e0989169$exports, "TwitterOauthAccount", () => $ff5f4339e0989169$export$31cc32c7bc65c2e0);
$parcel$export($ff5f4339e0989169$exports, "DiscordOauthAccount", () => $ff5f4339e0989169$export$cb35596f96141125);
$parcel$export($ff5f4339e0989169$exports, "GithubOauthAccount", () => $ff5f4339e0989169$export$55430b29b82d98a8);
$parcel$export($ff5f4339e0989169$exports, "LinkedInOauthAccount", () => $ff5f4339e0989169$export$25b982740d2e431c);
$parcel$export($ff5f4339e0989169$exports, "SpotifyOauthAccount", () => $ff5f4339e0989169$export$260592be4b1062f4);
$parcel$export($ff5f4339e0989169$exports, "InstagramOauthAccount", () => $ff5f4339e0989169$export$c683d35f732c1912);
$parcel$export($ff5f4339e0989169$exports, "TiktokOauthAccount", () => $ff5f4339e0989169$export$5e8143c3f32ac816);
$parcel$export($ff5f4339e0989169$exports, "AppleOauthAccount", () => $ff5f4339e0989169$export$67af8a2e34850611);
$parcel$export($ff5f4339e0989169$exports, "CustomJwtAccount", () => $ff5f4339e0989169$export$5da87a1cf22b57d0);
$parcel$export($ff5f4339e0989169$exports, "CrossAppEmbeddedWallet", () => $ff5f4339e0989169$export$ccd62fbf4b025fee);
$parcel$export($ff5f4339e0989169$exports, "CrossAppSmartWallet", () => $ff5f4339e0989169$export$8f1f4ad4f8276ad7);
$parcel$export($ff5f4339e0989169$exports, "CrossAppAccount", () => $ff5f4339e0989169$export$564035ac82295c40);
$parcel$export($ff5f4339e0989169$exports, "LinkedAccount", () => $ff5f4339e0989169$export$70ab2a322f92399);
$parcel$export($ff5f4339e0989169$exports, "SmsMfaMethod", () => $ff5f4339e0989169$export$3250154351927c42);
$parcel$export($ff5f4339e0989169$exports, "TotpMfaMethod", () => $ff5f4339e0989169$export$27837b14d6717da8);
$parcel$export($ff5f4339e0989169$exports, "PasskeyMfaMethod", () => $ff5f4339e0989169$export$dd0fcfda522da705);
$parcel$export($ff5f4339e0989169$exports, "LinkedMfaMethod", () => $ff5f4339e0989169$export$7d2a870b119c9ff4);
$parcel$export($ff5f4339e0989169$exports, "User", () => $ff5f4339e0989169$export$1f44aaf2ec115b54);
$parcel$export($ff5f4339e0989169$exports, "SessionUpdateActionEnum", () => $ff5f4339e0989169$export$bbd8e1b014079dc1);
$parcel$export($ff5f4339e0989169$exports, "AuthenticatedUser", () => $ff5f4339e0989169$export$7de20aee312d6767);
$parcel$export($ff5f4339e0989169$exports, "LoggedOutUser", () => $ff5f4339e0989169$export$8339543958fe900f);
$parcel$export($ff5f4339e0989169$exports, "MaybeUser", () => $ff5f4339e0989169$export$2b6251d563a4e626);


const $ff5f4339e0989169$export$b291b2a8281f162e = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("email"),
    address: (0, $83Qas$z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$e461068a75eb0fb3 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("phone"),
    phoneNumber: (0, $83Qas$z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$e72ed3751242ab66 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("wallet"),
    address: (0, $83Qas$z).string(),
    chain_id: (0, $83Qas$z).string().optional(),
    chain_type: (0, $83Qas$z).literal("ethereum"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $83Qas$z).literal("unknown"),
    wallet_client_type: (0, $83Qas$z).string().optional(),
    connector_type: (0, $83Qas$z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$d08a38b431a135bc = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("wallet"),
    address: (0, $83Qas$z).string(),
    chain_type: (0, $83Qas$z).literal("solana"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $83Qas$z).literal("unknown"),
    wallet_client_type: (0, $83Qas$z).string().optional(),
    connector_type: (0, $83Qas$z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$70b11e99555e84cb = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("farcaster"),
    fid: (0, $83Qas$z).number(),
    owner_address: (0, $aab3573de5b449b0$export$e1ae011f7e31ddec),
    username: (0, $83Qas$z).string().optional(),
    display_name: (0, $83Qas$z).string().optional(),
    bio: (0, $83Qas$z).string().optional(),
    // The `profile_picture` interface is deprecated in favor of `profile_picture_url`
    // as of `react-auth@1.55.2`.
    profile_picture: (0, $83Qas$z).string().optional(),
    profile_picture_url: (0, $83Qas$z).string().optional(),
    homepage_url: (0, $83Qas$z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable(),
    signer_public_key: (0, $83Qas$z).string().optional()
});
const $ff5f4339e0989169$export$c7ead6bbfb2c42a0 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("passkey"),
    created_with_browser: (0, $83Qas$z).string().optional(),
    created_with_os: (0, $83Qas$z).string().optional(),
    created_with_device: (0, $83Qas$z).string().optional(),
    credential_id: (0, $83Qas$z).string(),
    authenticator_name: (0, $83Qas$z).string().optional(),
    enrolled_in_mfa: (0, $83Qas$z).boolean(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$70ddd50daefc295f = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("telegram"),
    telegram_user_id: (0, $83Qas$z).string(),
    first_name: (0, $83Qas$z).string().nullish(),
    last_name: (0, $83Qas$z).string().nullish(),
    username: (0, $83Qas$z).string().nullish(),
    photo_url: (0, $83Qas$z).string().nullish(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
}).transform((value)=>({
        ...value,
        // TODO: We keep this for backwards compatibility. Remove this after we make sure no one uses this version of the response
        telegramUserId: value.telegram_user_id,
        firstName: value.first_name
    }));
const $ff5f4339e0989169$export$42e79fc33a58a2f5 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("wallet"),
    address: (0, $83Qas$z).string(),
    imported: (0, $83Qas$z).boolean().default(false),
    wallet_index: (0, $83Qas$z).number(),
    // CAIP-2 chain ID for Ethereum mainnet: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md
    chain_id: (0, $83Qas$z).literal("eip155:1"),
    chain_type: (0, $83Qas$z).literal("ethereum"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $83Qas$z).literal("privy"),
    wallet_client_type: (0, $83Qas$z).literal("privy"),
    connector_type: (0, $83Qas$z).literal("embedded"),
    recovery_method: (0, $83Qas$z).union([
        (0, $83Qas$z).literal("privy"),
        (0, $83Qas$z).literal("user-passcode"),
        (0, $83Qas$z).literal("google-drive"),
        (0, $83Qas$z).literal("icloud")
    ]),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$734c44f2e990ba65 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("wallet"),
    address: (0, $83Qas$z).string(),
    public_key: (0, $83Qas$z).string(),
    chain_type: (0, $83Qas$z).literal("solana"),
    // CAIP-2 chain ID for Solana mainnet based on CAIP-10 account address definition: https://namespaces.chainagnostic.org/solana/caip10
    chain_id: (0, $83Qas$z).literal("solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp"),
    imported: (0, $83Qas$z).boolean().default(false),
    wallet_client_type: (0, $83Qas$z).literal("privy"),
    connector_type: (0, $83Qas$z).literal("embedded"),
    // For now we only support automatic recovery, but eventually will support the other methods.
    // The EVM accounts connected to this seed must ALSO be recoverable by the same method.
    recovery_method: (0, $83Qas$z).literal("privy"),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$ab3c40bebc4d6f36 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("google_oauth"),
    subject: (0, $83Qas$z).string(),
    email: (0, $83Qas$z).string(),
    name: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$31cc32c7bc65c2e0 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("twitter_oauth"),
    subject: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullable(),
    name: (0, $83Qas$z).string().nullable(),
    profile_picture_url: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$cb35596f96141125 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("discord_oauth"),
    subject: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullable(),
    email: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$55430b29b82d98a8 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("github_oauth"),
    subject: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullable(),
    name: (0, $83Qas$z).string().nullable(),
    email: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$25b982740d2e431c = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("linkedin_oauth"),
    subject: (0, $83Qas$z).string(),
    name: (0, $83Qas$z).string().optional(),
    email: (0, $83Qas$z).string().nullable(),
    vanity_name: (0, $83Qas$z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$260592be4b1062f4 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("spotify_oauth"),
    subject: (0, $83Qas$z).string(),
    email: (0, $83Qas$z).string().nullable(),
    name: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$c683d35f732c1912 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("instagram_oauth"),
    subject: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$5e8143c3f32ac816 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("tiktok_oauth"),
    subject: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullable(),
    name: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$67af8a2e34850611 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("apple_oauth"),
    subject: (0, $83Qas$z).string(),
    email: (0, $83Qas$z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$5da87a1cf22b57d0 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("custom_auth"),
    custom_user_id: (0, $83Qas$z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$ccd62fbf4b025fee = (0, $83Qas$z).object({
    address: (0, $83Qas$z).string()
});
const $ff5f4339e0989169$export$8f1f4ad4f8276ad7 = (0, $83Qas$z).object({
    address: (0, $83Qas$z).string()
});
const $ff5f4339e0989169$export$564035ac82295c40 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("cross_app"),
    subject: (0, $83Qas$z).string(),
    provider_app_id: (0, $83Qas$z).string(),
    embedded_wallets: (0, $83Qas$z).array($ff5f4339e0989169$export$ccd62fbf4b025fee),
    smart_wallets: (0, $83Qas$z).array($ff5f4339e0989169$export$8f1f4ad4f8276ad7),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $83Qas$z).number(),
    first_verified_at: (0, $83Qas$z).number().nullable(),
    latest_verified_at: (0, $83Qas$z).number().nullable()
});
const $ff5f4339e0989169$export$70ab2a322f92399 = (0, $83Qas$z).union([
    $ff5f4339e0989169$export$b291b2a8281f162e,
    $ff5f4339e0989169$export$e461068a75eb0fb3,
    $ff5f4339e0989169$export$e72ed3751242ab66,
    $ff5f4339e0989169$export$d08a38b431a135bc,
    $ff5f4339e0989169$export$42e79fc33a58a2f5,
    $ff5f4339e0989169$export$734c44f2e990ba65,
    $ff5f4339e0989169$export$ab3c40bebc4d6f36,
    $ff5f4339e0989169$export$31cc32c7bc65c2e0,
    $ff5f4339e0989169$export$cb35596f96141125,
    $ff5f4339e0989169$export$55430b29b82d98a8,
    $ff5f4339e0989169$export$260592be4b1062f4,
    $ff5f4339e0989169$export$c683d35f732c1912,
    $ff5f4339e0989169$export$5e8143c3f32ac816,
    $ff5f4339e0989169$export$25b982740d2e431c,
    $ff5f4339e0989169$export$67af8a2e34850611,
    $ff5f4339e0989169$export$5da87a1cf22b57d0,
    $ff5f4339e0989169$export$70b11e99555e84cb,
    $ff5f4339e0989169$export$c7ead6bbfb2c42a0,
    $ff5f4339e0989169$export$70ddd50daefc295f,
    $ff5f4339e0989169$export$564035ac82295c40
]);
const $ff5f4339e0989169$export$3250154351927c42 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("sms"),
    verified_at: (0, $83Qas$z).number()
});
const $ff5f4339e0989169$export$27837b14d6717da8 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("totp"),
    verified_at: (0, $83Qas$z).number()
});
const $ff5f4339e0989169$export$dd0fcfda522da705 = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("passkey"),
    verified_at: (0, $83Qas$z).number()
});
const $ff5f4339e0989169$export$7d2a870b119c9ff4 = (0, $83Qas$z).union([
    $ff5f4339e0989169$export$3250154351927c42,
    $ff5f4339e0989169$export$27837b14d6717da8,
    $ff5f4339e0989169$export$dd0fcfda522da705
]);
const $ff5f4339e0989169$export$1f44aaf2ec115b54 = (0, $83Qas$z).object({
    id: (0, $83Qas$z).string(),
    linked_accounts: (0, $83Qas$z).array($ff5f4339e0989169$export$70ab2a322f92399),
    mfa_methods: (0, $83Qas$z).array($ff5f4339e0989169$export$7d2a870b119c9ff4),
    created_at: (0, $83Qas$z).number(),
    has_accepted_terms: (0, $83Qas$z).boolean(),
    is_guest: (0, $83Qas$z).boolean(),
    custom_metadata: (0, $83Qas$z).record((0, $83Qas$z).string(), (0, $83Qas$z).string()).optional()
});
const $ff5f4339e0989169$export$bbd8e1b014079dc1 = (0, $83Qas$z).enum([
    "set",
    "ignore",
    "clear"
]);
const $ff5f4339e0989169$export$7de20aee312d6767 = (0, $83Qas$z).object({
    user: $ff5f4339e0989169$export$1f44aaf2ec115b54,
    token: (0, $83Qas$z).string().min(1).nullable(),
    refresh_token: (0, $83Qas$z).string().min(1).nullable(),
    identity_token: (0, $83Qas$z).string().min(1).optional(),
    is_new_user: (0, $83Qas$z).boolean().optional(),
    oauth_tokens: (0, $83Qas$z).object({
        provider: (0, $83Qas$z).string(),
        access_token: (0, $83Qas$z).string(),
        // The following additional parameters are optional as some OAuth providers may not return them.
        // Being conservative with marking them optional for now.
        access_token_expires_in_seconds: (0, $83Qas$z).number().optional(),
        refresh_token: (0, $83Qas$z).string().optional(),
        refresh_token_expires_in_seconds: (0, $83Qas$z).number().optional(),
        scopes: (0, $83Qas$z).array((0, $83Qas$z).string()).optional()
    }).optional(),
    // TODO: deprecate once Deform moves off of the legacy method
    oauth_provider_tokens: (0, $83Qas$z).object({
        provider: (0, $83Qas$z).string(),
        access_token: (0, $83Qas$z).string(),
        // The following additional parameters are optional as some OAuth providers may not return them.
        // Being conservative with marking them optional for now.
        access_token_expires_in_seconds: (0, $83Qas$z).number().optional(),
        refresh_token: (0, $83Qas$z).string().optional(),
        scopes: (0, $83Qas$z).array((0, $83Qas$z).string()).optional()
    }).optional(),
    /**
   * Instructs the client on how to handle tokens received from /sessions endpoint
   * Seeks to mimic the behavior of cookies with manual storage
   *
   * - `set`: _write both tokens to storage_
   * - `clear`: _remove tokens from storage_
   * - `ignore`: _leave tokens storage unchanged_
   */ session_update_action: $ff5f4339e0989169$export$bbd8e1b014079dc1
});
const $ff5f4339e0989169$export$8339543958fe900f = (0, $83Qas$z).object({
    user: (0, $83Qas$z).null(),
    token: (0, $83Qas$z).null(),
    refresh_token: (0, $83Qas$z).null(),
    /** Instructs the client to clear any persisted tokens */ session_update_action: (0, $83Qas$z).literal($ff5f4339e0989169$export$bbd8e1b014079dc1.Values.clear)
});
const $ff5f4339e0989169$export$2b6251d563a4e626 = (0, $83Qas$z).union([
    $ff5f4339e0989169$export$7de20aee312d6767,
    $ff5f4339e0989169$export$8339543958fe900f
]);


var $ef9e821724d8ed45$exports = {};

$parcel$export($ef9e821724d8ed45$exports, "MfaPasskeyInitInput", () => $ef9e821724d8ed45$export$2d6bddbbdba11994);
$parcel$export($ef9e821724d8ed45$exports, "ResponsePasskeyInitMfa", () => $ef9e821724d8ed45$export$94c076e2d3cd7355);
$parcel$export($ef9e821724d8ed45$exports, "MfaPasskeyVerifyInput", () => $ef9e821724d8ed45$export$afc24607eeb5f33e);
$parcel$export($ef9e821724d8ed45$exports, "MfaEnrollmentPasskeyInput", () => $ef9e821724d8ed45$export$3c909b91f5e6545f);


const $ef9e821724d8ed45$export$2d6bddbbdba11994 = (0, $83Qas$zod).object({
    relying_party: (0, $83Qas$zod).string().url().optional()
}).strict();
const $ef9e821724d8ed45$export$94c076e2d3cd7355 = (0, $83Qas$zod).object({
    options: (0, $e4527135bfc60402$export$7b01305e0ab33ea3)
}).strict();
const $ef9e821724d8ed45$export$afc24607eeb5f33e = (0, $83Qas$zod).object({
    authenticator_response: (0, $e4527135bfc60402$export$96203b761bf7cadb),
    relying_party: (0, $83Qas$zod).string().url().optional()
}).strict();
const $ef9e821724d8ed45$export$3c909b91f5e6545f = (0, $83Qas$zod).object({
    credential_ids: (0, $83Qas$zod).array((0, $83Qas$zod).string())
});


var $23e46a9ca9430f57$exports = {};

$parcel$export($23e46a9ca9430f57$exports, "TelegramAuthenticateInput", () => $23e46a9ca9430f57$export$fb9405d8cbdd092d);
$parcel$export($23e46a9ca9430f57$exports, "TelegramLinkInput", () => $23e46a9ca9430f57$export$303a24e5db03ece6);
$parcel$export($23e46a9ca9430f57$exports, "TelegramUnlinkInput", () => $23e46a9ca9430f57$export$aad710dcc06a76fc);


const $23e46a9ca9430f57$export$fb9405d8cbdd092d = (0, $83Qas$zod).object({
    captcha_token: (0, $83Qas$zod).string().optional(),
    telegram_auth_result: (0, $83Qas$zod).object({
        id: (0, $83Qas$zod).number(),
        first_name: (0, $83Qas$zod).string(),
        auth_date: (0, $83Qas$zod).number(),
        hash: (0, $83Qas$zod).string(),
        username: (0, $83Qas$zod).string().optional(),
        last_name: (0, $83Qas$zod).string().optional(),
        photo_url: (0, $83Qas$zod).string().optional()
    }).optional(),
    telegram_web_app_data: (0, $83Qas$zod).object({
        // query id provided by Telegram. It's present when logging in from a chat button but it's not when logging in from a direct link to the mini-app
        query_id: (0, $83Qas$zod).string().optional(),
        // This is the timestamp the auth result was verified at, in seconds from epoch
        auth_date: (0, $83Qas$zod).number(),
        // hash provided by Telegram to verify the payload is authentic
        hash: (0, $83Qas$zod).string(),
        // user data provided by Telegram, it is a JSON string with a shape similar to TelegramAuthResult
        user: (0, $83Qas$zod).string(),
        // chat instance provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        chat_instance: (0, $83Qas$zod).string().optional(),
        // chat type provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        chat_type: (0, $83Qas$zod).string().optional(),
        // start param provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        start_param: (0, $83Qas$zod).string().optional()
    }).optional()
}).merge((0, $aab3573de5b449b0$export$1d53282181ae94da));
const $23e46a9ca9430f57$export$303a24e5db03ece6 = $23e46a9ca9430f57$export$fb9405d8cbdd092d.omit({
    telegram_web_app_data: true
}).required({
    telegram_auth_result: true
});
const $23e46a9ca9430f57$export$aad710dcc06a76fc = (0, $83Qas$zod).object({
    telegram_user_id: (0, $83Qas$zod).string()
});


var $af2ad3d86e19473c$exports = {};

$parcel$export($af2ad3d86e19473c$exports, "CrossAppConnectionsResponse", () => $af2ad3d86e19473c$export$39ad28598f91ae95);

const $af2ad3d86e19473c$export$39ad28598f91ae95 = (0, $83Qas$z).object({
    connections: (0, $83Qas$z).array((0, $83Qas$z).object({
        provider_app_id: (0, $83Qas$z).string(),
        provider_app_name: (0, $83Qas$z).string(),
        provider_app_icon_url: (0, $83Qas$z).string().url().nullable(),
        provider_app_custom_api_url: (0, $83Qas$z).string().url().nullable(),
        read_only: (0, $83Qas$z).boolean()
    }))
});


var $7bd1f4246cdff428$exports = {};

$parcel$export($7bd1f4246cdff428$exports, "AuthorizationCodeInput", () => $7bd1f4246cdff428$export$b20be2048fbb4ed4);

const $7bd1f4246cdff428$export$b20be2048fbb4ed4 = (0, $83Qas$z).object({
    redirect_to: (0, $83Qas$z).string(),
    state: (0, $83Qas$z).string(),
    code_challenge: (0, $83Qas$z).string()
});


var $aac8cc805e9d28dc$exports = {};

$parcel$export($aac8cc805e9d28dc$exports, "EmailIdTokenEntry", () => $aac8cc805e9d28dc$export$783ee7c583b70bea);
$parcel$export($aac8cc805e9d28dc$exports, "PhoneIdTokenEntry", () => $aac8cc805e9d28dc$export$5050d0079ef11bc6);
$parcel$export($aac8cc805e9d28dc$exports, "EthereumIdTokenEntry", () => $aac8cc805e9d28dc$export$2986f977b4f702ee);
$parcel$export($aac8cc805e9d28dc$exports, "FarcasterIdTokenEntry", () => $aac8cc805e9d28dc$export$7d91b89a98a32206);
$parcel$export($aac8cc805e9d28dc$exports, "PasskeyIdTokenEntry", () => $aac8cc805e9d28dc$export$3da0109396521ad1);
$parcel$export($aac8cc805e9d28dc$exports, "TelegramIdTokenEntry", () => $aac8cc805e9d28dc$export$883001ac4b913adc);
$parcel$export($aac8cc805e9d28dc$exports, "EthereumEmbeddedWalletIdTokenEntry", () => $aac8cc805e9d28dc$export$19eeeea304fd874e);
$parcel$export($aac8cc805e9d28dc$exports, "SolanaEmbeddedWalletIdTokenEntry", () => $aac8cc805e9d28dc$export$fb08b43dc8cf7e59);
$parcel$export($aac8cc805e9d28dc$exports, "GoogleOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$10bcfd05b1db0bfb);
$parcel$export($aac8cc805e9d28dc$exports, "TwitterOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$5cf36acae8eed78);
$parcel$export($aac8cc805e9d28dc$exports, "DiscordOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$18e770e71ddcac8b);
$parcel$export($aac8cc805e9d28dc$exports, "GithubOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$2fd8fb76725390c8);
$parcel$export($aac8cc805e9d28dc$exports, "LinkedInOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$7325b1b9ff5b138f);
$parcel$export($aac8cc805e9d28dc$exports, "SpotifyOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$5991ad07a5ac6061);
$parcel$export($aac8cc805e9d28dc$exports, "InstagramOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$d50f791175bcd25a);
$parcel$export($aac8cc805e9d28dc$exports, "TiktokOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$ad548be3bcd74188);
$parcel$export($aac8cc805e9d28dc$exports, "AppleOauthIdTokenEntry", () => $aac8cc805e9d28dc$export$b8a150e8882d7617);
$parcel$export($aac8cc805e9d28dc$exports, "CustomJwtIdTokenEntry", () => $aac8cc805e9d28dc$export$4a1709b5a6406ed8);
$parcel$export($aac8cc805e9d28dc$exports, "CrossAppIdTokenEntry", () => $aac8cc805e9d28dc$export$23abd82a93e245c8);
// This file contains zod schemas for the linked account entries that appear with the `user` claim of
// Privy's identity token. These are a subset of the full linked account object that only contains
// critical data about the account (e.g. type and address for email). When developers request more fields
// about an account to be returned in the identity token, we can add them to the schemas here.


const $aac8cc805e9d28dc$export$783ee7c583b70bea = (0, $ff5f4339e0989169$export$b291b2a8281f162e).pick({
    type: true,
    address: true
});
const $aac8cc805e9d28dc$export$5050d0079ef11bc6 = (0, $ff5f4339e0989169$export$e461068a75eb0fb3).pick({
    type: true,
    phoneNumber: true
}).transform(({ type: type, phoneNumber: phoneNumber })=>({
        type: type,
        phone_number: phoneNumber
    }));
const $aac8cc805e9d28dc$export$2986f977b4f702ee = (0, $ff5f4339e0989169$export$e72ed3751242ab66).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $aac8cc805e9d28dc$export$7d91b89a98a32206 = (0, $ff5f4339e0989169$export$70b11e99555e84cb).pick({
    type: true,
    fid: true,
    username: true
});
const $aac8cc805e9d28dc$export$3da0109396521ad1 = (0, $ff5f4339e0989169$export$c7ead6bbfb2c42a0).pick({
    type: true,
    credential_id: true
});
const $aac8cc805e9d28dc$export$883001ac4b913adc = (0, $83Qas$z).object({
    type: (0, $83Qas$z).literal("telegram"),
    telegram_user_id: (0, $83Qas$z).string(),
    username: (0, $83Qas$z).string().nullish()
});
const $aac8cc805e9d28dc$export$19eeeea304fd874e = (0, $ff5f4339e0989169$export$42e79fc33a58a2f5).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $aac8cc805e9d28dc$export$fb08b43dc8cf7e59 = (0, $ff5f4339e0989169$export$734c44f2e990ba65).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $aac8cc805e9d28dc$export$10bcfd05b1db0bfb = (0, $ff5f4339e0989169$export$ab3c40bebc4d6f36).pick({
    type: true,
    subject: true,
    email: true
});
const $aac8cc805e9d28dc$export$5cf36acae8eed78 = (0, $ff5f4339e0989169$export$31cc32c7bc65c2e0).pick({
    type: true,
    subject: true,
    username: true
});
const $aac8cc805e9d28dc$export$18e770e71ddcac8b = (0, $ff5f4339e0989169$export$cb35596f96141125).pick({
    type: true,
    subject: true,
    username: true
});
const $aac8cc805e9d28dc$export$2fd8fb76725390c8 = (0, $ff5f4339e0989169$export$55430b29b82d98a8).pick({
    type: true,
    subject: true,
    username: true
});
const $aac8cc805e9d28dc$export$7325b1b9ff5b138f = (0, $ff5f4339e0989169$export$25b982740d2e431c).pick({
    type: true,
    subject: true,
    email: true
});
const $aac8cc805e9d28dc$export$5991ad07a5ac6061 = (0, $ff5f4339e0989169$export$260592be4b1062f4).pick({
    type: true,
    subject: true,
    email: true
});
const $aac8cc805e9d28dc$export$d50f791175bcd25a = (0, $ff5f4339e0989169$export$c683d35f732c1912).pick({
    type: true,
    subject: true,
    username: true
});
const $aac8cc805e9d28dc$export$ad548be3bcd74188 = (0, $ff5f4339e0989169$export$5e8143c3f32ac816).pick({
    type: true,
    subject: true,
    username: true
});
const $aac8cc805e9d28dc$export$b8a150e8882d7617 = (0, $ff5f4339e0989169$export$67af8a2e34850611).pick({
    type: true,
    subject: true,
    email: true
});
const $aac8cc805e9d28dc$export$4a1709b5a6406ed8 = (0, $ff5f4339e0989169$export$5da87a1cf22b57d0).pick({
    type: true,
    custom_user_id: true
});
const $aac8cc805e9d28dc$export$23abd82a93e245c8 = (0, $ff5f4339e0989169$export$564035ac82295c40).pick({
    type: true,
    subject: true,
    provider_app_id: true,
    embedded_wallets: true,
    smart_wallets: true
});


var $51582535fc36093f$exports = {};

$parcel$export($51582535fc36093f$exports, "CoinbaseOnRampInitInput", () => $51582535fc36093f$export$f01d538aa036a41a);
$parcel$export($51582535fc36093f$exports, "CoinbaseOnRampInitResponse", () => $51582535fc36093f$export$4731ee7c27e49d6e);
$parcel$export($51582535fc36093f$exports, "CoinbaseOnRampStatusResponse", () => $51582535fc36093f$export$146ab3c80d47ce12);


const $51582535fc36093f$var$Blockchain = (0, $83Qas$z).enum([
    "ethereum",
    "bitcoin",
    "base",
    "avacchain",
    "optimism",
    "solana",
    "polygon",
    "arbitrum",
    "stellar"
]);
// Assets must be capitalized to work with Coinbase Onramp. We preserve 'eth' here for backwards
// compatibility with older SDKs when Coinbase Onramp did not require capitalization
const $51582535fc36093f$var$Assets = (0, $83Qas$z).enum([
    "eth",
    "ETH",
    "USDC"
]).transform((asset)=>asset.toUpperCase());
const $51582535fc36093f$export$f01d538aa036a41a = (0, $83Qas$z).object({
    addresses: (0, $83Qas$z).array((0, $83Qas$z).object({
        address: (0, $aab3573de5b449b0$export$e1ae011f7e31ddec),
        blockchains: (0, $83Qas$z).array($51582535fc36093f$var$Blockchain)
    })).max(1).min(1),
    assets: (0, $83Qas$z).array($51582535fc36093f$var$Assets).optional()
});
const $51582535fc36093f$export$4731ee7c27e49d6e = (0, $83Qas$z).object({
    app_id: (0, $83Qas$z).string(),
    session_token: (0, $83Qas$z).string(),
    channel_id: (0, $83Qas$z).string(),
    partner_user_id: (0, $83Qas$z).string()
});
const $51582535fc36093f$export$146ab3c80d47ce12 = (0, $83Qas$z).object({
    status: (0, $83Qas$z).enum([
        "pending",
        "success",
        "failure"
    ])
});


$parcel$exportWildcard($b119bab29258e0d9$exports, $812875ae794ecf76$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $fb8f2a65df112faf$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $8952c42c3be19228$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $0e5934ad39a0ffa1$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $87659a5aa8799d23$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $aab3573de5b449b0$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $8423ca9a1079f6d7$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $0777f26061100062$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $a666dad64df3d92a$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $bfb33230d402f72a$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $37a005f91989be72$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $8dc16b6adb6078fa$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $f91fb55607e72b51$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $e4527135bfc60402$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $dd6f6c1fe379a225$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $3cbf49bd3049c518$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $0b8e24f996a08aa3$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $c55f91d419fc395f$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $2a629451b24ff1c8$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $dbeea6fd043fc091$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $ff5f4339e0989169$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $ef9e821724d8ed45$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $23e46a9ca9430f57$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $af2ad3d86e19473c$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $7bd1f4246cdff428$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $aac8cc805e9d28dc$exports);
$parcel$exportWildcard($b119bab29258e0d9$exports, $51582535fc36093f$exports);




export {$b8155828bfbee467$export$30b7662faa307ee as AnalyticsEvent, $bf4e8a2ca74ea9d9$export$d67e1898834e3885 as AppConfig, $97c9540320dc068b$export$cca9af2746f9b569 as CustomJWTAuthenticate, $4a90d090fac716ad$export$7ed9dd4c28ecc8b6 as FarcasterInit, $4a90d090fac716ad$export$c5985ffc97c7fdbb as FarcasterAuthenticate, $4a90d090fac716ad$export$d51549470a941fce as FarcasterLink, $4a90d090fac716ad$export$6017b5ccb660b0c8 as FarcasterUnlink, $4a90d090fac716ad$export$a14f766edcdfe001 as FarcasterStatus, $2d6a6e81ed9cd5aa$export$bed08221a49a43ad as GuestAuthenticate, $93c65b76e5ca94a2$export$2ca9d9da76a9e126 as MfaPasswordlessSmsEnroll, $93c65b76e5ca94a2$export$3308cba1d8516f90 as MfaPasswordlessSmsVerify, $93c65b76e5ca94a2$export$cf4de95cb6da600f as MfaPasswordlessSmsInit, $93c65b76e5ca94a2$export$503f112538d6ca09 as MfaPasswordlessSmsUnenroll, $0c1aa725aa4f001d$export$8cde043d756d30a7 as MfaTotpEnroll, $0c1aa725aa4f001d$export$68c952ebd3c622ec as MfaTotpVerify, $0c1aa725aa4f001d$export$9cfb70b9ef75b776 as MfaTotpInit, $0c1aa725aa4f001d$export$2241a8a66d2e6498 as MfaTotpUnenroll, $918825365cfa7807$export$866d3936787f21ca as OAuthAuthenticate, $918825365cfa7807$export$7dd94108c3c957b6 as OAuthInit, $918825365cfa7807$export$b555c05c68965095 as OAuthLink, $918825365cfa7807$export$7246a40b726a26ea as OAuthUnlink, $7968453470d2e50e$export$2399ea1e38345a53 as PasswordlessAuthenticate, $7968453470d2e50e$export$90dfa8041dde28f7 as PasswordlessInit, $7968453470d2e50e$export$c9ba5b5d6d996387 as PasswordlessLink, $7968453470d2e50e$export$770e866201444bcd as PasswordlessUnlink, $7968453470d2e50e$export$95d6a76319eb3229 as PasswordlessUpdate, $ff2f0b212f862ec3$export$7e171ae579c30d39 as PasswordlessSmsAuthenticate, $ff2f0b212f862ec3$export$5c9a94090bb8ee8b as PasswordlessSmsInit, $ff2f0b212f862ec3$export$e1b8c2813604d498 as PasswordlessSmsLink, $ff2f0b212f862ec3$export$140d5f525bd53d81 as PasswordlessSmsUnlink, $ff2f0b212f862ec3$export$3a4526cc549be74f as PasswordlessSmsUpdate, $2047bf8d239cddb9$export$53205c78d6e0d37f as RecoveryKeyMaterial, $2047bf8d239cddb9$export$f75c08c106aa5263 as RecoveryOAuthInit, $2047bf8d239cddb9$export$c4a18348874706ae as RecoveryOAuthAuthenticate, $2047bf8d239cddb9$export$4af466c8016ee40a as RecoveryOAuthInitICloud, $2047bf8d239cddb9$export$7b1580cc69e35e8 as RecoveryOAuthCallbackICloudExpo, $2047bf8d239cddb9$export$bce86382d6c75077 as RecoveryConfigurationICloud, $48277fa58f01a4d6$export$99faf8b86d913c70 as RefreshSession, $48277fa58f01a4d6$export$cad1a703886b4e3a as Logout, $8263a276deed55ab$export$81e59ade100f8964 as SiweInit, $8263a276deed55ab$export$5caf5d9f95cfdd58 as SiweAuthenticate, $8263a276deed55ab$export$626623f305a79065 as SiweLink, $8263a276deed55ab$export$cb415236257a3f80 as SiweUnlink, $7b32172f6de04ef0$export$dd0e464f35c701ad as SiwsInit, $7b32172f6de04ef0$export$93f5e2370a853629 as SiwsAuthenticate, $7b32172f6de04ef0$export$c309d30a2a2815ee as SiwsLink, $7b32172f6de04ef0$export$553e646318a6c92f as SiwsUnlink, $5e8055405b755abe$export$42654a8cac183ef0 as FarcasterSignerInit, $5e8055405b755abe$export$7f204ee171a84aa2 as FarcasterSignerStatus, $bc8d7bb5b40ee608$export$5e38d17588971983 as MfaPasskeyInit, $bc8d7bb5b40ee608$export$52d08a59e63ca968 as MfaPasskeyVerify, $bc8d7bb5b40ee608$export$d79952c9effdb95b as MfaPasskeyEnrollment, $4af29a4ebf5db253$export$cae4629dc339a1d3 as AcceptTermsOnUser, $bfc4ba45a1fc4c5e$export$8cc9b733d1d27bf as PasskeyLink, $bfc4ba45a1fc4c5e$export$dd91b2a2913f179 as PasskeyAuthenticate, $bfc4ba45a1fc4c5e$export$fbd0752c8f04789a as PasskeyAuthenticateInit, $bfc4ba45a1fc4c5e$export$908a107a443c8819 as PasskeyLinkInit, $bfc4ba45a1fc4c5e$export$73b251c34ecfbbd2 as PasskeyUnlink, $aa12a54bc3b2d48d$export$7429ccdbcca0d2ad as TelegramAuthenticate, $aa12a54bc3b2d48d$export$db55e97c1ee79975 as TelegramLink, $aa12a54bc3b2d48d$export$513feadc9dae72fd as TelegramUnlink, $5d50be17907f558e$export$e2939f516ef264b4 as getPathWithParams, $9e6262beb329f0fb$export$b9aa74fef6e2a10e as GetCrossAppConnections, $656516c52c57c728$export$3251d162865f5fbf as OAuthProviderAuthorize, $656516c52c57c728$export$e2c685498411d101 as OAuthProviderVerify, $61a6e48966d976eb$export$4d795ff6fee0e993 as CoinbaseOnRampInit, $61a6e48966d976eb$export$56adef0ec94545fa as CoinbaseOnRampStatus, $812875ae794ecf76$export$5ac03ce02eff89f4 as AnalyticsEventInput, $fb8f2a65df112faf$export$570fdd1008d53ff0 as AccountIdFromPath, $fb8f2a65df112faf$export$85be2ce1a760a71d as AppIdFromPath, $fb8f2a65df112faf$export$68f5e1453c188a82 as Pagination, $8952c42c3be19228$export$dc5469a0fff5641e as AppResponseSchema, $0e5934ad39a0ffa1$export$95372738a64f3b79 as EmbeddedWalletConfigSchema, $0e5934ad39a0ffa1$export$367c50094464beeb as TelegramAuthConfigSchema, $0e5934ad39a0ffa1$export$fe80bb11c98cc392 as FundingMethodEnum, $0e5934ad39a0ffa1$export$f5d20e939383f95 as FundingMethodArraySchema, $0e5934ad39a0ffa1$export$4d23ce301550190a as FundingConfigSchema, $87659a5aa8799d23$export$ad1e2c0abc3d279d as AuthenticateJwtInput, $aab3573de5b449b0$export$22a69335b09c6066 as UserId, $aab3573de5b449b0$export$5a6b8595f311cbb0 as Email, $aab3573de5b449b0$export$6051224a4b788b5f as PasswordlessCode, $aab3573de5b449b0$export$e1ae011f7e31ddec as WalletAddress, $aab3573de5b449b0$export$52d99941c9f2e1af as normalizeEthereumAddress, $aab3573de5b449b0$export$a6c60903ce1c32fc as SolanaWalletAddress, $aab3573de5b449b0$export$3d623a7758cab61e as PhoneNumber, $aab3573de5b449b0$export$eeb94ade8d38f116 as normalizePhoneNumber, $aab3573de5b449b0$export$1987372119294ebb as isValidAddress, $aab3573de5b449b0$export$1d53282181ae94da as AuthenticateMode, $8423ca9a1079f6d7$export$ca08e252312baf4 as VerifyEmailInput, $8423ca9a1079f6d7$export$161958b06841b78e as AuthenticateEmailInput, $8423ca9a1079f6d7$export$ef1330eae1875774 as InitEmailInput, $8423ca9a1079f6d7$export$b0a53354e7257ebb as UnlinkEmailInput, $8423ca9a1079f6d7$export$9ad98d5ba2865624 as UpdateEmailInput, $0777f26061100062$export$d3aaee921473d84f as JsonWebKey, $0777f26061100062$export$c8fd35505bb1704a as JsonWebKeySet, $a666dad64df3d92a$export$6ddd112193bb4f42 as FarcasterConnectInitResponse, $a666dad64df3d92a$export$10e572b3d0f4fe4f as FarcasterConnectStatusCompletedResponse, $a666dad64df3d92a$export$430853814f05cdc4 as FarcasterConnectStatusPendingResponse, $a666dad64df3d92a$export$2c59d6b32b518c6e as FarcasterAuthenticateInput, $a666dad64df3d92a$export$40d6546edc9ccfd3 as FarcasterLinkInput, $a666dad64df3d92a$export$8dd30088cf36d3f5 as FarcasterInitInput, $a666dad64df3d92a$export$325ad26071643d09 as FarcasterUnlinkInput, $bfb33230d402f72a$export$494be569a247ec9a as AuthenticateGuestInput, $37a005f91989be72$export$2a073761fe65eeb as MfaVerifyResponse, $37a005f91989be72$export$5f85d1e59e4b8093 as MfaSmsInitInput, $37a005f91989be72$export$d79e41bdbeb33bc8 as MfaSmsAuthenticateEnrollInput, $37a005f91989be72$export$44ba6bee6f8c49bc as MfaSmsEnrollInput, $37a005f91989be72$export$826840b7b6a37c22 as MfaSmsVerifyInput, $37a005f91989be72$export$c5079467361455e6 as MfaTotpInput, $37a005f91989be72$export$2e0b8857111f6a95 as ResponseTotpInitMfa, $8dc16b6adb6078fa$export$16a6de659e03cb8c as ExternalOAuthProvider, $8dc16b6adb6078fa$export$96d4e4cf27d59030 as OAuthProvider, $8dc16b6adb6078fa$export$3a815e8ae7821b6e as AuthenticateOauthInput, $8dc16b6adb6078fa$export$25d68298fd675d8b as OAuthInitInput, $8dc16b6adb6078fa$export$5169dcfff85dfbff as LinkOAuthInput, $8dc16b6adb6078fa$export$100da24bb64f0e09 as OAuthUnlinkInput, $8dc16b6adb6078fa$export$98ade43b6adcef0b as OAuthInitResponse, $7d4d0e60b9c76b9d$export$b5266d2768e29008 as SUPPORTED_OAUTH_PROVIDERS, $f91fb55607e72b51$export$2441567ff18621e0 as EmptyObject, $f91fb55607e72b51$export$8aed050e83c45a78 as SuccessObject, $e4527135bfc60402$export$7b01305e0ab33ea3 as PasskeyAuthenticatorVerifyOptions, $e4527135bfc60402$export$1c8fd79a94eba09d as PasskeyAuthenticatorEnrollmentOptions, $e4527135bfc60402$export$96203b761bf7cadb as PasskeyAuthenticatorVerifyResponse, $e4527135bfc60402$export$49424bcfec0159aa as PasskeyAuthenticatorEnrollmentResponse, $e4527135bfc60402$export$8c48c68e39a52ea as PasskeyInitInput, $e4527135bfc60402$export$2741be68c8db9b43 as ResponsePasskeyInitAuthenticate, $e4527135bfc60402$export$b968e9f4fdc5f0b2 as ResponsePasskeyInitLink, $e4527135bfc60402$export$9bd14c1a22f65d2 as PasskeyLinkInput, $e4527135bfc60402$export$8525265affc59e4b as PasskeyAuthenticateInput, $e4527135bfc60402$export$1d32dd5b05e1d4dd as UnlinkPasskeyInput, $dd6f6c1fe379a225$export$5133f8b99ced310e as RecoveryKeyMaterialInput, $dd6f6c1fe379a225$export$52cacd413024ecd as RecoveryKeyMaterialResponse, $dd6f6c1fe379a225$export$8d23e6b066801647 as OAuthAuthenticateRecoveryResponse, $dd6f6c1fe379a225$export$eb513813549270a as OAuthInitRecoveryInput, $dd6f6c1fe379a225$export$425e6a14ed44c52f as OAuthInitICloudRecoveryInput, $dd6f6c1fe379a225$export$bec1efaa57e00983 as OAuthCallbackICloudExpoInput, $dd6f6c1fe379a225$export$ea90bf451d6cb856 as OAuthCallbackICloudExpoResponse, $dd6f6c1fe379a225$export$dfb75001bba12cf3 as RecoveryConfigurationICloudInput, $dd6f6c1fe379a225$export$a53156909e5ca052 as RecoveryConfigurationICloudResponse, $3cbf49bd3049c518$export$1c8c55e22e1d8d9 as SiweInput, $3cbf49bd3049c518$export$3435c4883fbaeb7f as AuthenticateSiweInput, $3cbf49bd3049c518$export$b5b65f417dc17888 as SiweAddressInput, $3cbf49bd3049c518$export$6c957bbdcf62135c as SiweInitInput, $3cbf49bd3049c518$export$8535d2f477730820 as SiweNonce, $0b8e24f996a08aa3$export$fa2883d1fd7d2f4d as SiwsInput, $0b8e24f996a08aa3$export$defac42dc31eb54f as AuthenticateSiwsInput, $0b8e24f996a08aa3$export$767d61e052aa619c as SiwsAddressInput, $0b8e24f996a08aa3$export$334006f5cf1e2e10 as SiwsInitInput, $0b8e24f996a08aa3$export$678e538743698961 as SiwsNonce, $c55f91d419fc395f$export$809802d572486db as FarcasterSignerInitInput, $c55f91d419fc395f$export$327461233dec5bf6 as FarcasterSignerInitResponse, $c55f91d419fc395f$export$2a58492abd647af5 as FarcasterSignerStatusResponse, $2a629451b24ff1c8$export$1bbb8302b8922d52 as VerifyPhoneInput, $2a629451b24ff1c8$export$d57783821ffbf377 as AuthenticatePhoneInput, $2a629451b24ff1c8$export$794b219acd817434 as PasswordlessSmsPhoneInput, $2a629451b24ff1c8$export$63fc94c9f0d4eea8 as UnlinkPhoneInput, $2a629451b24ff1c8$export$bbf6b1d1316adec as UpdatePhoneInput, $dbeea6fd043fc091$export$6aa3af8166ea5386 as RefreshTokenInput, $dbeea6fd043fc091$export$3ea9bae16087a241 as OptionalRefreshTokenInput, $dbeea6fd043fc091$export$874b0538e4f42518 as ForkedToken, $ff5f4339e0989169$export$b291b2a8281f162e as EmailAccount, $ff5f4339e0989169$export$e461068a75eb0fb3 as PhoneAccount, $ff5f4339e0989169$export$e72ed3751242ab66 as EthereumAccount, $ff5f4339e0989169$export$d08a38b431a135bc as SolanaAccount, $ff5f4339e0989169$export$70b11e99555e84cb as FarcasterAccount, $ff5f4339e0989169$export$c7ead6bbfb2c42a0 as PasskeyAccount, $ff5f4339e0989169$export$70ddd50daefc295f as TelegramAccount, $ff5f4339e0989169$export$42e79fc33a58a2f5 as EthereumEmbeddedWalletAccount, $ff5f4339e0989169$export$734c44f2e990ba65 as SolanaEmbeddedWalletAccount, $ff5f4339e0989169$export$ab3c40bebc4d6f36 as GoogleOauthAccount, $ff5f4339e0989169$export$31cc32c7bc65c2e0 as TwitterOauthAccount, $ff5f4339e0989169$export$cb35596f96141125 as DiscordOauthAccount, $ff5f4339e0989169$export$55430b29b82d98a8 as GithubOauthAccount, $ff5f4339e0989169$export$25b982740d2e431c as LinkedInOauthAccount, $ff5f4339e0989169$export$260592be4b1062f4 as SpotifyOauthAccount, $ff5f4339e0989169$export$c683d35f732c1912 as InstagramOauthAccount, $ff5f4339e0989169$export$5e8143c3f32ac816 as TiktokOauthAccount, $ff5f4339e0989169$export$67af8a2e34850611 as AppleOauthAccount, $ff5f4339e0989169$export$5da87a1cf22b57d0 as CustomJwtAccount, $ff5f4339e0989169$export$ccd62fbf4b025fee as CrossAppEmbeddedWallet, $ff5f4339e0989169$export$8f1f4ad4f8276ad7 as CrossAppSmartWallet, $ff5f4339e0989169$export$564035ac82295c40 as CrossAppAccount, $ff5f4339e0989169$export$70ab2a322f92399 as LinkedAccount, $ff5f4339e0989169$export$3250154351927c42 as SmsMfaMethod, $ff5f4339e0989169$export$27837b14d6717da8 as TotpMfaMethod, $ff5f4339e0989169$export$dd0fcfda522da705 as PasskeyMfaMethod, $ff5f4339e0989169$export$7d2a870b119c9ff4 as LinkedMfaMethod, $ff5f4339e0989169$export$1f44aaf2ec115b54 as User, $ff5f4339e0989169$export$bbd8e1b014079dc1 as SessionUpdateActionEnum, $ff5f4339e0989169$export$7de20aee312d6767 as AuthenticatedUser, $ff5f4339e0989169$export$8339543958fe900f as LoggedOutUser, $ff5f4339e0989169$export$2b6251d563a4e626 as MaybeUser, $ef9e821724d8ed45$export$2d6bddbbdba11994 as MfaPasskeyInitInput, $ef9e821724d8ed45$export$94c076e2d3cd7355 as ResponsePasskeyInitMfa, $ef9e821724d8ed45$export$afc24607eeb5f33e as MfaPasskeyVerifyInput, $ef9e821724d8ed45$export$3c909b91f5e6545f as MfaEnrollmentPasskeyInput, $23e46a9ca9430f57$export$fb9405d8cbdd092d as TelegramAuthenticateInput, $23e46a9ca9430f57$export$303a24e5db03ece6 as TelegramLinkInput, $23e46a9ca9430f57$export$aad710dcc06a76fc as TelegramUnlinkInput, $af2ad3d86e19473c$export$39ad28598f91ae95 as CrossAppConnectionsResponse, $7bd1f4246cdff428$export$b20be2048fbb4ed4 as AuthorizationCodeInput, $aac8cc805e9d28dc$export$783ee7c583b70bea as EmailIdTokenEntry, $aac8cc805e9d28dc$export$5050d0079ef11bc6 as PhoneIdTokenEntry, $aac8cc805e9d28dc$export$2986f977b4f702ee as EthereumIdTokenEntry, $aac8cc805e9d28dc$export$7d91b89a98a32206 as FarcasterIdTokenEntry, $aac8cc805e9d28dc$export$3da0109396521ad1 as PasskeyIdTokenEntry, $aac8cc805e9d28dc$export$883001ac4b913adc as TelegramIdTokenEntry, $aac8cc805e9d28dc$export$19eeeea304fd874e as EthereumEmbeddedWalletIdTokenEntry, $aac8cc805e9d28dc$export$fb08b43dc8cf7e59 as SolanaEmbeddedWalletIdTokenEntry, $aac8cc805e9d28dc$export$10bcfd05b1db0bfb as GoogleOauthIdTokenEntry, $aac8cc805e9d28dc$export$5cf36acae8eed78 as TwitterOauthIdTokenEntry, $aac8cc805e9d28dc$export$18e770e71ddcac8b as DiscordOauthIdTokenEntry, $aac8cc805e9d28dc$export$2fd8fb76725390c8 as GithubOauthIdTokenEntry, $aac8cc805e9d28dc$export$7325b1b9ff5b138f as LinkedInOauthIdTokenEntry, $aac8cc805e9d28dc$export$5991ad07a5ac6061 as SpotifyOauthIdTokenEntry, $aac8cc805e9d28dc$export$d50f791175bcd25a as InstagramOauthIdTokenEntry, $aac8cc805e9d28dc$export$ad548be3bcd74188 as TiktokOauthIdTokenEntry, $aac8cc805e9d28dc$export$b8a150e8882d7617 as AppleOauthIdTokenEntry, $aac8cc805e9d28dc$export$4a1709b5a6406ed8 as CustomJwtIdTokenEntry, $aac8cc805e9d28dc$export$23abd82a93e245c8 as CrossAppIdTokenEntry, $51582535fc36093f$export$f01d538aa036a41a as CoinbaseOnRampInitInput, $51582535fc36093f$export$4731ee7c27e49d6e as CoinbaseOnRampInitResponse, $51582535fc36093f$export$146ab3c80d47ce12 as CoinbaseOnRampStatusResponse};
