// @privy-io/public-api/dist/index.js
var $3LmS4$zod = require("zod");
var $3LmS4$privyioapibase = require("@privy-io/api-base");
var $3LmS4$bs58 = require("bs58");
var $3LmS4$ethers = require("ethers");
var $3LmS4$libphonenumberjsmax = require("libphonenumber-js/max");


function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || Object.prototype.hasOwnProperty.call(dest, key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $4b4e9f0316150d93$exports = {};
var $a0d8576077966429$exports = {};

$parcel$export($a0d8576077966429$exports, "AnalyticsEvent", () => $a0d8576077966429$export$30b7662faa307ee);
const $a0d8576077966429$export$30b7662faa307ee = {
    path: "/api/v1/analytics_events",
    method: "POST"
};


var $817900df99c0e13a$exports = {};

$parcel$export($817900df99c0e13a$exports, "AppConfig", () => $817900df99c0e13a$export$d67e1898834e3885);
const $817900df99c0e13a$export$d67e1898834e3885 = {
    path: "/api/v1/apps/:app_id",
    method: "GET"
};


var $de58c2d12ac26593$exports = {};

$parcel$export($de58c2d12ac26593$exports, "CustomJWTAuthenticate", () => $de58c2d12ac26593$export$cca9af2746f9b569);
const $de58c2d12ac26593$export$cca9af2746f9b569 = {
    path: "/api/v1/custom_jwt_account/authenticate",
    method: "POST"
};


var $f40b85231d3cfbcd$exports = {};

$parcel$export($f40b85231d3cfbcd$exports, "FarcasterInit", () => $f40b85231d3cfbcd$export$7ed9dd4c28ecc8b6);
$parcel$export($f40b85231d3cfbcd$exports, "FarcasterAuthenticate", () => $f40b85231d3cfbcd$export$c5985ffc97c7fdbb);
$parcel$export($f40b85231d3cfbcd$exports, "FarcasterLink", () => $f40b85231d3cfbcd$export$d51549470a941fce);
$parcel$export($f40b85231d3cfbcd$exports, "FarcasterUnlink", () => $f40b85231d3cfbcd$export$6017b5ccb660b0c8);
$parcel$export($f40b85231d3cfbcd$exports, "FarcasterStatus", () => $f40b85231d3cfbcd$export$a14f766edcdfe001);
const $f40b85231d3cfbcd$export$7ed9dd4c28ecc8b6 = {
    path: "/api/v1/farcaster/init",
    method: "POST"
};
const $f40b85231d3cfbcd$export$c5985ffc97c7fdbb = {
    path: "/api/v1/farcaster/authenticate",
    method: "POST"
};
const $f40b85231d3cfbcd$export$d51549470a941fce = {
    path: "/api/v1/farcaster/link",
    method: "POST"
};
const $f40b85231d3cfbcd$export$6017b5ccb660b0c8 = {
    path: "/api/v1/farcaster/unlink",
    method: "POST"
};
const $f40b85231d3cfbcd$export$a14f766edcdfe001 = {
    path: "/api/v1/farcaster/status",
    method: "GET"
};


var $a90b3840c091e164$exports = {};

$parcel$export($a90b3840c091e164$exports, "GuestAuthenticate", () => $a90b3840c091e164$export$bed08221a49a43ad);
const $a90b3840c091e164$export$bed08221a49a43ad = {
    path: "/api/v1/guest/authenticate",
    method: "POST"
};


var $ca6478b87782fb21$exports = {};

$parcel$export($ca6478b87782fb21$exports, "MfaPasswordlessSmsEnroll", () => $ca6478b87782fb21$export$2ca9d9da76a9e126);
$parcel$export($ca6478b87782fb21$exports, "MfaPasswordlessSmsVerify", () => $ca6478b87782fb21$export$3308cba1d8516f90);
$parcel$export($ca6478b87782fb21$exports, "MfaPasswordlessSmsInit", () => $ca6478b87782fb21$export$cf4de95cb6da600f);
$parcel$export($ca6478b87782fb21$exports, "MfaPasswordlessSmsUnenroll", () => $ca6478b87782fb21$export$503f112538d6ca09);
const $ca6478b87782fb21$export$2ca9d9da76a9e126 = {
    path: "/api/v1/mfa/passwordless_sms/enroll",
    method: "POST"
};
const $ca6478b87782fb21$export$3308cba1d8516f90 = {
    path: "/api/v1/mfa/passwordless_sms/verify",
    method: "POST"
};
const $ca6478b87782fb21$export$cf4de95cb6da600f = {
    path: "/api/v1/mfa/passwordless_sms/init",
    method: "POST"
};
const $ca6478b87782fb21$export$503f112538d6ca09 = {
    path: "/api/v1/mfa/passwordless_sms/unenroll",
    method: "POST"
};


var $a89c10acee2ab376$exports = {};

$parcel$export($a89c10acee2ab376$exports, "MfaTotpEnroll", () => $a89c10acee2ab376$export$8cde043d756d30a7);
$parcel$export($a89c10acee2ab376$exports, "MfaTotpVerify", () => $a89c10acee2ab376$export$68c952ebd3c622ec);
$parcel$export($a89c10acee2ab376$exports, "MfaTotpInit", () => $a89c10acee2ab376$export$9cfb70b9ef75b776);
$parcel$export($a89c10acee2ab376$exports, "MfaTotpUnenroll", () => $a89c10acee2ab376$export$2241a8a66d2e6498);
const $a89c10acee2ab376$export$8cde043d756d30a7 = {
    path: "/api/v1/mfa/totp/enroll",
    method: "POST"
};
const $a89c10acee2ab376$export$68c952ebd3c622ec = {
    path: "/api/v1/mfa/totp/verify",
    method: "POST"
};
const $a89c10acee2ab376$export$9cfb70b9ef75b776 = {
    path: "/api/v1/mfa/totp/init",
    method: "POST"
};
const $a89c10acee2ab376$export$2241a8a66d2e6498 = {
    path: "/api/v1/mfa/totp/unenroll",
    method: "POST"
};


var $2bdc1b6c51deda64$exports = {};

$parcel$export($2bdc1b6c51deda64$exports, "OAuthAuthenticate", () => $2bdc1b6c51deda64$export$866d3936787f21ca);
$parcel$export($2bdc1b6c51deda64$exports, "OAuthInit", () => $2bdc1b6c51deda64$export$7dd94108c3c957b6);
$parcel$export($2bdc1b6c51deda64$exports, "OAuthLink", () => $2bdc1b6c51deda64$export$b555c05c68965095);
$parcel$export($2bdc1b6c51deda64$exports, "OAuthUnlink", () => $2bdc1b6c51deda64$export$7246a40b726a26ea);
const $2bdc1b6c51deda64$export$866d3936787f21ca = {
    path: "/api/v1/oauth/authenticate",
    method: "POST"
};
const $2bdc1b6c51deda64$export$7dd94108c3c957b6 = {
    path: "/api/v1/oauth/init",
    method: "POST"
};
const $2bdc1b6c51deda64$export$b555c05c68965095 = {
    path: "/api/v1/oauth/link",
    method: "POST"
};
const $2bdc1b6c51deda64$export$7246a40b726a26ea = {
    path: "/api/v1/oauth/unlink",
    method: "POST"
};


var $ef7b73ae2aafc8b1$exports = {};

$parcel$export($ef7b73ae2aafc8b1$exports, "PasswordlessAuthenticate", () => $ef7b73ae2aafc8b1$export$2399ea1e38345a53);
$parcel$export($ef7b73ae2aafc8b1$exports, "PasswordlessInit", () => $ef7b73ae2aafc8b1$export$90dfa8041dde28f7);
$parcel$export($ef7b73ae2aafc8b1$exports, "PasswordlessLink", () => $ef7b73ae2aafc8b1$export$c9ba5b5d6d996387);
$parcel$export($ef7b73ae2aafc8b1$exports, "PasswordlessUnlink", () => $ef7b73ae2aafc8b1$export$770e866201444bcd);
$parcel$export($ef7b73ae2aafc8b1$exports, "PasswordlessUpdate", () => $ef7b73ae2aafc8b1$export$95d6a76319eb3229);
const $ef7b73ae2aafc8b1$export$2399ea1e38345a53 = {
    path: "/api/v1/passwordless/authenticate",
    method: "POST"
};
const $ef7b73ae2aafc8b1$export$90dfa8041dde28f7 = {
    path: "/api/v1/passwordless/init",
    method: "POST"
};
const $ef7b73ae2aafc8b1$export$c9ba5b5d6d996387 = {
    path: "/api/v1/passwordless/link",
    method: "POST"
};
const $ef7b73ae2aafc8b1$export$770e866201444bcd = {
    path: "/api/v1/passwordless/unlink",
    method: "POST"
};
const $ef7b73ae2aafc8b1$export$95d6a76319eb3229 = {
    path: "/api/v1/passwordless/update",
    method: "POST"
};


var $b3452035fae6153a$exports = {};

$parcel$export($b3452035fae6153a$exports, "PasswordlessSmsAuthenticate", () => $b3452035fae6153a$export$7e171ae579c30d39);
$parcel$export($b3452035fae6153a$exports, "PasswordlessSmsInit", () => $b3452035fae6153a$export$5c9a94090bb8ee8b);
$parcel$export($b3452035fae6153a$exports, "PasswordlessSmsLink", () => $b3452035fae6153a$export$e1b8c2813604d498);
$parcel$export($b3452035fae6153a$exports, "PasswordlessSmsUnlink", () => $b3452035fae6153a$export$140d5f525bd53d81);
$parcel$export($b3452035fae6153a$exports, "PasswordlessSmsUpdate", () => $b3452035fae6153a$export$3a4526cc549be74f);
const $b3452035fae6153a$export$7e171ae579c30d39 = {
    path: "/api/v1/passwordless_sms/authenticate",
    method: "POST"
};
const $b3452035fae6153a$export$5c9a94090bb8ee8b = {
    path: "/api/v1/passwordless_sms/init",
    method: "POST"
};
const $b3452035fae6153a$export$e1b8c2813604d498 = {
    path: "/api/v1/passwordless_sms/link",
    method: "POST"
};
const $b3452035fae6153a$export$140d5f525bd53d81 = {
    path: "/api/v1/passwordless_sms/unlink",
    method: "POST"
};
const $b3452035fae6153a$export$3a4526cc549be74f = {
    path: "/api/v1/passwordless_sms/update",
    method: "POST"
};


var $e2481eec2b83d332$exports = {};

$parcel$export($e2481eec2b83d332$exports, "RecoveryKeyMaterial", () => $e2481eec2b83d332$export$53205c78d6e0d37f);
$parcel$export($e2481eec2b83d332$exports, "RecoveryOAuthInit", () => $e2481eec2b83d332$export$f75c08c106aa5263);
$parcel$export($e2481eec2b83d332$exports, "RecoveryOAuthAuthenticate", () => $e2481eec2b83d332$export$c4a18348874706ae);
$parcel$export($e2481eec2b83d332$exports, "RecoveryOAuthInitICloud", () => $e2481eec2b83d332$export$4af466c8016ee40a);
$parcel$export($e2481eec2b83d332$exports, "RecoveryOAuthCallbackICloudExpo", () => $e2481eec2b83d332$export$7b1580cc69e35e8);
$parcel$export($e2481eec2b83d332$exports, "RecoveryConfigurationICloud", () => $e2481eec2b83d332$export$bce86382d6c75077);
const $e2481eec2b83d332$export$53205c78d6e0d37f = {
    path: "/api/v1/embedded_wallets/:address/recovery/key_material",
    method: "POST"
};
const $e2481eec2b83d332$export$f75c08c106aa5263 = {
    path: "/api/v1/recovery/oauth/init",
    method: "POST"
};
const $e2481eec2b83d332$export$c4a18348874706ae = {
    path: "/api/v1/recovery/oauth/authenticate",
    method: "POST"
};
const $e2481eec2b83d332$export$4af466c8016ee40a = {
    path: "/api/v1/recovery/oauth/init_icloud",
    method: "POST"
};
const $e2481eec2b83d332$export$7b1580cc69e35e8 = {
    path: "/api/v1/recovery/oauth/callback_icloud/:app_id/expo_ios",
    method: "GET"
};
const $e2481eec2b83d332$export$bce86382d6c75077 = {
    path: "/api/v1/recovery/configuration_icloud",
    method: "POST"
};


var $0dbb5f16a83f0322$exports = {};

$parcel$export($0dbb5f16a83f0322$exports, "RefreshSession", () => $0dbb5f16a83f0322$export$99faf8b86d913c70);
$parcel$export($0dbb5f16a83f0322$exports, "Logout", () => $0dbb5f16a83f0322$export$cad1a703886b4e3a);
const $0dbb5f16a83f0322$export$99faf8b86d913c70 = {
    path: "/api/v1/sessions",
    method: "POST"
};
const $0dbb5f16a83f0322$export$cad1a703886b4e3a = {
    path: "/api/v1/sessions/logout",
    method: "POST"
};


var $f2adde655311edcf$exports = {};

$parcel$export($f2adde655311edcf$exports, "SiweInit", () => $f2adde655311edcf$export$81e59ade100f8964);
$parcel$export($f2adde655311edcf$exports, "SiweAuthenticate", () => $f2adde655311edcf$export$5caf5d9f95cfdd58);
$parcel$export($f2adde655311edcf$exports, "SiweLink", () => $f2adde655311edcf$export$626623f305a79065);
$parcel$export($f2adde655311edcf$exports, "SiweUnlink", () => $f2adde655311edcf$export$cb415236257a3f80);
const $f2adde655311edcf$export$81e59ade100f8964 = {
    path: "/api/v1/siwe/init",
    method: "POST"
};
const $f2adde655311edcf$export$5caf5d9f95cfdd58 = {
    path: "/api/v1/siwe/authenticate",
    method: "POST"
};
const $f2adde655311edcf$export$626623f305a79065 = {
    path: "/api/v1/siwe/link",
    method: "POST"
};
const $f2adde655311edcf$export$cb415236257a3f80 = {
    path: "/api/v1/siwe/unlink",
    method: "POST"
};


var $4fb3d0a369c051d5$exports = {};

$parcel$export($4fb3d0a369c051d5$exports, "SiwsInit", () => $4fb3d0a369c051d5$export$dd0e464f35c701ad);
$parcel$export($4fb3d0a369c051d5$exports, "SiwsAuthenticate", () => $4fb3d0a369c051d5$export$93f5e2370a853629);
$parcel$export($4fb3d0a369c051d5$exports, "SiwsLink", () => $4fb3d0a369c051d5$export$c309d30a2a2815ee);
$parcel$export($4fb3d0a369c051d5$exports, "SiwsUnlink", () => $4fb3d0a369c051d5$export$553e646318a6c92f);
const $4fb3d0a369c051d5$export$dd0e464f35c701ad = {
    path: "/api/v1/siws/init",
    method: "POST"
};
const $4fb3d0a369c051d5$export$93f5e2370a853629 = {
    path: "/api/v1/siws/authenticate",
    method: "POST"
};
const $4fb3d0a369c051d5$export$c309d30a2a2815ee = {
    path: "/api/v1/siws/link",
    method: "POST"
};
const $4fb3d0a369c051d5$export$553e646318a6c92f = {
    path: "/api/v1/siws/unlink",
    method: "POST"
};


var $1d54822e9a448421$exports = {};

$parcel$export($1d54822e9a448421$exports, "FarcasterSignerInit", () => $1d54822e9a448421$export$42654a8cac183ef0);
$parcel$export($1d54822e9a448421$exports, "FarcasterSignerStatus", () => $1d54822e9a448421$export$7f204ee171a84aa2);
const $1d54822e9a448421$export$42654a8cac183ef0 = {
    path: "/api/v1/farcaster/signer/init",
    method: "POST"
};
const $1d54822e9a448421$export$7f204ee171a84aa2 = {
    path: "/api/v1/farcaster/signer/status",
    method: "POST"
};


var $9c59835f3068be20$exports = {};

$parcel$export($9c59835f3068be20$exports, "MfaPasskeyInit", () => $9c59835f3068be20$export$5e38d17588971983);
$parcel$export($9c59835f3068be20$exports, "MfaPasskeyVerify", () => $9c59835f3068be20$export$52d08a59e63ca968);
$parcel$export($9c59835f3068be20$exports, "MfaPasskeyEnrollment", () => $9c59835f3068be20$export$d79952c9effdb95b);
const $9c59835f3068be20$export$5e38d17588971983 = {
    path: "/api/v1/mfa/passkey/init",
    method: "POST"
};
const $9c59835f3068be20$export$52d08a59e63ca968 = {
    path: "/api/v1/mfa/passkey/verify",
    method: "POST"
};
const $9c59835f3068be20$export$d79952c9effdb95b = {
    path: "/api/v1/mfa/passkey/enrollment",
    method: "POST"
};


var $a5a7e4d1a170ec26$exports = {};

$parcel$export($a5a7e4d1a170ec26$exports, "AcceptTermsOnUser", () => $a5a7e4d1a170ec26$export$cae4629dc339a1d3);
const $a5a7e4d1a170ec26$export$cae4629dc339a1d3 = {
    path: "/api/v1/users/me/accept_terms",
    method: "POST"
};


var $d3a1c54e835ed9a0$exports = {};

$parcel$export($d3a1c54e835ed9a0$exports, "PasskeyLink", () => $d3a1c54e835ed9a0$export$8cc9b733d1d27bf);
$parcel$export($d3a1c54e835ed9a0$exports, "PasskeyAuthenticate", () => $d3a1c54e835ed9a0$export$dd91b2a2913f179);
$parcel$export($d3a1c54e835ed9a0$exports, "PasskeyAuthenticateInit", () => $d3a1c54e835ed9a0$export$fbd0752c8f04789a);
$parcel$export($d3a1c54e835ed9a0$exports, "PasskeyLinkInit", () => $d3a1c54e835ed9a0$export$908a107a443c8819);
$parcel$export($d3a1c54e835ed9a0$exports, "PasskeyUnlink", () => $d3a1c54e835ed9a0$export$73b251c34ecfbbd2);
const $d3a1c54e835ed9a0$export$8cc9b733d1d27bf = {
    path: "/api/v1/passkeys/link",
    method: "POST"
};
const $d3a1c54e835ed9a0$export$dd91b2a2913f179 = {
    path: "/api/v1/passkeys/authenticate",
    method: "POST"
};
const $d3a1c54e835ed9a0$export$fbd0752c8f04789a = {
    path: "/api/v1/passkeys/authenticate/init",
    method: "POST"
};
const $d3a1c54e835ed9a0$export$908a107a443c8819 = {
    path: "/api/v1/passkeys/link/init",
    method: "POST"
};
const $d3a1c54e835ed9a0$export$73b251c34ecfbbd2 = {
    path: "/api/v1/passkeys/unlink",
    method: "POST"
};


var $fd67ba46b9dfe5c6$exports = {};

$parcel$export($fd67ba46b9dfe5c6$exports, "TelegramAuthenticate", () => $fd67ba46b9dfe5c6$export$7429ccdbcca0d2ad);
$parcel$export($fd67ba46b9dfe5c6$exports, "TelegramLink", () => $fd67ba46b9dfe5c6$export$db55e97c1ee79975);
$parcel$export($fd67ba46b9dfe5c6$exports, "TelegramUnlink", () => $fd67ba46b9dfe5c6$export$513feadc9dae72fd);
const $fd67ba46b9dfe5c6$export$7429ccdbcca0d2ad = {
    path: "/api/v1/telegram/authenticate",
    method: "POST"
};
const $fd67ba46b9dfe5c6$export$db55e97c1ee79975 = {
    path: "/api/v1/telegram/link",
    method: "POST"
};
const $fd67ba46b9dfe5c6$export$513feadc9dae72fd = {
    path: "/api/v1/telegram/unlink",
    method: "POST"
};


var $17696c15666b26c3$exports = {};

$parcel$export($17696c15666b26c3$exports, "getPathWithParams", () => $17696c15666b26c3$export$e2939f516ef264b4);
const $17696c15666b26c3$export$e2939f516ef264b4 = (path, params)=>{
    if (!params) return path;
    return Object.entries(params).reduce((acc, [key, value])=>{
        return acc.replace(`:${key}`, `${value}`);
    }, path);
};


var $1894e6f7e3d075ef$exports = {};

$parcel$export($1894e6f7e3d075ef$exports, "GetCrossAppConnections", () => $1894e6f7e3d075ef$export$b9aa74fef6e2a10e);
const $1894e6f7e3d075ef$export$b9aa74fef6e2a10e = {
    path: "/api/v1/apps/:app_id/cross-app/connections",
    method: "GET"
};


var $a7b22576260d45a9$exports = {};

$parcel$export($a7b22576260d45a9$exports, "OAuthProviderAuthorize", () => $a7b22576260d45a9$export$3251d162865f5fbf);
$parcel$export($a7b22576260d45a9$exports, "OAuthProviderVerify", () => $a7b22576260d45a9$export$e2c685498411d101);
const $a7b22576260d45a9$export$3251d162865f5fbf = {
    path: "/oauth/authorization_code",
    method: "POST"
};
const $a7b22576260d45a9$export$e2c685498411d101 = {
    path: "/oauth/verify",
    method: "POST"
};


var $71f6f2eb8352bb65$exports = {};

$parcel$export($71f6f2eb8352bb65$exports, "CoinbaseOnRampInit", () => $71f6f2eb8352bb65$export$4d795ff6fee0e993);
$parcel$export($71f6f2eb8352bb65$exports, "CoinbaseOnRampStatus", () => $71f6f2eb8352bb65$export$56adef0ec94545fa);
const $71f6f2eb8352bb65$export$4d795ff6fee0e993 = {
    path: "/api/v1/funding/coinbase_on_ramp/init",
    method: "POST"
};
const $71f6f2eb8352bb65$export$56adef0ec94545fa = {
    path: "/api/v1/funding/coinbase_on_ramp/status",
    method: "GET"
};


$parcel$exportWildcard($4b4e9f0316150d93$exports, $a0d8576077966429$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $817900df99c0e13a$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $de58c2d12ac26593$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $f40b85231d3cfbcd$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $a90b3840c091e164$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $ca6478b87782fb21$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $a89c10acee2ab376$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $2bdc1b6c51deda64$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $ef7b73ae2aafc8b1$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $b3452035fae6153a$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $e2481eec2b83d332$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $0dbb5f16a83f0322$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $f2adde655311edcf$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $4fb3d0a369c051d5$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $1d54822e9a448421$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $9c59835f3068be20$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $a5a7e4d1a170ec26$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $d3a1c54e835ed9a0$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $fd67ba46b9dfe5c6$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $17696c15666b26c3$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $1894e6f7e3d075ef$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $a7b22576260d45a9$exports);
$parcel$exportWildcard($4b4e9f0316150d93$exports, $71f6f2eb8352bb65$exports);


var $7d14bc6e1da56942$exports = {};
var $6d13bfa775813546$exports = {};

$parcel$export($6d13bfa775813546$exports, "AnalyticsEventInput", () => $6d13bfa775813546$export$5ac03ce02eff89f4);

const $6d13bfa775813546$export$5ac03ce02eff89f4 = (0, $3LmS4$zod.z).object({
    event_name: (0, $3LmS4$zod.z).string().max(255),
    client_id: (0, $3LmS4$zod.z).string().max(255).nullable(),
    payload: (0, $3LmS4$zod.z).record((0, $3LmS4$zod.z).any()).optional()
});


var $920ac747dab06b9c$exports = {};

$parcel$export($920ac747dab06b9c$exports, "AccountIdFromPath", () => $920ac747dab06b9c$export$570fdd1008d53ff0);
$parcel$export($920ac747dab06b9c$exports, "AppIdFromPath", () => $920ac747dab06b9c$export$85be2ce1a760a71d);
$parcel$export($920ac747dab06b9c$exports, "Pagination", () => $920ac747dab06b9c$export$68f5e1453c188a82);


var $d69d14a89141bb36$exports = {};

$parcel$export($d69d14a89141bb36$exports, "UserId", () => $d69d14a89141bb36$export$22a69335b09c6066);
$parcel$export($d69d14a89141bb36$exports, "Email", () => $d69d14a89141bb36$export$5a6b8595f311cbb0);
$parcel$export($d69d14a89141bb36$exports, "PasswordlessCode", () => $d69d14a89141bb36$export$6051224a4b788b5f);
$parcel$export($d69d14a89141bb36$exports, "WalletAddress", () => $d69d14a89141bb36$export$e1ae011f7e31ddec);
$parcel$export($d69d14a89141bb36$exports, "normalizeEthereumAddress", () => $d69d14a89141bb36$export$52d99941c9f2e1af);
$parcel$export($d69d14a89141bb36$exports, "SolanaWalletAddress", () => $d69d14a89141bb36$export$a6c60903ce1c32fc);
$parcel$export($d69d14a89141bb36$exports, "PhoneNumber", () => $d69d14a89141bb36$export$3d623a7758cab61e);
$parcel$export($d69d14a89141bb36$exports, "normalizePhoneNumber", () => $d69d14a89141bb36$export$eeb94ade8d38f116);
$parcel$export($d69d14a89141bb36$exports, "isValidAddress", () => $d69d14a89141bb36$export$1987372119294ebb);
$parcel$export($d69d14a89141bb36$exports, "AuthenticateMode", () => $d69d14a89141bb36$export$1d53282181ae94da);




const $d69d14a89141bb36$export$22a69335b09c6066 = (0, $3LmS4$zod.z).string({
    required_error: "Privy DID must be included",
    invalid_type_error: "Privy DID is not a valid string"
}).min(1).transform((id)=>id.replace("did:privy:", ""));
const $d69d14a89141bb36$export$5a6b8595f311cbb0 = (0, $3LmS4$zod.z).string({
    required_error: "Email address must be provided",
    invalid_type_error: "Email address must be a valid string"
}).email("Invalid email address").transform((email)=>email.toLowerCase());
const $d69d14a89141bb36$export$6051224a4b788b5f = (0, $3LmS4$zod.z).string({
    required_error: "Verification code must be provided",
    invalid_type_error: "Invalid verification code"
}).length(6, "Verification code must have 6 digits.");
const $d69d14a89141bb36$export$e1ae011f7e31ddec = (0, $3LmS4$zod.z).string({
    required_error: "Ethereum address must be provided",
    invalid_type_error: "Invalid Ethereum address"
}).transform((address, ctx)=>{
    // After verifying that the address is a string, we attempt to
    // checksum it per EIP-55.
    const normalizedAddress = $d69d14a89141bb36$export$52d99941c9f2e1af(address);
    // If our normalization method returns falsy, the address is not
    // a valid Ethereum address, and we mark the input as invalid.
    if (!normalizedAddress) {
        ctx.addIssue({
            code: (0, $3LmS4$zod.z).ZodIssueCode.custom,
            message: "Invalid Ethereum address"
        });
        return (0, $3LmS4$zod.z).NEVER;
    }
    // Otherwise, we return the normalized (checksummed) address.
    return normalizedAddress;
});
const $d69d14a89141bb36$export$a6c60903ce1c32fc = (0, $3LmS4$zod.z).string({
    required_error: "Solana address must be provided",
    invalid_type_error: "Invalid Solana address"
}).trim().transform((address, ctx)=>{
    try {
        // Try to decode. If it fails the address is not a valid base58 string.
        (0, ($parcel$interopDefault($3LmS4$bs58))).decode(address);
    } catch (error) {
        ctx.addIssue({
            code: (0, $3LmS4$zod.z).ZodIssueCode.custom,
            message: "Invalid Solana address"
        });
        return (0, $3LmS4$zod.z).NEVER;
    }
    return address;
});
const $d69d14a89141bb36$export$3d623a7758cab61e = (0, $3LmS4$zod.z).string({
    required_error: "Phone number must be included",
    invalid_type_error: "Phone number is not a valid string"
}).refine((val)=>{
    try {
        return $d69d14a89141bb36$export$eeb94ade8d38f116(val) !== undefined;
    } catch (e) {
        return false;
    }
}, {
    message: "Phone number is not valid"
}).transform((val)=>$d69d14a89141bb36$export$eeb94ade8d38f116(val) || ""); // This operation is safe because we previously refine the schema and break if it's undefined
function $d69d14a89141bb36$export$eeb94ade8d38f116(inputPhoneNumber) {
    const phoneNumber = (0, $3LmS4$libphonenumberjsmax.parsePhoneNumber)(inputPhoneNumber, "US");
    // https://www.npmjs.com/package/libphonenumber-js#using-phone-number-validation-feature
    if (phoneNumber?.isPossible()) return phoneNumber.formatInternational();
}
function $d69d14a89141bb36$export$1987372119294ebb(address) {
    return !!$d69d14a89141bb36$export$52d99941c9f2e1af(address);
}
function $d69d14a89141bb36$export$52d99941c9f2e1af(address) {
    if (typeof address !== "string") return;
    try {
        return (0, $3LmS4$ethers.ethers).utils.getAddress(address.trim());
    } catch (e) {}
}
const $d69d14a89141bb36$export$1d53282181ae94da = (0, $3LmS4$zod.z).object({
    mode: (0, $3LmS4$zod.z).enum([
        "no-signup",
        "login-or-sign-up"
    ]).optional()
});


const $920ac747dab06b9c$export$570fdd1008d53ff0 = (0, $3LmS4$zod.z).object({
    account_id: (0, $3LmS4$zod.z).string({
        required_error: "Account ID must be provided",
        invalid_type_error: "Account ID is not a valid string"
    }).min(1)
});
const $920ac747dab06b9c$export$85be2ce1a760a71d = (0, $3LmS4$zod.z).object({
    app_id: (0, $3LmS4$zod.z).string({
        required_error: "App ID must be provided",
        invalid_type_error: "App ID is not a valid string"
    }).min(1)
}).transform(({ app_id: app_id })=>({
        app_id: app_id,
        appId: app_id
    })).catch(({ error: error })=>{
    throw new (0, $3LmS4$privyioapibase.InvalidInputError)(error.message, (0, $3LmS4$privyioapibase.PrivyErrorCode).MISSING_OR_INVALID_PRIVY_APP_ID);
});
const $920ac747dab06b9c$export$68f5e1453c188a82 = (0, $3LmS4$zod.z).object({
    cursor: (0, $d69d14a89141bb36$export$22a69335b09c6066).optional(),
    limit: (0, $3LmS4$zod.z).coerce.number().max(100, {
        message: "Query limit can be at maximum 100 records."
    }).optional()
});


var $8a7996e6cd7430a9$exports = {};

$parcel$export($8a7996e6cd7430a9$exports, "AppResponseSchema", () => $8a7996e6cd7430a9$export$dc5469a0fff5641e);

var $d2ef8d1854e48841$exports = {};

$parcel$export($d2ef8d1854e48841$exports, "EmbeddedWalletConfigSchema", () => $d2ef8d1854e48841$export$95372738a64f3b79);
$parcel$export($d2ef8d1854e48841$exports, "TelegramAuthConfigSchema", () => $d2ef8d1854e48841$export$367c50094464beeb);
$parcel$export($d2ef8d1854e48841$exports, "FundingMethodEnum", () => $d2ef8d1854e48841$export$fe80bb11c98cc392);
$parcel$export($d2ef8d1854e48841$exports, "FundingMethodArraySchema", () => $d2ef8d1854e48841$export$f5d20e939383f95);
$parcel$export($d2ef8d1854e48841$exports, "FundingConfigSchema", () => $d2ef8d1854e48841$export$4d23ce301550190a);


const $74792b5343d1d804$var$SUPPORTED_CHAIN_TYPES = [
    "evm",
    "solana"
];
const $74792b5343d1d804$export$b733e092f9f44b2c = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    /**
   * CAIP-2 formatted chain ID
   * Resource: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md
   */ chain: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    // Defaulting to `native` token for now, so not specifying anything beyond chain
    // Currency asset
    asset: (0, ($parcel$interopDefault($3LmS4$zod))).enum([
        "native-currency",
        "USDC"
    ]).optional()
});


const $d2ef8d1854e48841$export$95372738a64f3b79 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    /**
   * Whether to create embedded wallets on login. Note that this still requires distinct API
   * calls, rather than the server creating one on the /authenticate call.
   *
   * Default: 'users-without-wallets'
   *
   * Overridable from client.
   */ create_on_login: (0, ($parcel$interopDefault($3LmS4$zod))).enum([
        "users-without-wallets",
        "all-users",
        "off"
    ]),
    /**
   * Will be one or more of `user-passcode` | `google-drive` | `icloud`
   * - all enabled methods other than `password` require prior configuration
   */ user_owned_recovery_options: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).enum([
        "user-passcode",
        "google-drive",
        "icloud"
    ])),
    /**
   * If true, this will prompt the user use one of the enabled recovery methods
   * to secure the recovery share of their embedded wallet.
   *
   * Otherwise (the default), Privy will secure the recovery share, and the embedded wallet
   * will be created without showing any UIs to the user.
   *
   * Overridable from client.
   */ require_user_owned_recovery_on_create: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional(),
    /**
   * @deprecated use `require_user_owned_recovery_on_create`
   *
   * If true, this will prompt the user to enter a password to secure the recovery share of
   * their embedded wallet.
   *
   * Otherwise (the default), Privy will secure the recovery share, and the embedded wallet
   * will be created without showing any UIs to the user.
   *
   * Overridable from client.
   */ require_user_password_on_create: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
});
const $d2ef8d1854e48841$export$367c50094464beeb = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    bot_id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    bot_name: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    link_enabled: (0, ($parcel$interopDefault($3LmS4$zod))).boolean(),
    seamless_auth_enabled: (0, ($parcel$interopDefault($3LmS4$zod))).boolean()
});
const $d2ef8d1854e48841$export$fe80bb11c98cc392 = (0, ($parcel$interopDefault($3LmS4$zod))).enum([
    "moonpay",
    "coinbase-onramp",
    "external"
]);
const $d2ef8d1854e48841$export$f5d20e939383f95 = (0, ($parcel$interopDefault($3LmS4$zod))).array($d2ef8d1854e48841$export$fe80bb11c98cc392);
const $d2ef8d1854e48841$export$4d23ce301550190a = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    default_recommended_currency: (0, $74792b5343d1d804$export$b733e092f9f44b2c),
    default_recommended_amount: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    methods: $d2ef8d1854e48841$export$f5d20e939383f95,
    prompt_funding_on_wallet_creation: (0, ($parcel$interopDefault($3LmS4$zod))).boolean()
});


const $8a7996e6cd7430a9$export$dc5469a0fff5641e = (0, $3LmS4$zod.z).object({
    id: (0, $3LmS4$zod.z).string(),
    name: (0, $3LmS4$zod.z).string(),
    logo_url: (0, $3LmS4$zod.z).string().nullable(),
    icon_url: (0, $3LmS4$zod.z).string().nullable(),
    terms_and_conditions_url: (0, $3LmS4$zod.z).string().nullable(),
    privacy_policy_url: (0, $3LmS4$zod.z).string().nullable(),
    require_users_accept_terms: (0, $3LmS4$zod.z).boolean().nullable(),
    theme: (0, $3LmS4$zod.z).string(),
    accent_color: (0, $3LmS4$zod.z).string().nullable(),
    show_wallet_login_first: (0, $3LmS4$zod.z).boolean(),
    allowed_domains: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()),
    allowed_native_app_ids: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()),
    allowed_native_app_url_schemes: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()),
    wallet_auth: (0, $3LmS4$zod.z).boolean(),
    email_auth: (0, $3LmS4$zod.z).boolean(),
    sms_auth: (0, $3LmS4$zod.z).boolean(),
    google_oauth: (0, $3LmS4$zod.z).boolean(),
    twitter_oauth: (0, $3LmS4$zod.z).boolean(),
    discord_oauth: (0, $3LmS4$zod.z).boolean(),
    github_oauth: (0, $3LmS4$zod.z).boolean(),
    spotify_oauth: (0, $3LmS4$zod.z).boolean(),
    instagram_oauth: (0, $3LmS4$zod.z).boolean(),
    tiktok_oauth: (0, $3LmS4$zod.z).boolean(),
    linkedin_oauth: (0, $3LmS4$zod.z).boolean(),
    apple_oauth: (0, $3LmS4$zod.z).boolean(),
    farcaster_auth: (0, $3LmS4$zod.z).boolean(),
    passkey_auth: (0, $3LmS4$zod.z).boolean(),
    telegram_auth: (0, $3LmS4$zod.z).boolean(),
    guest_auth: (0, $3LmS4$zod.z).boolean(),
    solana_wallet_auth: (0, $3LmS4$zod.z).boolean(),
    // TODO: allowlist_enabled should be moved to AdminOnlyAppResponseSchema but is technically in older
    // versions of the server-auth getAppSettings response
    allowlist_enabled: (0, $3LmS4$zod.z).boolean(),
    allowlist_config: (0, $3LmS4$zod.z).object({
        error_title: (0, $3LmS4$zod.z).string().nullable(),
        error_detail: (0, $3LmS4$zod.z).string().nullable(),
        cta_text: (0, $3LmS4$zod.z).string().nullable(),
        cta_link: (0, $3LmS4$zod.z).string().nullable()
    }),
    wallet_connect_cloud_project_id: (0, $3LmS4$zod.z).string().nullable(),
    custom_api_url: (0, $3LmS4$zod.z).string().nullable(),
    embedded_wallet_config: (0, $d2ef8d1854e48841$export$95372738a64f3b79),
    enforce_wallet_uis: (0, $3LmS4$zod.z).boolean(),
    legacy_wallet_ui_config: (0, $3LmS4$zod.z).boolean(),
    fiat_on_ramp_enabled: (0, $3LmS4$zod.z).boolean(),
    captcha_enabled: (0, $3LmS4$zod.z).boolean(),
    twitter_oauth_on_mobile_enabled: (0, $3LmS4$zod.z).boolean(),
    mfa_methods: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).enum([
        "sms",
        "totp",
        "passkey"
    ])),
    captcha_site_key: (0, $3LmS4$zod.z).string().optional(),
    verification_key: (0, $3LmS4$zod.z).string(),
    telegram_auth_config: (0, $d2ef8d1854e48841$export$367c50094464beeb).optional(),
    funding_config: (0, $d2ef8d1854e48841$export$4d23ce301550190a).optional(),
    max_linked_wallets_per_user: (0, $3LmS4$zod.z).number().nullable(),
    farcaster_link_wallets_enabled: (0, $3LmS4$zod.z).boolean()
});



var $a32571b6d32f2bac$exports = {};

$parcel$export($a32571b6d32f2bac$exports, "AuthenticateJwtInput", () => $a32571b6d32f2bac$export$ad1e2c0abc3d279d);

const $a32571b6d32f2bac$export$ad1e2c0abc3d279d = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    /** A JWT from the custom authentication provider */ token: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
});



var $ba5673beb0ead784$exports = {};

$parcel$export($ba5673beb0ead784$exports, "VerifyEmailInput", () => $ba5673beb0ead784$export$ca08e252312baf4);
$parcel$export($ba5673beb0ead784$exports, "AuthenticateEmailInput", () => $ba5673beb0ead784$export$161958b06841b78e);
$parcel$export($ba5673beb0ead784$exports, "InitEmailInput", () => $ba5673beb0ead784$export$ef1330eae1875774);
$parcel$export($ba5673beb0ead784$exports, "UnlinkEmailInput", () => $ba5673beb0ead784$export$b0a53354e7257ebb);
$parcel$export($ba5673beb0ead784$exports, "UpdateEmailInput", () => $ba5673beb0ead784$export$9ad98d5ba2865624);


const $ba5673beb0ead784$export$ca08e252312baf4 = (0, $3LmS4$zod.z).object({
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f),
    email: (0, $d69d14a89141bb36$export$5a6b8595f311cbb0)
});
const $ba5673beb0ead784$export$161958b06841b78e = $ba5673beb0ead784$export$ca08e252312baf4.merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $ba5673beb0ead784$export$ef1330eae1875774 = (0, $3LmS4$zod.z).object({
    email: (0, $d69d14a89141bb36$export$5a6b8595f311cbb0),
    token: (0, $3LmS4$zod.z).string().optional()
});
const $ba5673beb0ead784$export$b0a53354e7257ebb = (0, $3LmS4$zod.z).object({
    address: (0, $d69d14a89141bb36$export$5a6b8595f311cbb0)
});
const $ba5673beb0ead784$export$9ad98d5ba2865624 = (0, $3LmS4$zod.z).object({
    oldAddress: (0, $d69d14a89141bb36$export$5a6b8595f311cbb0),
    newAddress: (0, $d69d14a89141bb36$export$5a6b8595f311cbb0),
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f)
});


var $1e7d24a9b3fdde80$exports = {};

$parcel$export($1e7d24a9b3fdde80$exports, "JsonWebKey", () => $1e7d24a9b3fdde80$export$d3aaee921473d84f);
$parcel$export($1e7d24a9b3fdde80$exports, "JsonWebKeySet", () => $1e7d24a9b3fdde80$export$c8fd35505bb1704a);

const $1e7d24a9b3fdde80$export$d3aaee921473d84f = (0, $3LmS4$zod.z).object({
    kty: (0, $3LmS4$zod.z).string(),
    use: (0, $3LmS4$zod.z).enum([
        "sig",
        "enc"
    ]).optional(),
    key_ops: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).enum([
        "sign",
        "verify",
        "encrypt",
        "decrypt",
        "wrapKey",
        "unwrapKey",
        "deriveKey",
        "deriveBits"
    ])).optional(),
    alg: (0, $3LmS4$zod.z).enum([
        "RS256"
    ]),
    kid: (0, $3LmS4$zod.z).string().optional(),
    x5u: (0, $3LmS4$zod.z).string().optional(),
    x5c: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()).optional(),
    x5t: (0, $3LmS4$zod.z).string().optional(),
    "x5t#S256": (0, $3LmS4$zod.z).string().optional(),
    n: (0, $3LmS4$zod.z).string().optional(),
    e: (0, $3LmS4$zod.z).string().optional(),
    d: (0, $3LmS4$zod.z).string().optional()
});
const $1e7d24a9b3fdde80$export$c8fd35505bb1704a = (0, $3LmS4$zod.z).object({
    keys: (0, $3LmS4$zod.z).array($1e7d24a9b3fdde80$export$d3aaee921473d84f)
});


var $566f5ba79f307877$exports = {};

$parcel$export($566f5ba79f307877$exports, "FarcasterConnectInitResponse", () => $566f5ba79f307877$export$6ddd112193bb4f42);
$parcel$export($566f5ba79f307877$exports, "FarcasterConnectStatusCompletedResponse", () => $566f5ba79f307877$export$10e572b3d0f4fe4f);
$parcel$export($566f5ba79f307877$exports, "FarcasterConnectStatusPendingResponse", () => $566f5ba79f307877$export$430853814f05cdc4);
$parcel$export($566f5ba79f307877$exports, "FarcasterAuthenticateInput", () => $566f5ba79f307877$export$2c59d6b32b518c6e);
$parcel$export($566f5ba79f307877$exports, "FarcasterLinkInput", () => $566f5ba79f307877$export$40d6546edc9ccfd3);
$parcel$export($566f5ba79f307877$exports, "FarcasterInitInput", () => $566f5ba79f307877$export$8dd30088cf36d3f5);
$parcel$export($566f5ba79f307877$exports, "FarcasterUnlinkInput", () => $566f5ba79f307877$export$325ad26071643d09);


const $566f5ba79f307877$export$6ddd112193bb4f42 = (0, $3LmS4$zod.z).object({
    channel_token: (0, $3LmS4$zod.z).string(),
    connect_uri: (0, $3LmS4$zod.z).string()
});
const $566f5ba79f307877$export$10e572b3d0f4fe4f = (0, $3LmS4$zod.z).object({
    state: (0, $3LmS4$zod.z).literal("completed"),
    nonce: (0, $3LmS4$zod.z).string(),
    message: (0, $3LmS4$zod.z).string(),
    signature: (0, $3LmS4$zod.z).string(),
    fid: (0, $3LmS4$zod.z).number(),
    username: (0, $3LmS4$zod.z).string(),
    display_name: (0, $3LmS4$zod.z).string(),
    bio: (0, $3LmS4$zod.z).string(),
    pfp_url: (0, $3LmS4$zod.z).string()
});
const $566f5ba79f307877$export$430853814f05cdc4 = (0, $3LmS4$zod.z).object({
    state: (0, $3LmS4$zod.z).literal("pending"),
    nonce: (0, $3LmS4$zod.z).string()
});
const $566f5ba79f307877$export$2c59d6b32b518c6e = (0, $3LmS4$zod.z).object({
    channel_token: (0, $3LmS4$zod.z).string(),
    message: (0, $3LmS4$zod.z).string(),
    signature: (0, $3LmS4$zod.z).string(),
    fid: (0, $3LmS4$zod.z).number()
}).merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $566f5ba79f307877$export$40d6546edc9ccfd3 = (0, $3LmS4$zod.z).object({
    channel_token: (0, $3LmS4$zod.z).string(),
    message: (0, $3LmS4$zod.z).string(),
    signature: (0, $3LmS4$zod.z).string(),
    fid: (0, $3LmS4$zod.z).number()
});
const $566f5ba79f307877$export$8dd30088cf36d3f5 = (0, $3LmS4$zod.z).object({
    token: (0, $3LmS4$zod.z).string().optional(),
    redirect_url: (0, $3LmS4$zod.z).string().url().optional(),
    relying_party: (0, $3LmS4$zod.z).string().optional()
});
const $566f5ba79f307877$export$325ad26071643d09 = (0, $3LmS4$zod.z).object({
    fid: (0, $3LmS4$zod.z).number()
});


var $f7e0547cf75f2094$exports = {};

$parcel$export($f7e0547cf75f2094$exports, "AuthenticateGuestInput", () => $f7e0547cf75f2094$export$494be569a247ec9a);

const $f7e0547cf75f2094$export$494be569a247ec9a = (0, $3LmS4$zod.z).object({
    /**
     * Locally-persisted guest secret.
     * This credential should have as much entropy as embedded wallet entropy.
     */ guest_credential: (0, $3LmS4$zod.z).string().length(43)
}).strict();


var $63a1d8c3bf348ff9$exports = {};

$parcel$export($63a1d8c3bf348ff9$exports, "MfaVerifyResponse", () => $63a1d8c3bf348ff9$export$2a073761fe65eeb);
$parcel$export($63a1d8c3bf348ff9$exports, "MfaSmsInitInput", () => $63a1d8c3bf348ff9$export$5f85d1e59e4b8093);
$parcel$export($63a1d8c3bf348ff9$exports, "MfaSmsAuthenticateEnrollInput", () => $63a1d8c3bf348ff9$export$d79e41bdbeb33bc8);
$parcel$export($63a1d8c3bf348ff9$exports, "MfaSmsEnrollInput", () => $63a1d8c3bf348ff9$export$44ba6bee6f8c49bc);
$parcel$export($63a1d8c3bf348ff9$exports, "MfaSmsVerifyInput", () => $63a1d8c3bf348ff9$export$826840b7b6a37c22);
$parcel$export($63a1d8c3bf348ff9$exports, "MfaTotpInput", () => $63a1d8c3bf348ff9$export$c5079467361455e6);
$parcel$export($63a1d8c3bf348ff9$exports, "ResponseTotpInitMfa", () => $63a1d8c3bf348ff9$export$2e0b8857111f6a95);


const $63a1d8c3bf348ff9$export$2a073761fe65eeb = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    token: (0, ($parcel$interopDefault($3LmS4$zod))).string()
}).strict();
/**
 * SMS MFA enrollment and verification inputs.
 */ const $63a1d8c3bf348ff9$var$MfaSmsInitVerifyInput = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    action: (0, ($parcel$interopDefault($3LmS4$zod))).literal("verify")
}).strict();
const $63a1d8c3bf348ff9$var$MfaSmsInitEnrollInput = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    action: (0, ($parcel$interopDefault($3LmS4$zod))).literal("enroll"),
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e)
}).strict();
const $63a1d8c3bf348ff9$export$5f85d1e59e4b8093 = (0, ($parcel$interopDefault($3LmS4$zod))).union([
    $63a1d8c3bf348ff9$var$MfaSmsInitVerifyInput,
    $63a1d8c3bf348ff9$var$MfaSmsInitEnrollInput
]);
const $63a1d8c3bf348ff9$export$d79e41bdbeb33bc8 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    action: (0, ($parcel$interopDefault($3LmS4$zod))).literal("enroll").optional(),
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f),
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e)
}).strict();
const $63a1d8c3bf348ff9$export$44ba6bee6f8c49bc = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f),
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e)
}).strict();
const $63a1d8c3bf348ff9$export$826840b7b6a37c22 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f)
}).strict();
const $63a1d8c3bf348ff9$export$c5079467361455e6 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f)
}).strict();
const $63a1d8c3bf348ff9$export$2e0b8857111f6a95 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    totpSecret: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    totpAuthUrl: (0, ($parcel$interopDefault($3LmS4$zod))).string()
}).strict();


var $1a93a9315817630b$exports = {};

$parcel$export($1a93a9315817630b$exports, "ExternalOAuthProvider", () => $1a93a9315817630b$export$16a6de659e03cb8c);
$parcel$export($1a93a9315817630b$exports, "OAuthProvider", () => $1a93a9315817630b$export$96d4e4cf27d59030);
$parcel$export($1a93a9315817630b$exports, "AuthenticateOauthInput", () => $1a93a9315817630b$export$3a815e8ae7821b6e);
$parcel$export($1a93a9315817630b$exports, "OAuthInitInput", () => $1a93a9315817630b$export$25d68298fd675d8b);
$parcel$export($1a93a9315817630b$exports, "LinkOAuthInput", () => $1a93a9315817630b$export$5169dcfff85dfbff);
$parcel$export($1a93a9315817630b$exports, "OAuthUnlinkInput", () => $1a93a9315817630b$export$100da24bb64f0e09);
$parcel$export($1a93a9315817630b$exports, "OAuthInitResponse", () => $1a93a9315817630b$export$98ade43b6adcef0b);
$parcel$export($1a93a9315817630b$exports, "SUPPORTED_OAUTH_PROVIDERS", () => $bb47c7a05fd82065$export$b5266d2768e29008);


// Casting as const makes it immutable.
const $bb47c7a05fd82065$export$b5266d2768e29008 = [
    "google",
    "discord",
    "twitter",
    "github",
    "spotify",
    "instagram",
    "tiktok",
    "linkedin",
    "apple"
];



const $1a93a9315817630b$export$16a6de659e03cb8c = (0, $3LmS4$zod.z).enum((0, $bb47c7a05fd82065$export$b5266d2768e29008), {
    invalid_type_error: "Please specify a supported provider.",
    required_error: "Please specify a supported provider."
});
const $1a93a9315817630b$export$96d4e4cf27d59030 = $1a93a9315817630b$export$16a6de659e03cb8c.or((0, $3LmS4$zod.z).custom((value)=>{
    return typeof value === "string" ? /^privy:[a-z0-9]{25}$/i.test(value) : false;
}, {
    message: "Please specify a supported provider."
}));
const $1a93a9315817630b$export$3a815e8ae7821b6e = (0, $3LmS4$zod.z).object({
    authorization_code: (0, $3LmS4$zod.z).string(),
    state_code: (0, $3LmS4$zod.z).string(),
    code_verifier: (0, $3LmS4$zod.z).string().min(43).max(128).optional(),
    code_type: (0, $3LmS4$zod.z).enum([
        "raw"
    ]).optional()
}).merge((0, $d69d14a89141bb36$export$1d53282181ae94da)).catch(()=>{
    throw new (0, $3LmS4$privyioapibase.UnauthorizedError)("Please specify authorization_code and state_code.", (0, $3LmS4$privyioapibase.PrivyErrorCode).INVALID_CREDENTIALS);
});
const $1a93a9315817630b$export$25d68298fd675d8b = (0, $3LmS4$zod.z).object({
    redirect_to: (0, $3LmS4$zod.z).string({
        required_error: "Please provide a redirect_to value."
    }),
    provider: $1a93a9315817630b$export$96d4e4cf27d59030,
    token: (0, $3LmS4$zod.z).string().optional(),
    state_code: (0, $3LmS4$zod.z).string().optional(),
    code_challenge: (0, $3LmS4$zod.z).string().optional()
});
const $1a93a9315817630b$export$5169dcfff85dfbff = (0, $3LmS4$zod.z).object({
    authorization_code: (0, $3LmS4$zod.z).string(),
    state_code: (0, $3LmS4$zod.z).string(),
    code_verifier: (0, $3LmS4$zod.z).string().min(43).max(128).optional(),
    code_type: (0, $3LmS4$zod.z).enum([
        "raw"
    ]).optional()
});
const $1a93a9315817630b$export$100da24bb64f0e09 = (0, $3LmS4$zod.z).object({
    subject: (0, $3LmS4$zod.z).string(),
    provider: $1a93a9315817630b$export$96d4e4cf27d59030
});
const $1a93a9315817630b$export$98ade43b6adcef0b = (0, $3LmS4$zod.z).object({
    url: (0, $3LmS4$zod.z).string()
});


var $0dbe2ed1fbc05d05$exports = {};

$parcel$export($0dbe2ed1fbc05d05$exports, "EmptyObject", () => $0dbe2ed1fbc05d05$export$2441567ff18621e0);
$parcel$export($0dbe2ed1fbc05d05$exports, "SuccessObject", () => $0dbe2ed1fbc05d05$export$8aed050e83c45a78);

const $0dbe2ed1fbc05d05$export$2441567ff18621e0 = (0, $3LmS4$zod.z).record((0, $3LmS4$zod.z).string(), (0, $3LmS4$zod.z).never());
const $0dbe2ed1fbc05d05$export$8aed050e83c45a78 = (0, $3LmS4$zod.z).object({
    success: (0, $3LmS4$zod.z).boolean()
});


var $4e0c5fecfa91527e$exports = {};

$parcel$export($4e0c5fecfa91527e$exports, "PasskeyAuthenticatorVerifyOptions", () => $4e0c5fecfa91527e$export$7b01305e0ab33ea3);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyAuthenticatorEnrollmentOptions", () => $4e0c5fecfa91527e$export$1c8fd79a94eba09d);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyAuthenticatorVerifyResponse", () => $4e0c5fecfa91527e$export$96203b761bf7cadb);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyAuthenticatorEnrollmentResponse", () => $4e0c5fecfa91527e$export$49424bcfec0159aa);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyInitInput", () => $4e0c5fecfa91527e$export$8c48c68e39a52ea);
$parcel$export($4e0c5fecfa91527e$exports, "ResponsePasskeyInitAuthenticate", () => $4e0c5fecfa91527e$export$2741be68c8db9b43);
$parcel$export($4e0c5fecfa91527e$exports, "ResponsePasskeyInitLink", () => $4e0c5fecfa91527e$export$b968e9f4fdc5f0b2);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyLinkInput", () => $4e0c5fecfa91527e$export$9bd14c1a22f65d2);
$parcel$export($4e0c5fecfa91527e$exports, "PasskeyAuthenticateInput", () => $4e0c5fecfa91527e$export$8525265affc59e4b);
$parcel$export($4e0c5fecfa91527e$exports, "UnlinkPasskeyInput", () => $4e0c5fecfa91527e$export$1d32dd5b05e1d4dd);

const $4e0c5fecfa91527e$export$7b01305e0ab33ea3 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    challenge: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    timeout: (0, ($parcel$interopDefault($3LmS4$zod))).number().optional(),
    rp_id: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    allow_credentials: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).object({
        id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        type: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        transports: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).string()).optional()
    })).optional(),
    user_verification: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    extensions: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        app_id: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        cred_props: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional(),
        hmac_create_secret: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
    }).optional()
});
const $4e0c5fecfa91527e$export$1c8fd79a94eba09d = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    rp: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        name: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        id: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }),
    user: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        name: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        display_name: (0, ($parcel$interopDefault($3LmS4$zod))).string()
    }),
    challenge: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    pub_key_cred_params: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).object({
        alg: (0, ($parcel$interopDefault($3LmS4$zod))).number(),
        type: (0, ($parcel$interopDefault($3LmS4$zod))).literal("public-key")
    })),
    timeout: (0, ($parcel$interopDefault($3LmS4$zod))).number().optional(),
    exclude_credentials: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).object({
        id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        type: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        transports: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).string()).optional()
    })).optional(),
    authenticator_selection: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        authenticator_attachment: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        require_resident_key: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional(),
        resident_key: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        user_verification: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }).optional(),
    attestation: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    extensions: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        app_id: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        cred_props: (0, ($parcel$interopDefault($3LmS4$zod))).object({
            rk: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
    }).optional()
});
const $4e0c5fecfa91527e$export$96203b761bf7cadb = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    raw_id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    response: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        client_data_json: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        authenticator_data: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        signature: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        user_handle: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }),
    authenticator_attachment: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    client_extension_results: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        app_id: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional(),
        cred_props: (0, ($parcel$interopDefault($3LmS4$zod))).object({
            rk: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
    }),
    type: (0, ($parcel$interopDefault($3LmS4$zod))).literal("public-key")
});
const $4e0c5fecfa91527e$export$49424bcfec0159aa = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    raw_id: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    response: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        client_data_json: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        attestation_object: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        authenticator_data: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        // TODO: Type this to AuthenticatorTransportFuture[]
        transports: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).any()).optional(),
        public_key_algorithm: (0, ($parcel$interopDefault($3LmS4$zod))).number().optional(),
        public_key: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }),
    authenticator_attachment: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    client_extension_results: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        app_id: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional(),
        cred_props: (0, ($parcel$interopDefault($3LmS4$zod))).object({
            rk: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
        }).optional(),
        hmac_create_secret: (0, ($parcel$interopDefault($3LmS4$zod))).boolean().optional()
    }),
    type: (0, ($parcel$interopDefault($3LmS4$zod))).literal("public-key")
});
const $4e0c5fecfa91527e$export$8c48c68e39a52ea = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional(),
    token: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
}).strict();
const $4e0c5fecfa91527e$export$2741be68c8db9b43 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional(),
    options: $4e0c5fecfa91527e$export$7b01305e0ab33ea3
}).strict();
const $4e0c5fecfa91527e$export$b968e9f4fdc5f0b2 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional(),
    options: $4e0c5fecfa91527e$export$1c8fd79a94eba09d
}).strict();
const $4e0c5fecfa91527e$export$9bd14c1a22f65d2 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional(),
    authenticator_response: $4e0c5fecfa91527e$export$49424bcfec0159aa
}).strict();
const $4e0c5fecfa91527e$export$8525265affc59e4b = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional(),
    challenge: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
    authenticator_response: $4e0c5fecfa91527e$export$96203b761bf7cadb
}).strict();
const $4e0c5fecfa91527e$export$1d32dd5b05e1d4dd = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    credential_id: (0, ($parcel$interopDefault($3LmS4$zod))).string()
});


var $57535d9c1ccc4808$exports = {};

$parcel$export($57535d9c1ccc4808$exports, "RecoveryKeyMaterialInput", () => $57535d9c1ccc4808$export$5133f8b99ced310e);
$parcel$export($57535d9c1ccc4808$exports, "RecoveryKeyMaterialResponse", () => $57535d9c1ccc4808$export$52cacd413024ecd);
$parcel$export($57535d9c1ccc4808$exports, "OAuthAuthenticateRecoveryResponse", () => $57535d9c1ccc4808$export$8d23e6b066801647);
$parcel$export($57535d9c1ccc4808$exports, "OAuthInitRecoveryInput", () => $57535d9c1ccc4808$export$eb513813549270a);
$parcel$export($57535d9c1ccc4808$exports, "OAuthInitICloudRecoveryInput", () => $57535d9c1ccc4808$export$425e6a14ed44c52f);
$parcel$export($57535d9c1ccc4808$exports, "OAuthCallbackICloudExpoInput", () => $57535d9c1ccc4808$export$bec1efaa57e00983);
$parcel$export($57535d9c1ccc4808$exports, "OAuthCallbackICloudExpoResponse", () => $57535d9c1ccc4808$export$ea90bf451d6cb856);
$parcel$export($57535d9c1ccc4808$exports, "RecoveryConfigurationICloudInput", () => $57535d9c1ccc4808$export$dfb75001bba12cf3);
$parcel$export($57535d9c1ccc4808$exports, "RecoveryConfigurationICloudResponse", () => $57535d9c1ccc4808$export$a53156909e5ca052);



const $57535d9c1ccc4808$export$5133f8b99ced310e = (0, $3LmS4$zod.z).object({
    chain_type: (0, $3LmS4$zod.z).string().optional()
});
const $57535d9c1ccc4808$export$52cacd413024ecd = (0, $3LmS4$zod.z).object({
    recovery_type: (0, $3LmS4$zod.z).enum([
        "user_passcode_derived_recovery_key",
        "privy_passcode_derived_recovery_key",
        "privy_generated_recovery_key",
        "google_drive_recovery_secret",
        "icloud_recovery_secret"
    ]),
    recovery_key_derivation_salt: (0, $3LmS4$zod.z).string(),
    recovery_code: (0, $3LmS4$zod.z).string(),
    recovery_key: (0, $3LmS4$zod.z).string(),
    file_id: (0, $3LmS4$zod.z).string(),
    icloud_record_name: (0, $3LmS4$zod.z).string()
});
const $57535d9c1ccc4808$export$8d23e6b066801647 = (0, $3LmS4$zod.z).object({
    access_token: (0, $3LmS4$zod.z).string()
});
const $57535d9c1ccc4808$export$eb513813549270a = (0, $3LmS4$zod.z).object({
    redirect_to: (0, $3LmS4$zod.z).string({
        required_error: "Please provide a redirect_to value."
    }),
    token: (0, $3LmS4$zod.z).string().optional(),
    state_code: (0, $3LmS4$zod.z).string().optional(),
    code_challenge: (0, $3LmS4$zod.z).string().optional()
});
const $57535d9c1ccc4808$export$425e6a14ed44c52f = (0, $3LmS4$zod.z).object({
    client_type: (0, $3LmS4$zod.z).enum([
        "web",
        "expo-ios"
    ])
});
const $57535d9c1ccc4808$export$bec1efaa57e00983 = (0, $3LmS4$zod.z).object({
    /**
     * The authorization code from apple
     */ ckWebAuthToken: (0, $3LmS4$zod.z).string()
}).transform((input)=>({
        ckWebAuthToken: input.ckWebAuthToken
    })).catch(()=>{
    throw new (0, $3LmS4$privyioapibase.UnauthorizedError)("Missing ckWebAuthToken.", (0, $3LmS4$privyioapibase.PrivyErrorCode).INVALID_DATA);
});
const $57535d9c1ccc4808$export$ea90bf451d6cb856 = (0, $0dbe2ed1fbc05d05$export$2441567ff18621e0);
const $57535d9c1ccc4808$export$dfb75001bba12cf3 = (0, $3LmS4$zod.z).object({
    client_type: (0, $3LmS4$zod.z).enum([
        "web",
        "expo-ios"
    ])
});
const $57535d9c1ccc4808$export$a53156909e5ca052 = (0, $3LmS4$zod.z).object({
    api_token: (0, $3LmS4$zod.z).string(),
    container_identifier: (0, $3LmS4$zod.z).string(),
    environment: (0, $3LmS4$zod.z).string()
});


var $3cb5f11d58fcde38$exports = {};

$parcel$export($3cb5f11d58fcde38$exports, "SiweInput", () => $3cb5f11d58fcde38$export$1c8c55e22e1d8d9);
$parcel$export($3cb5f11d58fcde38$exports, "AuthenticateSiweInput", () => $3cb5f11d58fcde38$export$3435c4883fbaeb7f);
$parcel$export($3cb5f11d58fcde38$exports, "SiweAddressInput", () => $3cb5f11d58fcde38$export$b5b65f417dc17888);
$parcel$export($3cb5f11d58fcde38$exports, "SiweInitInput", () => $3cb5f11d58fcde38$export$6c957bbdcf62135c);
$parcel$export($3cb5f11d58fcde38$exports, "SiweNonce", () => $3cb5f11d58fcde38$export$8535d2f477730820);


const $3cb5f11d58fcde38$export$1c8c55e22e1d8d9 = (0, $3LmS4$zod.z).object({
    message: (0, $3LmS4$zod.z).string({
        invalid_type_error: "Invalid SIWE message"
    }),
    signature: (0, $3LmS4$zod.z).string({
        invalid_type_error: "Invalid SIWE signature"
    }),
    chainId: (0, $3LmS4$zod.z).string().max(41).nullable().default(null),
    walletClientType: (0, $3LmS4$zod.z).string().max(64).nullable().default(null),
    connectorType: (0, $3LmS4$zod.z).string().max(64).nullable().default(null)
});
const $3cb5f11d58fcde38$export$3435c4883fbaeb7f = $3cb5f11d58fcde38$export$1c8c55e22e1d8d9.merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $3cb5f11d58fcde38$export$b5b65f417dc17888 = (0, $3LmS4$zod.z).object({
    address: (0, $d69d14a89141bb36$export$e1ae011f7e31ddec)
});
const $3cb5f11d58fcde38$export$6c957bbdcf62135c = (0, $3LmS4$zod.z).object({
    address: (0, $d69d14a89141bb36$export$e1ae011f7e31ddec),
    token: (0, $3LmS4$zod.z).string().optional()
});
const $3cb5f11d58fcde38$export$8535d2f477730820 = (0, $3LmS4$zod.z).object({
    nonce: (0, $3LmS4$zod.z).string(),
    address: (0, $3LmS4$zod.z).string(),
    expires_at: (0, $3LmS4$zod.z).string()
});


var $71ff3083479ae275$exports = {};

$parcel$export($71ff3083479ae275$exports, "SiwsInput", () => $71ff3083479ae275$export$fa2883d1fd7d2f4d);
$parcel$export($71ff3083479ae275$exports, "AuthenticateSiwsInput", () => $71ff3083479ae275$export$defac42dc31eb54f);
$parcel$export($71ff3083479ae275$exports, "SiwsAddressInput", () => $71ff3083479ae275$export$767d61e052aa619c);
$parcel$export($71ff3083479ae275$exports, "SiwsInitInput", () => $71ff3083479ae275$export$334006f5cf1e2e10);
$parcel$export($71ff3083479ae275$exports, "SiwsNonce", () => $71ff3083479ae275$export$678e538743698961);


const $71ff3083479ae275$export$fa2883d1fd7d2f4d = (0, $3LmS4$zod.z).object({
    message: (0, $3LmS4$zod.z).string({
        invalid_type_error: "Invalid SIWS message"
    }),
    signature: (0, $3LmS4$zod.z).string({
        invalid_type_error: "Invalid SIWS signature"
    }),
    walletClientType: (0, $3LmS4$zod.z).string().max(64).nullable().default(null),
    connectorType: (0, $3LmS4$zod.z).string().max(64).nullable().default(null)
});
const $71ff3083479ae275$export$defac42dc31eb54f = $71ff3083479ae275$export$fa2883d1fd7d2f4d.merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $71ff3083479ae275$export$767d61e052aa619c = (0, $3LmS4$zod.z).object({
    address: (0, $d69d14a89141bb36$export$a6c60903ce1c32fc)
});
const $71ff3083479ae275$export$334006f5cf1e2e10 = (0, $3LmS4$zod.z).object({
    address: (0, $d69d14a89141bb36$export$a6c60903ce1c32fc),
    token: (0, $3LmS4$zod.z).string().optional()
});
const $71ff3083479ae275$export$678e538743698961 = (0, $3LmS4$zod.z).object({
    nonce: (0, $3LmS4$zod.z).string(),
    address: (0, $3LmS4$zod.z).string(),
    expires_at: (0, $3LmS4$zod.z).string()
});


var $9a680a15fcf32da7$exports = {};

$parcel$export($9a680a15fcf32da7$exports, "FarcasterSignerInitInput", () => $9a680a15fcf32da7$export$809802d572486db);
$parcel$export($9a680a15fcf32da7$exports, "FarcasterSignerInitResponse", () => $9a680a15fcf32da7$export$327461233dec5bf6);
$parcel$export($9a680a15fcf32da7$exports, "FarcasterSignerStatusResponse", () => $9a680a15fcf32da7$export$2a58492abd647af5);

const $9a680a15fcf32da7$export$809802d572486db = (0, $3LmS4$zod.z).object({
    // TODO: Down the line, we should support specifying developer's own app FID
    // request_fid: z.bigint().optional(),
    /** Must be an ed25519 key, NOT secp256k1 (ethereum native curve) */ ed25519_public_key: (0, $3LmS4$zod.z).string().refine((key)=>key.startsWith("0x") && key.length === 66),
    /**
   * The deadline for a signer request to succeed. After the deadline, the request will no longer
   * succeed. Defaults to 1 hour.
   */ deadline: (0, $3LmS4$zod.z).bigint().optional().default(()=>{
        const now = Math.floor(Date.now() / 1000);
        const oneHour = 3600;
        return BigInt(now + oneHour);
    })
});
const $9a680a15fcf32da7$export$327461233dec5bf6 = (0, $3LmS4$zod.z).discriminatedUnion("status", [
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("pending_approval"),
        signer_approval_url: (0, $3LmS4$zod.z).string()
    }),
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("approved"),
        /** The FID of the user who approved the signer */ fid: (0, $3LmS4$zod.z).bigint()
    }),
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("revoked"),
        /** The FID of the user who approved the signer */ fid: (0, $3LmS4$zod.z).bigint()
    })
]);
const $9a680a15fcf32da7$export$2a58492abd647af5 = (0, $3LmS4$zod.z).discriminatedUnion("status", [
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("pending_approval")
    }),
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("approved"),
        /** The FID of the user who approved the signer */ fid: (0, $3LmS4$zod.z).bigint()
    }),
    (0, $3LmS4$zod.z).object({
        /** The public key reflected back from the API
     * ie 0x2087e48968ca16a8954d0da041de84b66392b2821c2af42bc28aa079bcbe1dfe
     */ public_key: (0, $3LmS4$zod.z).string(),
        status: (0, $3LmS4$zod.z).literal("revoked"),
        /** The FID of the user who approved the signer */ fid: (0, $3LmS4$zod.z).bigint()
    })
]);


var $1f44e55a6d27396e$exports = {};

$parcel$export($1f44e55a6d27396e$exports, "VerifyPhoneInput", () => $1f44e55a6d27396e$export$1bbb8302b8922d52);
$parcel$export($1f44e55a6d27396e$exports, "AuthenticatePhoneInput", () => $1f44e55a6d27396e$export$d57783821ffbf377);
$parcel$export($1f44e55a6d27396e$exports, "PasswordlessSmsPhoneInput", () => $1f44e55a6d27396e$export$794b219acd817434);
$parcel$export($1f44e55a6d27396e$exports, "UnlinkPhoneInput", () => $1f44e55a6d27396e$export$63fc94c9f0d4eea8);
$parcel$export($1f44e55a6d27396e$exports, "UpdatePhoneInput", () => $1f44e55a6d27396e$export$bbf6b1d1316adec);



const $1f44e55a6d27396e$export$1bbb8302b8922d52 = (0, $3LmS4$zod.z).object({
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f).catch(()=>{
        throw new (0, $3LmS4$privyioapibase.LegacyInvalidInputError)("Invalid SMS verification code", (0, $3LmS4$privyioapibase.PrivyErrorCode).INVALID_CREDENTIALS);
    }),
    // Not a mistake, it's actually camelCase
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e).catch(()=>{
        throw new (0, $3LmS4$privyioapibase.LegacyInvalidInputError)("Invalid phone number", (0, $3LmS4$privyioapibase.PrivyErrorCode).INVALID_CREDENTIALS);
    })
});
const $1f44e55a6d27396e$export$d57783821ffbf377 = $1f44e55a6d27396e$export$1bbb8302b8922d52.merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $1f44e55a6d27396e$export$794b219acd817434 = (0, $3LmS4$zod.z).object({
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e).catch(()=>{
        throw new (0, $3LmS4$privyioapibase.LegacyInvalidInputError)("Invalid phone number", (0, $3LmS4$privyioapibase.PrivyErrorCode).INVALID_DATA);
    }),
    token: (0, $3LmS4$zod.z).string().optional()
});
const $1f44e55a6d27396e$export$63fc94c9f0d4eea8 = (0, $3LmS4$zod.z).object({
    // Not a mistake, it's actually camelCase
    phoneNumber: (0, $d69d14a89141bb36$export$3d623a7758cab61e)
});
const $1f44e55a6d27396e$export$bbf6b1d1316adec = (0, $3LmS4$zod.z).object({
    old_phone_number: (0, $d69d14a89141bb36$export$3d623a7758cab61e),
    new_phone_number: (0, $d69d14a89141bb36$export$3d623a7758cab61e),
    code: (0, $d69d14a89141bb36$export$6051224a4b788b5f)
});


var $9fe9ce7c5c0924f5$exports = {};

$parcel$export($9fe9ce7c5c0924f5$exports, "RefreshTokenInput", () => $9fe9ce7c5c0924f5$export$6aa3af8166ea5386);
$parcel$export($9fe9ce7c5c0924f5$exports, "OptionalRefreshTokenInput", () => $9fe9ce7c5c0924f5$export$3ea9bae16087a241);
$parcel$export($9fe9ce7c5c0924f5$exports, "ForkedToken", () => $9fe9ce7c5c0924f5$export$874b0538e4f42518);

const $9fe9ce7c5c0924f5$export$6aa3af8166ea5386 = (0, $3LmS4$zod.z).object({
    refresh_token: (0, $3LmS4$zod.z).string()
});
const $9fe9ce7c5c0924f5$export$3ea9bae16087a241 = $9fe9ce7c5c0924f5$export$6aa3af8166ea5386.partial();
const $9fe9ce7c5c0924f5$export$874b0538e4f42518 = (0, $3LmS4$zod.z).object({
    token: (0, $3LmS4$zod.z).string(),
    refresh_token: (0, $3LmS4$zod.z).string(),
    new_session_refresh_token: (0, $3LmS4$zod.z).string()
});


var $17c0cc089a701b88$exports = {};

$parcel$export($17c0cc089a701b88$exports, "EmailAccount", () => $17c0cc089a701b88$export$b291b2a8281f162e);
$parcel$export($17c0cc089a701b88$exports, "PhoneAccount", () => $17c0cc089a701b88$export$e461068a75eb0fb3);
$parcel$export($17c0cc089a701b88$exports, "EthereumAccount", () => $17c0cc089a701b88$export$e72ed3751242ab66);
$parcel$export($17c0cc089a701b88$exports, "SolanaAccount", () => $17c0cc089a701b88$export$d08a38b431a135bc);
$parcel$export($17c0cc089a701b88$exports, "FarcasterAccount", () => $17c0cc089a701b88$export$70b11e99555e84cb);
$parcel$export($17c0cc089a701b88$exports, "PasskeyAccount", () => $17c0cc089a701b88$export$c7ead6bbfb2c42a0);
$parcel$export($17c0cc089a701b88$exports, "TelegramAccount", () => $17c0cc089a701b88$export$70ddd50daefc295f);
$parcel$export($17c0cc089a701b88$exports, "EthereumEmbeddedWalletAccount", () => $17c0cc089a701b88$export$42e79fc33a58a2f5);
$parcel$export($17c0cc089a701b88$exports, "SolanaEmbeddedWalletAccount", () => $17c0cc089a701b88$export$734c44f2e990ba65);
$parcel$export($17c0cc089a701b88$exports, "GoogleOauthAccount", () => $17c0cc089a701b88$export$ab3c40bebc4d6f36);
$parcel$export($17c0cc089a701b88$exports, "TwitterOauthAccount", () => $17c0cc089a701b88$export$31cc32c7bc65c2e0);
$parcel$export($17c0cc089a701b88$exports, "DiscordOauthAccount", () => $17c0cc089a701b88$export$cb35596f96141125);
$parcel$export($17c0cc089a701b88$exports, "GithubOauthAccount", () => $17c0cc089a701b88$export$55430b29b82d98a8);
$parcel$export($17c0cc089a701b88$exports, "LinkedInOauthAccount", () => $17c0cc089a701b88$export$25b982740d2e431c);
$parcel$export($17c0cc089a701b88$exports, "SpotifyOauthAccount", () => $17c0cc089a701b88$export$260592be4b1062f4);
$parcel$export($17c0cc089a701b88$exports, "InstagramOauthAccount", () => $17c0cc089a701b88$export$c683d35f732c1912);
$parcel$export($17c0cc089a701b88$exports, "TiktokOauthAccount", () => $17c0cc089a701b88$export$5e8143c3f32ac816);
$parcel$export($17c0cc089a701b88$exports, "AppleOauthAccount", () => $17c0cc089a701b88$export$67af8a2e34850611);
$parcel$export($17c0cc089a701b88$exports, "CustomJwtAccount", () => $17c0cc089a701b88$export$5da87a1cf22b57d0);
$parcel$export($17c0cc089a701b88$exports, "CrossAppEmbeddedWallet", () => $17c0cc089a701b88$export$ccd62fbf4b025fee);
$parcel$export($17c0cc089a701b88$exports, "CrossAppSmartWallet", () => $17c0cc089a701b88$export$8f1f4ad4f8276ad7);
$parcel$export($17c0cc089a701b88$exports, "CrossAppAccount", () => $17c0cc089a701b88$export$564035ac82295c40);
$parcel$export($17c0cc089a701b88$exports, "LinkedAccount", () => $17c0cc089a701b88$export$70ab2a322f92399);
$parcel$export($17c0cc089a701b88$exports, "SmsMfaMethod", () => $17c0cc089a701b88$export$3250154351927c42);
$parcel$export($17c0cc089a701b88$exports, "TotpMfaMethod", () => $17c0cc089a701b88$export$27837b14d6717da8);
$parcel$export($17c0cc089a701b88$exports, "PasskeyMfaMethod", () => $17c0cc089a701b88$export$dd0fcfda522da705);
$parcel$export($17c0cc089a701b88$exports, "LinkedMfaMethod", () => $17c0cc089a701b88$export$7d2a870b119c9ff4);
$parcel$export($17c0cc089a701b88$exports, "User", () => $17c0cc089a701b88$export$1f44aaf2ec115b54);
$parcel$export($17c0cc089a701b88$exports, "SessionUpdateActionEnum", () => $17c0cc089a701b88$export$bbd8e1b014079dc1);
$parcel$export($17c0cc089a701b88$exports, "AuthenticatedUser", () => $17c0cc089a701b88$export$7de20aee312d6767);
$parcel$export($17c0cc089a701b88$exports, "LoggedOutUser", () => $17c0cc089a701b88$export$8339543958fe900f);
$parcel$export($17c0cc089a701b88$exports, "MaybeUser", () => $17c0cc089a701b88$export$2b6251d563a4e626);


const $17c0cc089a701b88$export$b291b2a8281f162e = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("email"),
    address: (0, $3LmS4$zod.z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$e461068a75eb0fb3 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("phone"),
    phoneNumber: (0, $3LmS4$zod.z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$e72ed3751242ab66 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("wallet"),
    address: (0, $3LmS4$zod.z).string(),
    chain_id: (0, $3LmS4$zod.z).string().optional(),
    chain_type: (0, $3LmS4$zod.z).literal("ethereum"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $3LmS4$zod.z).literal("unknown"),
    wallet_client_type: (0, $3LmS4$zod.z).string().optional(),
    connector_type: (0, $3LmS4$zod.z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$d08a38b431a135bc = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("wallet"),
    address: (0, $3LmS4$zod.z).string(),
    chain_type: (0, $3LmS4$zod.z).literal("solana"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $3LmS4$zod.z).literal("unknown"),
    wallet_client_type: (0, $3LmS4$zod.z).string().optional(),
    connector_type: (0, $3LmS4$zod.z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$70b11e99555e84cb = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("farcaster"),
    fid: (0, $3LmS4$zod.z).number(),
    owner_address: (0, $d69d14a89141bb36$export$e1ae011f7e31ddec),
    username: (0, $3LmS4$zod.z).string().optional(),
    display_name: (0, $3LmS4$zod.z).string().optional(),
    bio: (0, $3LmS4$zod.z).string().optional(),
    // The `profile_picture` interface is deprecated in favor of `profile_picture_url`
    // as of `react-auth@1.55.2`.
    profile_picture: (0, $3LmS4$zod.z).string().optional(),
    profile_picture_url: (0, $3LmS4$zod.z).string().optional(),
    homepage_url: (0, $3LmS4$zod.z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    signer_public_key: (0, $3LmS4$zod.z).string().optional()
});
const $17c0cc089a701b88$export$c7ead6bbfb2c42a0 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("passkey"),
    created_with_browser: (0, $3LmS4$zod.z).string().optional(),
    created_with_os: (0, $3LmS4$zod.z).string().optional(),
    created_with_device: (0, $3LmS4$zod.z).string().optional(),
    credential_id: (0, $3LmS4$zod.z).string(),
    authenticator_name: (0, $3LmS4$zod.z).string().optional(),
    enrolled_in_mfa: (0, $3LmS4$zod.z).boolean(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$70ddd50daefc295f = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("telegram"),
    telegram_user_id: (0, $3LmS4$zod.z).string(),
    first_name: (0, $3LmS4$zod.z).string().nullish(),
    last_name: (0, $3LmS4$zod.z).string().nullish(),
    username: (0, $3LmS4$zod.z).string().nullish(),
    photo_url: (0, $3LmS4$zod.z).string().nullish(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
}).transform((value)=>({
        ...value,
        // TODO: We keep this for backwards compatibility. Remove this after we make sure no one uses this version of the response
        telegramUserId: value.telegram_user_id,
        firstName: value.first_name
    }));
const $17c0cc089a701b88$export$42e79fc33a58a2f5 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("wallet"),
    address: (0, $3LmS4$zod.z).string(),
    imported: (0, $3LmS4$zod.z).boolean().default(false),
    wallet_index: (0, $3LmS4$zod.z).number(),
    // CAIP-2 chain ID for Ethereum mainnet: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md
    chain_id: (0, $3LmS4$zod.z).literal("eip155:1"),
    chain_type: (0, $3LmS4$zod.z).literal("ethereum"),
    /**
   * @deprecated Use `wallet_client_type` instead.
   */ wallet_client: (0, $3LmS4$zod.z).literal("privy"),
    wallet_client_type: (0, $3LmS4$zod.z).literal("privy"),
    connector_type: (0, $3LmS4$zod.z).literal("embedded"),
    recovery_method: (0, $3LmS4$zod.z).union([
        (0, $3LmS4$zod.z).literal("privy"),
        (0, $3LmS4$zod.z).literal("user-passcode"),
        (0, $3LmS4$zod.z).literal("google-drive"),
        (0, $3LmS4$zod.z).literal("icloud")
    ]),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$734c44f2e990ba65 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("wallet"),
    address: (0, $3LmS4$zod.z).string(),
    public_key: (0, $3LmS4$zod.z).string(),
    chain_type: (0, $3LmS4$zod.z).literal("solana"),
    // CAIP-2 chain ID for Solana mainnet based on CAIP-10 account address definition: https://namespaces.chainagnostic.org/solana/caip10
    chain_id: (0, $3LmS4$zod.z).literal("solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp"),
    imported: (0, $3LmS4$zod.z).boolean().default(false),
    wallet_client_type: (0, $3LmS4$zod.z).literal("privy"),
    connector_type: (0, $3LmS4$zod.z).literal("embedded"),
    // For now we only support automatic recovery, but eventually will support the other methods.
    // The EVM accounts connected to this seed must ALSO be recoverable by the same method.
    recovery_method: (0, $3LmS4$zod.z).literal("privy"),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$ab3c40bebc4d6f36 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("google_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    email: (0, $3LmS4$zod.z).string(),
    name: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$31cc32c7bc65c2e0 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("twitter_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullable(),
    name: (0, $3LmS4$zod.z).string().nullable(),
    profile_picture_url: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$cb35596f96141125 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("discord_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullable(),
    email: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$55430b29b82d98a8 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("github_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullable(),
    name: (0, $3LmS4$zod.z).string().nullable(),
    email: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$25b982740d2e431c = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("linkedin_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    name: (0, $3LmS4$zod.z).string().optional(),
    email: (0, $3LmS4$zod.z).string().nullable(),
    vanity_name: (0, $3LmS4$zod.z).string().optional(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$260592be4b1062f4 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("spotify_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    email: (0, $3LmS4$zod.z).string().nullable(),
    name: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$c683d35f732c1912 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("instagram_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$5e8143c3f32ac816 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("tiktok_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullable(),
    name: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$67af8a2e34850611 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("apple_oauth"),
    subject: (0, $3LmS4$zod.z).string(),
    email: (0, $3LmS4$zod.z).string().nullable(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$5da87a1cf22b57d0 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("custom_auth"),
    custom_user_id: (0, $3LmS4$zod.z).string(),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$ccd62fbf4b025fee = (0, $3LmS4$zod.z).object({
    address: (0, $3LmS4$zod.z).string()
});
const $17c0cc089a701b88$export$8f1f4ad4f8276ad7 = (0, $3LmS4$zod.z).object({
    address: (0, $3LmS4$zod.z).string()
});
const $17c0cc089a701b88$export$564035ac82295c40 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("cross_app"),
    subject: (0, $3LmS4$zod.z).string(),
    provider_app_id: (0, $3LmS4$zod.z).string(),
    embedded_wallets: (0, $3LmS4$zod.z).array($17c0cc089a701b88$export$ccd62fbf4b025fee),
    smart_wallets: (0, $3LmS4$zod.z).array($17c0cc089a701b88$export$8f1f4ad4f8276ad7),
    /** @deprecated use `first_verified_at` instead. */ verified_at: (0, $3LmS4$zod.z).number(),
    first_verified_at: (0, $3LmS4$zod.z).number().nullable(),
    latest_verified_at: (0, $3LmS4$zod.z).number().nullable()
});
const $17c0cc089a701b88$export$70ab2a322f92399 = (0, $3LmS4$zod.z).union([
    $17c0cc089a701b88$export$b291b2a8281f162e,
    $17c0cc089a701b88$export$e461068a75eb0fb3,
    $17c0cc089a701b88$export$e72ed3751242ab66,
    $17c0cc089a701b88$export$d08a38b431a135bc,
    $17c0cc089a701b88$export$42e79fc33a58a2f5,
    $17c0cc089a701b88$export$734c44f2e990ba65,
    $17c0cc089a701b88$export$ab3c40bebc4d6f36,
    $17c0cc089a701b88$export$31cc32c7bc65c2e0,
    $17c0cc089a701b88$export$cb35596f96141125,
    $17c0cc089a701b88$export$55430b29b82d98a8,
    $17c0cc089a701b88$export$260592be4b1062f4,
    $17c0cc089a701b88$export$c683d35f732c1912,
    $17c0cc089a701b88$export$5e8143c3f32ac816,
    $17c0cc089a701b88$export$25b982740d2e431c,
    $17c0cc089a701b88$export$67af8a2e34850611,
    $17c0cc089a701b88$export$5da87a1cf22b57d0,
    $17c0cc089a701b88$export$70b11e99555e84cb,
    $17c0cc089a701b88$export$c7ead6bbfb2c42a0,
    $17c0cc089a701b88$export$70ddd50daefc295f,
    $17c0cc089a701b88$export$564035ac82295c40
]);
const $17c0cc089a701b88$export$3250154351927c42 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("sms"),
    verified_at: (0, $3LmS4$zod.z).number()
});
const $17c0cc089a701b88$export$27837b14d6717da8 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("totp"),
    verified_at: (0, $3LmS4$zod.z).number()
});
const $17c0cc089a701b88$export$dd0fcfda522da705 = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("passkey"),
    verified_at: (0, $3LmS4$zod.z).number()
});
const $17c0cc089a701b88$export$7d2a870b119c9ff4 = (0, $3LmS4$zod.z).union([
    $17c0cc089a701b88$export$3250154351927c42,
    $17c0cc089a701b88$export$27837b14d6717da8,
    $17c0cc089a701b88$export$dd0fcfda522da705
]);
const $17c0cc089a701b88$export$1f44aaf2ec115b54 = (0, $3LmS4$zod.z).object({
    id: (0, $3LmS4$zod.z).string(),
    linked_accounts: (0, $3LmS4$zod.z).array($17c0cc089a701b88$export$70ab2a322f92399),
    mfa_methods: (0, $3LmS4$zod.z).array($17c0cc089a701b88$export$7d2a870b119c9ff4),
    created_at: (0, $3LmS4$zod.z).number(),
    has_accepted_terms: (0, $3LmS4$zod.z).boolean(),
    is_guest: (0, $3LmS4$zod.z).boolean(),
    custom_metadata: (0, $3LmS4$zod.z).record((0, $3LmS4$zod.z).string(), (0, $3LmS4$zod.z).string()).optional()
});
const $17c0cc089a701b88$export$bbd8e1b014079dc1 = (0, $3LmS4$zod.z).enum([
    "set",
    "ignore",
    "clear"
]);
const $17c0cc089a701b88$export$7de20aee312d6767 = (0, $3LmS4$zod.z).object({
    user: $17c0cc089a701b88$export$1f44aaf2ec115b54,
    token: (0, $3LmS4$zod.z).string().min(1).nullable(),
    refresh_token: (0, $3LmS4$zod.z).string().min(1).nullable(),
    identity_token: (0, $3LmS4$zod.z).string().min(1).optional(),
    is_new_user: (0, $3LmS4$zod.z).boolean().optional(),
    oauth_tokens: (0, $3LmS4$zod.z).object({
        provider: (0, $3LmS4$zod.z).string(),
        access_token: (0, $3LmS4$zod.z).string(),
        // The following additional parameters are optional as some OAuth providers may not return them.
        // Being conservative with marking them optional for now.
        access_token_expires_in_seconds: (0, $3LmS4$zod.z).number().optional(),
        refresh_token: (0, $3LmS4$zod.z).string().optional(),
        refresh_token_expires_in_seconds: (0, $3LmS4$zod.z).number().optional(),
        scopes: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()).optional()
    }).optional(),
    // TODO: deprecate once Deform moves off of the legacy method
    oauth_provider_tokens: (0, $3LmS4$zod.z).object({
        provider: (0, $3LmS4$zod.z).string(),
        access_token: (0, $3LmS4$zod.z).string(),
        // The following additional parameters are optional as some OAuth providers may not return them.
        // Being conservative with marking them optional for now.
        access_token_expires_in_seconds: (0, $3LmS4$zod.z).number().optional(),
        refresh_token: (0, $3LmS4$zod.z).string().optional(),
        scopes: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).string()).optional()
    }).optional(),
    /**
   * Instructs the client on how to handle tokens received from /sessions endpoint
   * Seeks to mimic the behavior of cookies with manual storage
   *
   * - `set`: _write both tokens to storage_
   * - `clear`: _remove tokens from storage_
   * - `ignore`: _leave tokens storage unchanged_
   */ session_update_action: $17c0cc089a701b88$export$bbd8e1b014079dc1
});
const $17c0cc089a701b88$export$8339543958fe900f = (0, $3LmS4$zod.z).object({
    user: (0, $3LmS4$zod.z).null(),
    token: (0, $3LmS4$zod.z).null(),
    refresh_token: (0, $3LmS4$zod.z).null(),
    /** Instructs the client to clear any persisted tokens */ session_update_action: (0, $3LmS4$zod.z).literal($17c0cc089a701b88$export$bbd8e1b014079dc1.Values.clear)
});
const $17c0cc089a701b88$export$2b6251d563a4e626 = (0, $3LmS4$zod.z).union([
    $17c0cc089a701b88$export$7de20aee312d6767,
    $17c0cc089a701b88$export$8339543958fe900f
]);


var $4c07922918dc06b2$exports = {};

$parcel$export($4c07922918dc06b2$exports, "MfaPasskeyInitInput", () => $4c07922918dc06b2$export$2d6bddbbdba11994);
$parcel$export($4c07922918dc06b2$exports, "ResponsePasskeyInitMfa", () => $4c07922918dc06b2$export$94c076e2d3cd7355);
$parcel$export($4c07922918dc06b2$exports, "MfaPasskeyVerifyInput", () => $4c07922918dc06b2$export$afc24607eeb5f33e);
$parcel$export($4c07922918dc06b2$exports, "MfaEnrollmentPasskeyInput", () => $4c07922918dc06b2$export$3c909b91f5e6545f);


const $4c07922918dc06b2$export$2d6bddbbdba11994 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional()
}).strict();
const $4c07922918dc06b2$export$94c076e2d3cd7355 = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    options: (0, $4e0c5fecfa91527e$export$7b01305e0ab33ea3)
}).strict();
const $4c07922918dc06b2$export$afc24607eeb5f33e = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    authenticator_response: (0, $4e0c5fecfa91527e$export$96203b761bf7cadb),
    relying_party: (0, ($parcel$interopDefault($3LmS4$zod))).string().url().optional()
}).strict();
const $4c07922918dc06b2$export$3c909b91f5e6545f = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    credential_ids: (0, ($parcel$interopDefault($3LmS4$zod))).array((0, ($parcel$interopDefault($3LmS4$zod))).string())
});


var $441f2ef476215a7d$exports = {};

$parcel$export($441f2ef476215a7d$exports, "TelegramAuthenticateInput", () => $441f2ef476215a7d$export$fb9405d8cbdd092d);
$parcel$export($441f2ef476215a7d$exports, "TelegramLinkInput", () => $441f2ef476215a7d$export$303a24e5db03ece6);
$parcel$export($441f2ef476215a7d$exports, "TelegramUnlinkInput", () => $441f2ef476215a7d$export$aad710dcc06a76fc);


const $441f2ef476215a7d$export$fb9405d8cbdd092d = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    captcha_token: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
    telegram_auth_result: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        id: (0, ($parcel$interopDefault($3LmS4$zod))).number(),
        first_name: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        auth_date: (0, ($parcel$interopDefault($3LmS4$zod))).number(),
        hash: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        username: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        last_name: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        photo_url: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }).optional(),
    telegram_web_app_data: (0, ($parcel$interopDefault($3LmS4$zod))).object({
        // query id provided by Telegram. It's present when logging in from a chat button but it's not when logging in from a direct link to the mini-app
        query_id: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        // This is the timestamp the auth result was verified at, in seconds from epoch
        auth_date: (0, ($parcel$interopDefault($3LmS4$zod))).number(),
        // hash provided by Telegram to verify the payload is authentic
        hash: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        // user data provided by Telegram, it is a JSON string with a shape similar to TelegramAuthResult
        user: (0, ($parcel$interopDefault($3LmS4$zod))).string(),
        // chat instance provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        chat_instance: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        // chat type provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        chat_type: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional(),
        // start param provided by Telegram. It's present when logging in from a direct link to the mini-app but not when logging in from a chat button
        start_param: (0, ($parcel$interopDefault($3LmS4$zod))).string().optional()
    }).optional()
}).merge((0, $d69d14a89141bb36$export$1d53282181ae94da));
const $441f2ef476215a7d$export$303a24e5db03ece6 = $441f2ef476215a7d$export$fb9405d8cbdd092d.omit({
    telegram_web_app_data: true
}).required({
    telegram_auth_result: true
});
const $441f2ef476215a7d$export$aad710dcc06a76fc = (0, ($parcel$interopDefault($3LmS4$zod))).object({
    telegram_user_id: (0, ($parcel$interopDefault($3LmS4$zod))).string()
});


var $9999926dec3dbd87$exports = {};

$parcel$export($9999926dec3dbd87$exports, "CrossAppConnectionsResponse", () => $9999926dec3dbd87$export$39ad28598f91ae95);

const $9999926dec3dbd87$export$39ad28598f91ae95 = (0, $3LmS4$zod.z).object({
    connections: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).object({
        provider_app_id: (0, $3LmS4$zod.z).string(),
        provider_app_name: (0, $3LmS4$zod.z).string(),
        provider_app_icon_url: (0, $3LmS4$zod.z).string().url().nullable(),
        provider_app_custom_api_url: (0, $3LmS4$zod.z).string().url().nullable(),
        read_only: (0, $3LmS4$zod.z).boolean()
    }))
});


var $104ad8f0cb58a420$exports = {};

$parcel$export($104ad8f0cb58a420$exports, "AuthorizationCodeInput", () => $104ad8f0cb58a420$export$b20be2048fbb4ed4);

const $104ad8f0cb58a420$export$b20be2048fbb4ed4 = (0, $3LmS4$zod.z).object({
    redirect_to: (0, $3LmS4$zod.z).string(),
    state: (0, $3LmS4$zod.z).string(),
    code_challenge: (0, $3LmS4$zod.z).string()
});


var $35c7f086131bd89c$exports = {};

$parcel$export($35c7f086131bd89c$exports, "EmailIdTokenEntry", () => $35c7f086131bd89c$export$783ee7c583b70bea);
$parcel$export($35c7f086131bd89c$exports, "PhoneIdTokenEntry", () => $35c7f086131bd89c$export$5050d0079ef11bc6);
$parcel$export($35c7f086131bd89c$exports, "EthereumIdTokenEntry", () => $35c7f086131bd89c$export$2986f977b4f702ee);
$parcel$export($35c7f086131bd89c$exports, "FarcasterIdTokenEntry", () => $35c7f086131bd89c$export$7d91b89a98a32206);
$parcel$export($35c7f086131bd89c$exports, "PasskeyIdTokenEntry", () => $35c7f086131bd89c$export$3da0109396521ad1);
$parcel$export($35c7f086131bd89c$exports, "TelegramIdTokenEntry", () => $35c7f086131bd89c$export$883001ac4b913adc);
$parcel$export($35c7f086131bd89c$exports, "EthereumEmbeddedWalletIdTokenEntry", () => $35c7f086131bd89c$export$19eeeea304fd874e);
$parcel$export($35c7f086131bd89c$exports, "SolanaEmbeddedWalletIdTokenEntry", () => $35c7f086131bd89c$export$fb08b43dc8cf7e59);
$parcel$export($35c7f086131bd89c$exports, "GoogleOauthIdTokenEntry", () => $35c7f086131bd89c$export$10bcfd05b1db0bfb);
$parcel$export($35c7f086131bd89c$exports, "TwitterOauthIdTokenEntry", () => $35c7f086131bd89c$export$5cf36acae8eed78);
$parcel$export($35c7f086131bd89c$exports, "DiscordOauthIdTokenEntry", () => $35c7f086131bd89c$export$18e770e71ddcac8b);
$parcel$export($35c7f086131bd89c$exports, "GithubOauthIdTokenEntry", () => $35c7f086131bd89c$export$2fd8fb76725390c8);
$parcel$export($35c7f086131bd89c$exports, "LinkedInOauthIdTokenEntry", () => $35c7f086131bd89c$export$7325b1b9ff5b138f);
$parcel$export($35c7f086131bd89c$exports, "SpotifyOauthIdTokenEntry", () => $35c7f086131bd89c$export$5991ad07a5ac6061);
$parcel$export($35c7f086131bd89c$exports, "InstagramOauthIdTokenEntry", () => $35c7f086131bd89c$export$d50f791175bcd25a);
$parcel$export($35c7f086131bd89c$exports, "TiktokOauthIdTokenEntry", () => $35c7f086131bd89c$export$ad548be3bcd74188);
$parcel$export($35c7f086131bd89c$exports, "AppleOauthIdTokenEntry", () => $35c7f086131bd89c$export$b8a150e8882d7617);
$parcel$export($35c7f086131bd89c$exports, "CustomJwtIdTokenEntry", () => $35c7f086131bd89c$export$4a1709b5a6406ed8);
$parcel$export($35c7f086131bd89c$exports, "CrossAppIdTokenEntry", () => $35c7f086131bd89c$export$23abd82a93e245c8);
// This file contains zod schemas for the linked account entries that appear with the `user` claim of
// Privy's identity token. These are a subset of the full linked account object that only contains
// critical data about the account (e.g. type and address for email). When developers request more fields
// about an account to be returned in the identity token, we can add them to the schemas here.


const $35c7f086131bd89c$export$783ee7c583b70bea = (0, $17c0cc089a701b88$export$b291b2a8281f162e).pick({
    type: true,
    address: true
});
const $35c7f086131bd89c$export$5050d0079ef11bc6 = (0, $17c0cc089a701b88$export$e461068a75eb0fb3).pick({
    type: true,
    phoneNumber: true
}).transform(({ type: type, phoneNumber: phoneNumber })=>({
        type: type,
        phone_number: phoneNumber
    }));
const $35c7f086131bd89c$export$2986f977b4f702ee = (0, $17c0cc089a701b88$export$e72ed3751242ab66).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $35c7f086131bd89c$export$7d91b89a98a32206 = (0, $17c0cc089a701b88$export$70b11e99555e84cb).pick({
    type: true,
    fid: true,
    username: true
});
const $35c7f086131bd89c$export$3da0109396521ad1 = (0, $17c0cc089a701b88$export$c7ead6bbfb2c42a0).pick({
    type: true,
    credential_id: true
});
const $35c7f086131bd89c$export$883001ac4b913adc = (0, $3LmS4$zod.z).object({
    type: (0, $3LmS4$zod.z).literal("telegram"),
    telegram_user_id: (0, $3LmS4$zod.z).string(),
    username: (0, $3LmS4$zod.z).string().nullish()
});
const $35c7f086131bd89c$export$19eeeea304fd874e = (0, $17c0cc089a701b88$export$42e79fc33a58a2f5).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $35c7f086131bd89c$export$fb08b43dc8cf7e59 = (0, $17c0cc089a701b88$export$734c44f2e990ba65).pick({
    type: true,
    address: true,
    chain_type: true,
    wallet_client_type: true
});
const $35c7f086131bd89c$export$10bcfd05b1db0bfb = (0, $17c0cc089a701b88$export$ab3c40bebc4d6f36).pick({
    type: true,
    subject: true,
    email: true
});
const $35c7f086131bd89c$export$5cf36acae8eed78 = (0, $17c0cc089a701b88$export$31cc32c7bc65c2e0).pick({
    type: true,
    subject: true,
    username: true
});
const $35c7f086131bd89c$export$18e770e71ddcac8b = (0, $17c0cc089a701b88$export$cb35596f96141125).pick({
    type: true,
    subject: true,
    username: true
});
const $35c7f086131bd89c$export$2fd8fb76725390c8 = (0, $17c0cc089a701b88$export$55430b29b82d98a8).pick({
    type: true,
    subject: true,
    username: true
});
const $35c7f086131bd89c$export$7325b1b9ff5b138f = (0, $17c0cc089a701b88$export$25b982740d2e431c).pick({
    type: true,
    subject: true,
    email: true
});
const $35c7f086131bd89c$export$5991ad07a5ac6061 = (0, $17c0cc089a701b88$export$260592be4b1062f4).pick({
    type: true,
    subject: true,
    email: true
});
const $35c7f086131bd89c$export$d50f791175bcd25a = (0, $17c0cc089a701b88$export$c683d35f732c1912).pick({
    type: true,
    subject: true,
    username: true
});
const $35c7f086131bd89c$export$ad548be3bcd74188 = (0, $17c0cc089a701b88$export$5e8143c3f32ac816).pick({
    type: true,
    subject: true,
    username: true
});
const $35c7f086131bd89c$export$b8a150e8882d7617 = (0, $17c0cc089a701b88$export$67af8a2e34850611).pick({
    type: true,
    subject: true,
    email: true
});
const $35c7f086131bd89c$export$4a1709b5a6406ed8 = (0, $17c0cc089a701b88$export$5da87a1cf22b57d0).pick({
    type: true,
    custom_user_id: true
});
const $35c7f086131bd89c$export$23abd82a93e245c8 = (0, $17c0cc089a701b88$export$564035ac82295c40).pick({
    type: true,
    subject: true,
    provider_app_id: true,
    embedded_wallets: true,
    smart_wallets: true
});


var $1408493d32450b5a$exports = {};

$parcel$export($1408493d32450b5a$exports, "CoinbaseOnRampInitInput", () => $1408493d32450b5a$export$f01d538aa036a41a);
$parcel$export($1408493d32450b5a$exports, "CoinbaseOnRampInitResponse", () => $1408493d32450b5a$export$4731ee7c27e49d6e);
$parcel$export($1408493d32450b5a$exports, "CoinbaseOnRampStatusResponse", () => $1408493d32450b5a$export$146ab3c80d47ce12);


const $1408493d32450b5a$var$Blockchain = (0, $3LmS4$zod.z).enum([
    "ethereum",
    "bitcoin",
    "base",
    "avacchain",
    "optimism",
    "solana",
    "polygon",
    "arbitrum",
    "stellar"
]);
// Assets must be capitalized to work with Coinbase Onramp. We preserve 'eth' here for backwards
// compatibility with older SDKs when Coinbase Onramp did not require capitalization
const $1408493d32450b5a$var$Assets = (0, $3LmS4$zod.z).enum([
    "eth",
    "ETH",
    "USDC"
]).transform((asset)=>asset.toUpperCase());
const $1408493d32450b5a$export$f01d538aa036a41a = (0, $3LmS4$zod.z).object({
    addresses: (0, $3LmS4$zod.z).array((0, $3LmS4$zod.z).object({
        address: (0, $d69d14a89141bb36$export$e1ae011f7e31ddec),
        blockchains: (0, $3LmS4$zod.z).array($1408493d32450b5a$var$Blockchain)
    })).max(1).min(1),
    assets: (0, $3LmS4$zod.z).array($1408493d32450b5a$var$Assets).optional()
});
const $1408493d32450b5a$export$4731ee7c27e49d6e = (0, $3LmS4$zod.z).object({
    app_id: (0, $3LmS4$zod.z).string(),
    session_token: (0, $3LmS4$zod.z).string(),
    channel_id: (0, $3LmS4$zod.z).string(),
    partner_user_id: (0, $3LmS4$zod.z).string()
});
const $1408493d32450b5a$export$146ab3c80d47ce12 = (0, $3LmS4$zod.z).object({
    status: (0, $3LmS4$zod.z).enum([
        "pending",
        "success",
        "failure"
    ])
});


$parcel$exportWildcard($7d14bc6e1da56942$exports, $6d13bfa775813546$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $920ac747dab06b9c$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $8a7996e6cd7430a9$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $d2ef8d1854e48841$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $a32571b6d32f2bac$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $d69d14a89141bb36$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $ba5673beb0ead784$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $1e7d24a9b3fdde80$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $566f5ba79f307877$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $f7e0547cf75f2094$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $63a1d8c3bf348ff9$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $1a93a9315817630b$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $0dbe2ed1fbc05d05$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $4e0c5fecfa91527e$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $57535d9c1ccc4808$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $3cb5f11d58fcde38$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $71ff3083479ae275$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $9a680a15fcf32da7$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $1f44e55a6d27396e$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $9fe9ce7c5c0924f5$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $17c0cc089a701b88$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $4c07922918dc06b2$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $441f2ef476215a7d$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $9999926dec3dbd87$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $104ad8f0cb58a420$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $35c7f086131bd89c$exports);
$parcel$exportWildcard($7d14bc6e1da56942$exports, $1408493d32450b5a$exports);


$parcel$exportWildcard(module.exports, $4b4e9f0316150d93$exports);
$parcel$exportWildcard(module.exports, $7d14bc6e1da56942$exports);
